// @prisma/client/internal/class.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * WARNING: This is an internal file that is subject to change!
 *
 * 🛑 Under no circumstances should you import this file directly! 🛑
 *
 * Please import the `PrismaClient` class from the `client.ts` file instead.
 */

import * as runtime from "@prisma/client/runtime/client"
import type * as Prisma from "./prismaNamespace"


const config: runtime.GetPrismaClientConfig = {
  "previewFeatures": [],
  "clientVersion": "7.2.0",
  "engineVersion": "0c8ef2ce45c83248ab3df073180d5eda9e8be7a3",
  "activeProvider": "sqlite",
  "inlineSchema": "// This is your Prisma schema file,\n// learn more about it in the docs: https://pris.ly/d/prisma-schema\n\ngenerator client {\n  provider = \"prisma-client\"\n  output   = \"node_modules/@prisma/client\"\n}\n\ndatasource db {\n  provider = \"sqlite\"\n}\n\nmodel Evento {\n  id        Int            @id @default(autoincrement())\n  sigla     String         @unique\n  descricao String\n  edicoes   EdicaoEvento[]\n}\n\nmodel EdicaoEvento {\n  id        Int       @id @default(autoincrement())\n  ano       Int\n  eventoId  Int\n  evento    Evento    @relation(fields: [eventoId], references: [id], onDelete: Cascade)\n  artigos   Artigo[]\n  revisores Revisor[]\n\n  @@unique([eventoId, ano])\n}\n\nmodel Artigo {\n  id          Int          @id @default(autoincrement())\n  titulo      String\n  autores     Json // Array de strings (ex: [\"João Silva\", \"Maria Santos\"])\n  resumo      String\n  areas       Json // Array de strings (ex: [\"Banco de dados\", \"IA\"])\n  pdfUrl      String? // URL do arquivo PDF (opcional)\n  edicaoId    Int\n  edicao      EdicaoEvento @relation(fields: [edicaoId], references: [id], onDelete: Cascade)\n  atribuicoes Atribuicao[]\n  perguntas   Pergunta[]\n}\n\nmodel Revisor {\n  id          Int          @id @default(autoincrement())\n  nome        String\n  areas       Json // Array de strings (ex: [\"Banco de dados\", \"Segurança\"])\n  edicaoId    Int\n  edicao      EdicaoEvento @relation(fields: [edicaoId], references: [id], onDelete: Cascade)\n  atribuicoes Atribuicao[]\n}\n\nmodel Atribuicao {\n  id        Int        @id @default(autoincrement())\n  artigoId  Int\n  revisorId Int\n  artigo    Artigo     @relation(fields: [artigoId], references: [id], onDelete: Cascade)\n  revisor   Revisor    @relation(fields: [revisorId], references: [id], onDelete: Cascade)\n  respostas Resposta[]\n\n  @@unique([artigoId, revisorId])\n}\n\nmodel Pergunta {\n  id        Int        @id @default(autoincrement())\n  texto     String\n  tipo      String // Tipo da pergunta (ex: \"texto\", \"escala\", \"multipla\")\n  artigoId  Int\n  artigo    Artigo     @relation(fields: [artigoId], references: [id], onDelete: Cascade)\n  respostas Resposta[]\n}\n\nmodel Resposta {\n  id           Int        @id @default(autoincrement())\n  resposta     String\n  atribuicaoId Int\n  perguntaId   Int\n  atribuicao   Atribuicao @relation(fields: [atribuicaoId], references: [id], onDelete: Cascade)\n  pergunta     Pergunta   @relation(fields: [perguntaId], references: [id], onDelete: Cascade)\n\n  @@unique([atribuicaoId, perguntaId])\n}\n",
  "runtimeDataModel": {
    "models": {},
    "enums": {},
    "types": {}
  }
}

config.runtimeDataModel = JSON.parse("{\"models\":{\"Evento\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"sigla\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"descricao\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"edicoes\",\"kind\":\"object\",\"type\":\"EdicaoEvento\",\"relationName\":\"EdicaoEventoToEvento\"}],\"dbName\":null},\"EdicaoEvento\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"ano\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"eventoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"evento\",\"kind\":\"object\",\"type\":\"Evento\",\"relationName\":\"EdicaoEventoToEvento\"},{\"name\":\"artigos\",\"kind\":\"object\",\"type\":\"Artigo\",\"relationName\":\"ArtigoToEdicaoEvento\"},{\"name\":\"revisores\",\"kind\":\"object\",\"type\":\"Revisor\",\"relationName\":\"EdicaoEventoToRevisor\"}],\"dbName\":null},\"Artigo\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"titulo\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"autores\",\"kind\":\"scalar\",\"type\":\"Json\"},{\"name\":\"resumo\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"areas\",\"kind\":\"scalar\",\"type\":\"Json\"},{\"name\":\"pdfUrl\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"edicaoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"edicao\",\"kind\":\"object\",\"type\":\"EdicaoEvento\",\"relationName\":\"ArtigoToEdicaoEvento\"},{\"name\":\"atribuicoes\",\"kind\":\"object\",\"type\":\"Atribuicao\",\"relationName\":\"ArtigoToAtribuicao\"},{\"name\":\"perguntas\",\"kind\":\"object\",\"type\":\"Pergunta\",\"relationName\":\"ArtigoToPergunta\"}],\"dbName\":null},\"Revisor\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"nome\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"areas\",\"kind\":\"scalar\",\"type\":\"Json\"},{\"name\":\"edicaoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"edicao\",\"kind\":\"object\",\"type\":\"EdicaoEvento\",\"relationName\":\"EdicaoEventoToRevisor\"},{\"name\":\"atribuicoes\",\"kind\":\"object\",\"type\":\"Atribuicao\",\"relationName\":\"AtribuicaoToRevisor\"}],\"dbName\":null},\"Atribuicao\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"artigoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"revisorId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"artigo\",\"kind\":\"object\",\"type\":\"Artigo\",\"relationName\":\"ArtigoToAtribuicao\"},{\"name\":\"revisor\",\"kind\":\"object\",\"type\":\"Revisor\",\"relationName\":\"AtribuicaoToRevisor\"},{\"name\":\"respostas\",\"kind\":\"object\",\"type\":\"Resposta\",\"relationName\":\"AtribuicaoToResposta\"}],\"dbName\":null},\"Pergunta\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"texto\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"tipo\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"artigoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"artigo\",\"kind\":\"object\",\"type\":\"Artigo\",\"relationName\":\"ArtigoToPergunta\"},{\"name\":\"respostas\",\"kind\":\"object\",\"type\":\"Resposta\",\"relationName\":\"PerguntaToResposta\"}],\"dbName\":null},\"Resposta\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"resposta\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"atribuicaoId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"perguntaId\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"atribuicao\",\"kind\":\"object\",\"type\":\"Atribuicao\",\"relationName\":\"AtribuicaoToResposta\"},{\"name\":\"pergunta\",\"kind\":\"object\",\"type\":\"Pergunta\",\"relationName\":\"PerguntaToResposta\"}],\"dbName\":null}},\"enums\":{},\"types\":{}}")

async function decodeBase64AsWasm(wasmBase64: string): Promise<WebAssembly.Module> {
  const { Buffer } = await import('node:buffer')
  const wasmArray = Buffer.from(wasmBase64, 'base64')
  return new WebAssembly.Module(wasmArray)
}

config.compilerWasm = {
  getRuntime: async () => await import("@prisma/client/runtime/query_compiler_bg.sqlite.mjs"),

  getQueryCompilerWasmModule: async () => {
    const { wasm } = await import("@prisma/client/runtime/query_compiler_bg.sqlite.wasm-base64.mjs")
    return await decodeBase64AsWasm(wasm)
  }
}



export type LogOptions<ClientOptions extends Prisma.PrismaClientOptions> =
  'log' extends keyof ClientOptions ? ClientOptions['log'] extends Array<Prisma.LogLevel | Prisma.LogDefinition> ? Prisma.GetEvents<ClientOptions['log']> : never : never

export interface PrismaClientConstructor {
    /**
   * ## Prisma Client
   * 
   * Type-safe database client for TypeScript
   * @example
   * ```
   * const prisma = new PrismaClient()
   * // Fetch zero or more Eventos
   * const eventos = await prisma.evento.findMany()
   * ```
   * 
   * Read more in our [docs](https://pris.ly/d/client).
   */

  new <
    Options extends Prisma.PrismaClientOptions = Prisma.PrismaClientOptions,
    LogOpts extends LogOptions<Options> = LogOptions<Options>,
    OmitOpts extends Prisma.PrismaClientOptions['omit'] = Options extends { omit: infer U } ? U : Prisma.PrismaClientOptions['omit'],
    ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs
  >(options: Prisma.Subset<Options, Prisma.PrismaClientOptions> ): PrismaClient<LogOpts, OmitOpts, ExtArgs>
}

/**
 * ## Prisma Client
 * 
 * Type-safe database client for TypeScript
 * @example
 * ```
 * const prisma = new PrismaClient()
 * // Fetch zero or more Eventos
 * const eventos = await prisma.evento.findMany()
 * ```
 * 
 * Read more in our [docs](https://pris.ly/d/client).
 */

export interface PrismaClient<
  in LogOpts extends Prisma.LogLevel = never,
  in out OmitOpts extends Prisma.PrismaClientOptions['omit'] = undefined,
  in out ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs
> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['other'] }

  $on<V extends LogOpts>(eventType: V, callback: (event: V extends 'query' ? Prisma.QueryEvent : Prisma.LogEvent) => void): PrismaClient;

  /**
   * Connect with the database
   */
  $connect(): runtime.Types.Utils.JsPromise<void>;

  /**
   * Disconnect from the database
   */
  $disconnect(): runtime.Types.Utils.JsPromise<void>;

/**
   * Executes a prepared raw query and returns the number of affected rows.
   * @example
   * ```
   * const result = await prisma.$executeRaw`UPDATE User SET cool = ${true} WHERE email = ${'user@email.com'};`
   * ```
   *
   * Read more in our [docs](https://pris.ly/d/raw-queries).
   */
  $executeRaw<T = unknown>(query: TemplateStringsArray | Prisma.Sql, ...values: any[]): Prisma.PrismaPromise<number>;

  /**
   * Executes a raw query and returns the number of affected rows.
   * Susceptible to SQL injections, see documentation.
   * @example
   * ```
   * const result = await prisma.$executeRawUnsafe('UPDATE User SET cool = $1 WHERE email = $2 ;', true, 'user@email.com')
   * ```
   *
   * Read more in our [docs](https://pris.ly/d/raw-queries).
   */
  $executeRawUnsafe<T = unknown>(query: string, ...values: any[]): Prisma.PrismaPromise<number>;

  /**
   * Performs a prepared raw query and returns the `SELECT` data.
   * @example
   * ```
   * const result = await prisma.$queryRaw`SELECT * FROM User WHERE id = ${1} OR email = ${'user@email.com'};`
   * ```
   *
   * Read more in our [docs](https://pris.ly/d/raw-queries).
   */
  $queryRaw<T = unknown>(query: TemplateStringsArray | Prisma.Sql, ...values: any[]): Prisma.PrismaPromise<T>;

  /**
   * Performs a raw query and returns the `SELECT` data.
   * Susceptible to SQL injections, see documentation.
   * @example
   * ```
   * const result = await prisma.$queryRawUnsafe('SELECT * FROM User WHERE id = $1 OR email = $2;', 1, 'user@email.com')
   * ```
   *
   * Read more in our [docs](https://pris.ly/d/raw-queries).
   */
  $queryRawUnsafe<T = unknown>(query: string, ...values: any[]): Prisma.PrismaPromise<T>;


  /**
   * Allows the running of a sequence of read/write operations that are guaranteed to either succeed or fail as a whole.
   * @example
   * ```
   * const [george, bob, alice] = await prisma.$transaction([
   *   prisma.user.create({ data: { name: 'George' } }),
   *   prisma.user.create({ data: { name: 'Bob' } }),
   *   prisma.user.create({ data: { name: 'Alice' } }),
   * ])
   * ```
   * 
   * Read more in our [docs](https://www.prisma.io/docs/concepts/components/prisma-client/transactions).
   */
  $transaction<P extends Prisma.PrismaPromise<any>[]>(arg: [...P], options?: { isolationLevel?: Prisma.TransactionIsolationLevel }): runtime.Types.Utils.JsPromise<runtime.Types.Utils.UnwrapTuple<P>>

  $transaction<R>(fn: (prisma: Omit<PrismaClient, runtime.ITXClientDenyList>) => runtime.Types.Utils.JsPromise<R>, options?: { maxWait?: number, timeout?: number, isolationLevel?: Prisma.TransactionIsolationLevel }): runtime.Types.Utils.JsPromise<R>

  $extends: runtime.Types.Extensions.ExtendsHook<"extends", Prisma.TypeMapCb<OmitOpts>, ExtArgs, runtime.Types.Utils.Call<Prisma.TypeMapCb<OmitOpts>, {
    extArgs: ExtArgs
  }>>

      /**
   * `prisma.evento`: Exposes CRUD operations for the **Evento** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Eventos
    * const eventos = await prisma.evento.findMany()
    * ```
    */
  get evento(): Prisma.EventoDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.edicaoEvento`: Exposes CRUD operations for the **EdicaoEvento** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more EdicaoEventos
    * const edicaoEventos = await prisma.edicaoEvento.findMany()
    * ```
    */
  get edicaoEvento(): Prisma.EdicaoEventoDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.artigo`: Exposes CRUD operations for the **Artigo** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Artigos
    * const artigos = await prisma.artigo.findMany()
    * ```
    */
  get artigo(): Prisma.ArtigoDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.revisor`: Exposes CRUD operations for the **Revisor** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Revisors
    * const revisors = await prisma.revisor.findMany()
    * ```
    */
  get revisor(): Prisma.RevisorDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.atribuicao`: Exposes CRUD operations for the **Atribuicao** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Atribuicaos
    * const atribuicaos = await prisma.atribuicao.findMany()
    * ```
    */
  get atribuicao(): Prisma.AtribuicaoDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.pergunta`: Exposes CRUD operations for the **Pergunta** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Perguntas
    * const perguntas = await prisma.pergunta.findMany()
    * ```
    */
  get pergunta(): Prisma.PerguntaDelegate<ExtArgs, { omit: OmitOpts }>;

  /**
   * `prisma.resposta`: Exposes CRUD operations for the **Resposta** model.
    * Example usage:
    * ```ts
    * // Fetch zero or more Respostas
    * const respostas = await prisma.resposta.findMany()
    * ```
    */
  get resposta(): Prisma.RespostaDelegate<ExtArgs, { omit: OmitOpts }>;
}

export function getPrismaClientClass(): PrismaClientConstructor {
  return runtime.getPrismaClient(config) as unknown as PrismaClientConstructor
}
