// @prisma/client/internal/prismaNamespace.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * WARNING: This is an internal file that is subject to change!
 *
 * 🛑 Under no circumstances should you import this file directly! 🛑
 *
 * All exports from this file are wrapped under a `Prisma` namespace object in the client.ts file.
 * While this enables partial backward compatibility, it is not part of the stable public API.
 *
 * If you are looking for your Models, Enums, and Input Types, please import them from the respective
 * model files in the `model` directory!
 */

import * as runtime from "@prisma/client/runtime/client"
import type * as Prisma from "../models"
import { type PrismaClient } from "./class"

export type * from '../models'

export type DMMF = typeof runtime.DMMF

export type PrismaPromise<T> = runtime.Types.Public.PrismaPromise<T>

/**
 * Prisma Errors
 */

export const PrismaClientKnownRequestError = runtime.PrismaClientKnownRequestError
export type PrismaClientKnownRequestError = runtime.PrismaClientKnownRequestError

export const PrismaClientUnknownRequestError = runtime.PrismaClientUnknownRequestError
export type PrismaClientUnknownRequestError = runtime.PrismaClientUnknownRequestError

export const PrismaClientRustPanicError = runtime.PrismaClientRustPanicError
export type PrismaClientRustPanicError = runtime.PrismaClientRustPanicError

export const PrismaClientInitializationError = runtime.PrismaClientInitializationError
export type PrismaClientInitializationError = runtime.PrismaClientInitializationError

export const PrismaClientValidationError = runtime.PrismaClientValidationError
export type PrismaClientValidationError = runtime.PrismaClientValidationError

/**
 * Re-export of sql-template-tag
 */
export const sql = runtime.sqltag
export const empty = runtime.empty
export const join = runtime.join
export const raw = runtime.raw
export const Sql = runtime.Sql
export type Sql = runtime.Sql



/**
 * Decimal.js
 */
export const Decimal = runtime.Decimal
export type Decimal = runtime.Decimal

export type DecimalJsLike = runtime.DecimalJsLike

/**
* Extensions
*/
export type Extension = runtime.Types.Extensions.UserArgs
export const getExtensionContext = runtime.Extensions.getExtensionContext
export type Args<T, F extends runtime.Operation> = runtime.Types.Public.Args<T, F>
export type Payload<T, F extends runtime.Operation = never> = runtime.Types.Public.Payload<T, F>
export type Result<T, A, F extends runtime.Operation> = runtime.Types.Public.Result<T, A, F>
export type Exact<A, W> = runtime.Types.Public.Exact<A, W>

export type PrismaVersion = {
  client: string
  engine: string
}

/**
 * Prisma Client JS version: 7.2.0
 * Query Engine version: 0c8ef2ce45c83248ab3df073180d5eda9e8be7a3
 */
export const prismaVersion: PrismaVersion = {
  client: "7.2.0",
  engine: "0c8ef2ce45c83248ab3df073180d5eda9e8be7a3"
}

/**
 * Utility Types
 */

export type Bytes = runtime.Bytes
export type JsonObject = runtime.JsonObject
export type JsonArray = runtime.JsonArray
export type JsonValue = runtime.JsonValue
export type InputJsonObject = runtime.InputJsonObject
export type InputJsonArray = runtime.InputJsonArray
export type InputJsonValue = runtime.InputJsonValue


export const NullTypes = {
  DbNull: runtime.NullTypes.DbNull as (new (secret: never) => typeof runtime.DbNull),
  JsonNull: runtime.NullTypes.JsonNull as (new (secret: never) => typeof runtime.JsonNull),
  AnyNull: runtime.NullTypes.AnyNull as (new (secret: never) => typeof runtime.AnyNull),
}
/**
 * Helper for filtering JSON entries that have `null` on the database (empty on the db)
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const DbNull = runtime.DbNull

/**
 * Helper for filtering JSON entries that have JSON `null` values (not empty on the db)
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const JsonNull = runtime.JsonNull

/**
 * Helper for filtering JSON entries that are `Prisma.DbNull` or `Prisma.JsonNull`
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const AnyNull = runtime.AnyNull


type SelectAndInclude = {
  select: any
  include: any
}

type SelectAndOmit = {
  select: any
  omit: any
}

/**
 * From T, pick a set of properties whose keys are in the union K
 */
type Prisma__Pick<T, K extends keyof T> = {
    [P in K]: T[P];
};

export type Enumerable<T> = T | Array<T>;

/**
 * Subset
 * @desc From `T` pick properties that exist in `U`. Simple version of Intersection
 */
export type Subset<T, U> = {
  [key in keyof T]: key extends keyof U ? T[key] : never;
};

/**
 * SelectSubset
 * @desc From `T` pick properties that exist in `U`. Simple version of Intersection.
 * Additionally, it validates, if both select and include are present. If the case, it errors.
 */
export type SelectSubset<T, U> = {
  [key in keyof T]: key extends keyof U ? T[key] : never
} &
  (T extends SelectAndInclude
    ? 'Please either choose `select` or `include`.'
    : T extends SelectAndOmit
      ? 'Please either choose `select` or `omit`.'
      : {})

/**
 * Subset + Intersection
 * @desc From `T` pick properties that exist in `U` and intersect `K`
 */
export type SubsetIntersection<T, U, K> = {
  [key in keyof T]: key extends keyof U ? T[key] : never
} &
  K

type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };

/**
 * XOR is needed to have a real mutually exclusive union type
 * https://stackoverflow.com/questions/42123407/does-typescript-support-mutually-exclusive-types
 */
export type XOR<T, U> =
  T extends object ?
  U extends object ?
    (Without<T, U> & U) | (Without<U, T> & T)
  : U : T


/**
 * Is T a Record?
 */
type IsObject<T extends any> = T extends Array<any>
? False
: T extends Date
? False
: T extends Uint8Array
? False
: T extends BigInt
? False
: T extends object
? True
: False


/**
 * If it's T[], return T
 */
export type UnEnumerate<T extends unknown> = T extends Array<infer U> ? U : T

/**
 * From ts-toolbelt
 */

type __Either<O extends object, K extends Key> = Omit<O, K> &
  {
    // Merge all but K
    [P in K]: Prisma__Pick<O, P & keyof O> // With K possibilities
  }[K]

type EitherStrict<O extends object, K extends Key> = Strict<__Either<O, K>>

type EitherLoose<O extends object, K extends Key> = ComputeRaw<__Either<O, K>>

type _Either<
  O extends object,
  K extends Key,
  strict extends Boolean
> = {
  1: EitherStrict<O, K>
  0: EitherLoose<O, K>
}[strict]

export type Either<
  O extends object,
  K extends Key,
  strict extends Boolean = 1
> = O extends unknown ? _Either<O, K, strict> : never

export type Union = any

export type PatchUndefined<O extends object, O1 extends object> = {
  [K in keyof O]: O[K] extends undefined ? At<O1, K> : O[K]
} & {}

/** Helper Types for "Merge" **/
export type IntersectOf<U extends Union> = (
  U extends unknown ? (k: U) => void : never
) extends (k: infer I) => void
  ? I
  : never

export type Overwrite<O extends object, O1 extends object> = {
    [K in keyof O]: K extends keyof O1 ? O1[K] : O[K];
} & {};

type _Merge<U extends object> = IntersectOf<Overwrite<U, {
    [K in keyof U]-?: At<U, K>;
}>>;

type Key = string | number | symbol;
type AtStrict<O extends object, K extends Key> = O[K & keyof O];
type AtLoose<O extends object, K extends Key> = O extends unknown ? AtStrict<O, K> : never;
export type At<O extends object, K extends Key, strict extends Boolean = 1> = {
    1: AtStrict<O, K>;
    0: AtLoose<O, K>;
}[strict];

export type ComputeRaw<A extends any> = A extends Function ? A : {
  [K in keyof A]: A[K];
} & {};

export type OptionalFlat<O> = {
  [K in keyof O]?: O[K];
} & {};

type _Record<K extends keyof any, T> = {
  [P in K]: T;
};

// cause typescript not to expand types and preserve names
type NoExpand<T> = T extends unknown ? T : never;

// this type assumes the passed object is entirely optional
export type AtLeast<O extends object, K extends string> = NoExpand<
  O extends unknown
  ? | (K extends keyof O ? { [P in K]: O[P] } & O : O)
    | {[P in keyof O as P extends K ? P : never]-?: O[P]} & O
  : never>;

type _Strict<U, _U = U> = U extends unknown ? U & OptionalFlat<_Record<Exclude<Keys<_U>, keyof U>, never>> : never;

export type Strict<U extends object> = ComputeRaw<_Strict<U>>;
/** End Helper Types for "Merge" **/

export type Merge<U extends object> = ComputeRaw<_Merge<Strict<U>>>;

export type Boolean = True | False

export type True = 1

export type False = 0

export type Not<B extends Boolean> = {
  0: 1
  1: 0
}[B]

export type Extends<A1 extends any, A2 extends any> = [A1] extends [never]
  ? 0 // anything `never` is false
  : A1 extends A2
  ? 1
  : 0

export type Has<U extends Union, U1 extends Union> = Not<
  Extends<Exclude<U1, U>, U1>
>

export type Or<B1 extends Boolean, B2 extends Boolean> = {
  0: {
    0: 0
    1: 1
  }
  1: {
    0: 1
    1: 1
  }
}[B1][B2]

export type Keys<U extends Union> = U extends unknown ? keyof U : never

export type GetScalarType<T, O> = O extends object ? {
  [P in keyof T]: P extends keyof O
    ? O[P]
    : never
} : never

type FieldPaths<
  T,
  U = Omit<T, '_avg' | '_sum' | '_count' | '_min' | '_max'>
> = IsObject<T> extends True ? U : T

export type GetHavingFields<T> = {
  [K in keyof T]: Or<
    Or<Extends<'OR', K>, Extends<'AND', K>>,
    Extends<'NOT', K>
  > extends True
    ? // infer is only needed to not hit TS limit
      // based on the brilliant idea of Pierre-Antoine Mills
      // https://github.com/microsoft/TypeScript/issues/30188#issuecomment-478938437
      T[K] extends infer TK
      ? GetHavingFields<UnEnumerate<TK> extends object ? Merge<UnEnumerate<TK>> : never>
      : never
    : {} extends FieldPaths<T[K]>
    ? never
    : K
}[keyof T]

/**
 * Convert tuple to union
 */
type _TupleToUnion<T> = T extends (infer E)[] ? E : never
type TupleToUnion<K extends readonly any[]> = _TupleToUnion<K>
export type MaybeTupleToUnion<T> = T extends any[] ? TupleToUnion<T> : T

/**
 * Like `Pick`, but additionally can also accept an array of keys
 */
export type PickEnumerable<T, K extends Enumerable<keyof T> | keyof T> = Prisma__Pick<T, MaybeTupleToUnion<K>>

/**
 * Exclude all keys with underscores
 */
export type ExcludeUnderscoreKeys<T extends string> = T extends `_${string}` ? never : T


export type FieldRef<Model, FieldType> = runtime.FieldRef<Model, FieldType>

type FieldRefInputType<Model, FieldType> = Model extends never ? never : FieldRef<Model, FieldType>


export const ModelName = {
  Evento: 'Evento',
  EdicaoEvento: 'EdicaoEvento',
  Artigo: 'Artigo',
  Revisor: 'Revisor',
  Atribuicao: 'Atribuicao',
  Pergunta: 'Pergunta',
  Resposta: 'Resposta'
} as const

export type ModelName = (typeof ModelName)[keyof typeof ModelName]



export interface TypeMapCb<GlobalOmitOptions = {}> extends runtime.Types.Utils.Fn<{extArgs: runtime.Types.Extensions.InternalArgs }, runtime.Types.Utils.Record<string, any>> {
  returns: TypeMap<this['params']['extArgs'], GlobalOmitOptions>
}

export type TypeMap<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> = {
  globalOmitOptions: {
    omit: GlobalOmitOptions
  }
  meta: {
    modelProps: "evento" | "edicaoEvento" | "artigo" | "revisor" | "atribuicao" | "pergunta" | "resposta"
    txIsolationLevel: TransactionIsolationLevel
  }
  model: {
    Evento: {
      payload: Prisma.$EventoPayload<ExtArgs>
      fields: Prisma.EventoFieldRefs
      operations: {
        findUnique: {
          args: Prisma.EventoFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.EventoFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        findFirst: {
          args: Prisma.EventoFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.EventoFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        findMany: {
          args: Prisma.EventoFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>[]
        }
        create: {
          args: Prisma.EventoCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        createMany: {
          args: Prisma.EventoCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.EventoCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>[]
        }
        delete: {
          args: Prisma.EventoDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        update: {
          args: Prisma.EventoUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        deleteMany: {
          args: Prisma.EventoDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.EventoUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.EventoUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>[]
        }
        upsert: {
          args: Prisma.EventoUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EventoPayload>
        }
        aggregate: {
          args: Prisma.EventoAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateEvento>
        }
        groupBy: {
          args: Prisma.EventoGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.EventoGroupByOutputType>[]
        }
        count: {
          args: Prisma.EventoCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.EventoCountAggregateOutputType> | number
        }
      }
    }
    EdicaoEvento: {
      payload: Prisma.$EdicaoEventoPayload<ExtArgs>
      fields: Prisma.EdicaoEventoFieldRefs
      operations: {
        findUnique: {
          args: Prisma.EdicaoEventoFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.EdicaoEventoFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        findFirst: {
          args: Prisma.EdicaoEventoFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.EdicaoEventoFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        findMany: {
          args: Prisma.EdicaoEventoFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>[]
        }
        create: {
          args: Prisma.EdicaoEventoCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        createMany: {
          args: Prisma.EdicaoEventoCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.EdicaoEventoCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>[]
        }
        delete: {
          args: Prisma.EdicaoEventoDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        update: {
          args: Prisma.EdicaoEventoUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        deleteMany: {
          args: Prisma.EdicaoEventoDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.EdicaoEventoUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.EdicaoEventoUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>[]
        }
        upsert: {
          args: Prisma.EdicaoEventoUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$EdicaoEventoPayload>
        }
        aggregate: {
          args: Prisma.EdicaoEventoAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateEdicaoEvento>
        }
        groupBy: {
          args: Prisma.EdicaoEventoGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.EdicaoEventoGroupByOutputType>[]
        }
        count: {
          args: Prisma.EdicaoEventoCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.EdicaoEventoCountAggregateOutputType> | number
        }
      }
    }
    Artigo: {
      payload: Prisma.$ArtigoPayload<ExtArgs>
      fields: Prisma.ArtigoFieldRefs
      operations: {
        findUnique: {
          args: Prisma.ArtigoFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.ArtigoFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        findFirst: {
          args: Prisma.ArtigoFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.ArtigoFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        findMany: {
          args: Prisma.ArtigoFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>[]
        }
        create: {
          args: Prisma.ArtigoCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        createMany: {
          args: Prisma.ArtigoCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.ArtigoCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>[]
        }
        delete: {
          args: Prisma.ArtigoDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        update: {
          args: Prisma.ArtigoUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        deleteMany: {
          args: Prisma.ArtigoDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.ArtigoUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.ArtigoUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>[]
        }
        upsert: {
          args: Prisma.ArtigoUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$ArtigoPayload>
        }
        aggregate: {
          args: Prisma.ArtigoAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateArtigo>
        }
        groupBy: {
          args: Prisma.ArtigoGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.ArtigoGroupByOutputType>[]
        }
        count: {
          args: Prisma.ArtigoCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.ArtigoCountAggregateOutputType> | number
        }
      }
    }
    Revisor: {
      payload: Prisma.$RevisorPayload<ExtArgs>
      fields: Prisma.RevisorFieldRefs
      operations: {
        findUnique: {
          args: Prisma.RevisorFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.RevisorFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        findFirst: {
          args: Prisma.RevisorFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.RevisorFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        findMany: {
          args: Prisma.RevisorFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>[]
        }
        create: {
          args: Prisma.RevisorCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        createMany: {
          args: Prisma.RevisorCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.RevisorCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>[]
        }
        delete: {
          args: Prisma.RevisorDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        update: {
          args: Prisma.RevisorUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        deleteMany: {
          args: Prisma.RevisorDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.RevisorUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.RevisorUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>[]
        }
        upsert: {
          args: Prisma.RevisorUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RevisorPayload>
        }
        aggregate: {
          args: Prisma.RevisorAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateRevisor>
        }
        groupBy: {
          args: Prisma.RevisorGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.RevisorGroupByOutputType>[]
        }
        count: {
          args: Prisma.RevisorCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.RevisorCountAggregateOutputType> | number
        }
      }
    }
    Atribuicao: {
      payload: Prisma.$AtribuicaoPayload<ExtArgs>
      fields: Prisma.AtribuicaoFieldRefs
      operations: {
        findUnique: {
          args: Prisma.AtribuicaoFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.AtribuicaoFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        findFirst: {
          args: Prisma.AtribuicaoFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.AtribuicaoFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        findMany: {
          args: Prisma.AtribuicaoFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>[]
        }
        create: {
          args: Prisma.AtribuicaoCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        createMany: {
          args: Prisma.AtribuicaoCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.AtribuicaoCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>[]
        }
        delete: {
          args: Prisma.AtribuicaoDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        update: {
          args: Prisma.AtribuicaoUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        deleteMany: {
          args: Prisma.AtribuicaoDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.AtribuicaoUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.AtribuicaoUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>[]
        }
        upsert: {
          args: Prisma.AtribuicaoUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$AtribuicaoPayload>
        }
        aggregate: {
          args: Prisma.AtribuicaoAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateAtribuicao>
        }
        groupBy: {
          args: Prisma.AtribuicaoGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AtribuicaoGroupByOutputType>[]
        }
        count: {
          args: Prisma.AtribuicaoCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AtribuicaoCountAggregateOutputType> | number
        }
      }
    }
    Pergunta: {
      payload: Prisma.$PerguntaPayload<ExtArgs>
      fields: Prisma.PerguntaFieldRefs
      operations: {
        findUnique: {
          args: Prisma.PerguntaFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.PerguntaFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        findFirst: {
          args: Prisma.PerguntaFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.PerguntaFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        findMany: {
          args: Prisma.PerguntaFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>[]
        }
        create: {
          args: Prisma.PerguntaCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        createMany: {
          args: Prisma.PerguntaCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.PerguntaCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>[]
        }
        delete: {
          args: Prisma.PerguntaDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        update: {
          args: Prisma.PerguntaUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        deleteMany: {
          args: Prisma.PerguntaDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.PerguntaUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.PerguntaUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>[]
        }
        upsert: {
          args: Prisma.PerguntaUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$PerguntaPayload>
        }
        aggregate: {
          args: Prisma.PerguntaAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregatePergunta>
        }
        groupBy: {
          args: Prisma.PerguntaGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.PerguntaGroupByOutputType>[]
        }
        count: {
          args: Prisma.PerguntaCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.PerguntaCountAggregateOutputType> | number
        }
      }
    }
    Resposta: {
      payload: Prisma.$RespostaPayload<ExtArgs>
      fields: Prisma.RespostaFieldRefs
      operations: {
        findUnique: {
          args: Prisma.RespostaFindUniqueArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload> | null
        }
        findUniqueOrThrow: {
          args: Prisma.RespostaFindUniqueOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        findFirst: {
          args: Prisma.RespostaFindFirstArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload> | null
        }
        findFirstOrThrow: {
          args: Prisma.RespostaFindFirstOrThrowArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        findMany: {
          args: Prisma.RespostaFindManyArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>[]
        }
        create: {
          args: Prisma.RespostaCreateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        createMany: {
          args: Prisma.RespostaCreateManyArgs<ExtArgs>
          result: BatchPayload
        }
        createManyAndReturn: {
          args: Prisma.RespostaCreateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>[]
        }
        delete: {
          args: Prisma.RespostaDeleteArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        update: {
          args: Prisma.RespostaUpdateArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        deleteMany: {
          args: Prisma.RespostaDeleteManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateMany: {
          args: Prisma.RespostaUpdateManyArgs<ExtArgs>
          result: BatchPayload
        }
        updateManyAndReturn: {
          args: Prisma.RespostaUpdateManyAndReturnArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>[]
        }
        upsert: {
          args: Prisma.RespostaUpsertArgs<ExtArgs>
          result: runtime.Types.Utils.PayloadToResult<Prisma.$RespostaPayload>
        }
        aggregate: {
          args: Prisma.RespostaAggregateArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.AggregateResposta>
        }
        groupBy: {
          args: Prisma.RespostaGroupByArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.RespostaGroupByOutputType>[]
        }
        count: {
          args: Prisma.RespostaCountArgs<ExtArgs>
          result: runtime.Types.Utils.Optional<Prisma.RespostaCountAggregateOutputType> | number
        }
      }
    }
  }
} & {
  other: {
    payload: any
    operations: {
      $executeRaw: {
        args: [query: TemplateStringsArray | Sql, ...values: any[]],
        result: any
      }
      $executeRawUnsafe: {
        args: [query: string, ...values: any[]],
        result: any
      }
      $queryRaw: {
        args: [query: TemplateStringsArray | Sql, ...values: any[]],
        result: any
      }
      $queryRawUnsafe: {
        args: [query: string, ...values: any[]],
        result: any
      }
    }
  }
}

/**
 * Enums
 */

export const TransactionIsolationLevel = runtime.makeStrictEnum({
  Serializable: 'Serializable'
} as const)

export type TransactionIsolationLevel = (typeof TransactionIsolationLevel)[keyof typeof TransactionIsolationLevel]


export const EventoScalarFieldEnum = {
  id: 'id',
  sigla: 'sigla',
  descricao: 'descricao'
} as const

export type EventoScalarFieldEnum = (typeof EventoScalarFieldEnum)[keyof typeof EventoScalarFieldEnum]


export const EdicaoEventoScalarFieldEnum = {
  id: 'id',
  ano: 'ano',
  eventoId: 'eventoId'
} as const

export type EdicaoEventoScalarFieldEnum = (typeof EdicaoEventoScalarFieldEnum)[keyof typeof EdicaoEventoScalarFieldEnum]


export const ArtigoScalarFieldEnum = {
  id: 'id',
  titulo: 'titulo',
  autores: 'autores',
  resumo: 'resumo',
  areas: 'areas',
  pdfUrl: 'pdfUrl',
  edicaoId: 'edicaoId'
} as const

export type ArtigoScalarFieldEnum = (typeof ArtigoScalarFieldEnum)[keyof typeof ArtigoScalarFieldEnum]


export const RevisorScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  areas: 'areas',
  edicaoId: 'edicaoId'
} as const

export type RevisorScalarFieldEnum = (typeof RevisorScalarFieldEnum)[keyof typeof RevisorScalarFieldEnum]


export const AtribuicaoScalarFieldEnum = {
  id: 'id',
  artigoId: 'artigoId',
  revisorId: 'revisorId'
} as const

export type AtribuicaoScalarFieldEnum = (typeof AtribuicaoScalarFieldEnum)[keyof typeof AtribuicaoScalarFieldEnum]


export const PerguntaScalarFieldEnum = {
  id: 'id',
  texto: 'texto',
  tipo: 'tipo',
  artigoId: 'artigoId'
} as const

export type PerguntaScalarFieldEnum = (typeof PerguntaScalarFieldEnum)[keyof typeof PerguntaScalarFieldEnum]


export const RespostaScalarFieldEnum = {
  id: 'id',
  resposta: 'resposta',
  atribuicaoId: 'atribuicaoId',
  perguntaId: 'perguntaId'
} as const

export type RespostaScalarFieldEnum = (typeof RespostaScalarFieldEnum)[keyof typeof RespostaScalarFieldEnum]


export const SortOrder = {
  asc: 'asc',
  desc: 'desc'
} as const

export type SortOrder = (typeof SortOrder)[keyof typeof SortOrder]


export const JsonNullValueInput = {
  JsonNull: JsonNull
} as const

export type JsonNullValueInput = (typeof JsonNullValueInput)[keyof typeof JsonNullValueInput]


export const JsonNullValueFilter = {
  DbNull: DbNull,
  JsonNull: JsonNull,
  AnyNull: AnyNull
} as const

export type JsonNullValueFilter = (typeof JsonNullValueFilter)[keyof typeof JsonNullValueFilter]


export const QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
} as const

export type QueryMode = (typeof QueryMode)[keyof typeof QueryMode]


export const NullsOrder = {
  first: 'first',
  last: 'last'
} as const

export type NullsOrder = (typeof NullsOrder)[keyof typeof NullsOrder]



/**
 * Field references
 */


/**
 * Reference to a field of type 'Int'
 */
export type IntFieldRefInput<$PrismaModel> = FieldRefInputType<$PrismaModel, 'Int'>
    


/**
 * Reference to a field of type 'String'
 */
export type StringFieldRefInput<$PrismaModel> = FieldRefInputType<$PrismaModel, 'String'>
    


/**
 * Reference to a field of type 'Json'
 */
export type JsonFieldRefInput<$PrismaModel> = FieldRefInputType<$PrismaModel, 'Json'>
    


/**
 * Reference to a field of type 'QueryMode'
 */
export type EnumQueryModeFieldRefInput<$PrismaModel> = FieldRefInputType<$PrismaModel, 'QueryMode'>
    


/**
 * Reference to a field of type 'Float'
 */
export type FloatFieldRefInput<$PrismaModel> = FieldRefInputType<$PrismaModel, 'Float'>
    

/**
 * Batch Payload for updateMany & deleteMany & createMany
 */
export type BatchPayload = {
  count: number
}

export const defineExtension = runtime.Extensions.defineExtension as unknown as runtime.Types.Extensions.ExtendsHook<"define", TypeMapCb, runtime.Types.Extensions.DefaultArgs>
export type DefaultPrismaClient = PrismaClient
export type ErrorFormat = 'pretty' | 'colorless' | 'minimal'
export type PrismaClientOptions = ({
  /**
   * Instance of a Driver Adapter, e.g., like one provided by `@prisma/adapter-pg`.
   */
  adapter: runtime.SqlDriverAdapterFactory
  accelerateUrl?: never
} | {
  /**
   * Prisma Accelerate URL allowing the client to connect through Accelerate instead of a direct database.
   */
  accelerateUrl: string
  adapter?: never
}) & {
  /**
   * @default "colorless"
   */
  errorFormat?: ErrorFormat
  /**
   * @example
   * ```
   * // Shorthand for `emit: 'stdout'`
   * log: ['query', 'info', 'warn', 'error']
   * 
   * // Emit as events only
   * log: [
   *   { emit: 'event', level: 'query' },
   *   { emit: 'event', level: 'info' },
   *   { emit: 'event', level: 'warn' }
   *   { emit: 'event', level: 'error' }
   * ]
   * 
   * / Emit as events and log to stdout
   * og: [
   *  { emit: 'stdout', level: 'query' },
   *  { emit: 'stdout', level: 'info' },
   *  { emit: 'stdout', level: 'warn' }
   *  { emit: 'stdout', level: 'error' }
   * 
   * ```
   * Read more in our [docs](https://pris.ly/d/logging).
   */
  log?: (LogLevel | LogDefinition)[]
  /**
   * The default values for transactionOptions
   * maxWait ?= 2000
   * timeout ?= 5000
   */
  transactionOptions?: {
    maxWait?: number
    timeout?: number
    isolationLevel?: TransactionIsolationLevel
  }
  /**
   * Global configuration for omitting model fields by default.
   * 
   * @example
   * ```
   * const prisma = new PrismaClient({
   *   omit: {
   *     user: {
   *       password: true
   *     }
   *   }
   * })
   * ```
   */
  omit?: GlobalOmitConfig
  /**
   * SQL commenter plugins that add metadata to SQL queries as comments.
   * Comments follow the sqlcommenter format: https://google.github.io/sqlcommenter/
   * 
   * @example
   * ```
   * const prisma = new PrismaClient({
   *   adapter,
   *   comments: [
   *     traceContext(),
   *     queryInsights(),
   *   ],
   * })
   * ```
   */
  comments?: runtime.SqlCommenterPlugin[]
}
export type GlobalOmitConfig = {
  evento?: Prisma.EventoOmit
  edicaoEvento?: Prisma.EdicaoEventoOmit
  artigo?: Prisma.ArtigoOmit
  revisor?: Prisma.RevisorOmit
  atribuicao?: Prisma.AtribuicaoOmit
  pergunta?: Prisma.PerguntaOmit
  resposta?: Prisma.RespostaOmit
}

/* Types for Logging */
export type LogLevel = 'info' | 'query' | 'warn' | 'error'
export type LogDefinition = {
  level: LogLevel
  emit: 'stdout' | 'event'
}

export type CheckIsLogLevel<T> = T extends LogLevel ? T : never;

export type GetLogType<T> = CheckIsLogLevel<
  T extends LogDefinition ? T['level'] : T
>;

export type GetEvents<T extends any[]> = T extends Array<LogLevel | LogDefinition>
  ? GetLogType<T[number]>
  : never;

export type QueryEvent = {
  timestamp: Date
  query: string
  params: string
  duration: number
  target: string
}

export type LogEvent = {
  timestamp: Date
  message: string
  target: string
}
/* End Types for Logging */


export type PrismaAction =
  | 'findUnique'
  | 'findUniqueOrThrow'
  | 'findMany'
  | 'findFirst'
  | 'findFirstOrThrow'
  | 'create'
  | 'createMany'
  | 'createManyAndReturn'
  | 'update'
  | 'updateMany'
  | 'updateManyAndReturn'
  | 'upsert'
  | 'delete'
  | 'deleteMany'
  | 'executeRaw'
  | 'queryRaw'
  | 'aggregate'
  | 'count'
  | 'runCommandRaw'
  | 'findRaw'
  | 'groupBy'

/**
 * `PrismaClient` proxy available in interactive transactions.
 */
export type TransactionClient = Omit<DefaultPrismaClient, runtime.ITXClientDenyList>
