// @prisma/client/internal/prismaNamespaceBrowser.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * WARNING: This is an internal file that is subject to change!
 *
 * 🛑 Under no circumstances should you import this file directly! 🛑
 *
 * All exports from this file are wrapped under a `Prisma` namespace object in the browser.ts file.
 * While this enables partial backward compatibility, it is not part of the stable public API.
 *
 * If you are looking for your Models, Enums, and Input Types, please import them from the respective
 * model files in the `model` directory!
 */

import * as runtime from "@prisma/client/runtime/index-browser"

export type * from '../models'
export type * from './prismaNamespace'

export const Decimal = runtime.Decimal


export const NullTypes = {
  DbNull: runtime.NullTypes.DbNull as (new (secret: never) => typeof runtime.DbNull),
  JsonNull: runtime.NullTypes.JsonNull as (new (secret: never) => typeof runtime.JsonNull),
  AnyNull: runtime.NullTypes.AnyNull as (new (secret: never) => typeof runtime.AnyNull),
}
/**
 * Helper for filtering JSON entries that have `null` on the database (empty on the db)
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const DbNull = runtime.DbNull

/**
 * Helper for filtering JSON entries that have JSON `null` values (not empty on the db)
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const JsonNull = runtime.JsonNull

/**
 * Helper for filtering JSON entries that are `Prisma.DbNull` or `Prisma.JsonNull`
 *
 * @see https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-a-json-field
 */
export const AnyNull = runtime.AnyNull


export const ModelName = {
  Evento: 'Evento',
  EdicaoEvento: 'EdicaoEvento',
  Artigo: 'Artigo',
  Revisor: 'Revisor',
  Atribuicao: 'Atribuicao',
  Pergunta: 'Pergunta',
  Resposta: 'Resposta'
} as const

export type ModelName = (typeof ModelName)[keyof typeof ModelName]

/*
 * Enums
 */

export const TransactionIsolationLevel = {
  Serializable: 'Serializable'
} as const

export type TransactionIsolationLevel = (typeof TransactionIsolationLevel)[keyof typeof TransactionIsolationLevel]


export const EventoScalarFieldEnum = {
  id: 'id',
  sigla: 'sigla',
  descricao: 'descricao'
} as const

export type EventoScalarFieldEnum = (typeof EventoScalarFieldEnum)[keyof typeof EventoScalarFieldEnum]


export const EdicaoEventoScalarFieldEnum = {
  id: 'id',
  ano: 'ano',
  eventoId: 'eventoId'
} as const

export type EdicaoEventoScalarFieldEnum = (typeof EdicaoEventoScalarFieldEnum)[keyof typeof EdicaoEventoScalarFieldEnum]


export const ArtigoScalarFieldEnum = {
  id: 'id',
  titulo: 'titulo',
  autores: 'autores',
  resumo: 'resumo',
  areas: 'areas',
  pdfUrl: 'pdfUrl',
  edicaoId: 'edicaoId'
} as const

export type ArtigoScalarFieldEnum = (typeof ArtigoScalarFieldEnum)[keyof typeof ArtigoScalarFieldEnum]


export const RevisorScalarFieldEnum = {
  id: 'id',
  nome: 'nome',
  areas: 'areas',
  edicaoId: 'edicaoId'
} as const

export type RevisorScalarFieldEnum = (typeof RevisorScalarFieldEnum)[keyof typeof RevisorScalarFieldEnum]


export const AtribuicaoScalarFieldEnum = {
  id: 'id',
  artigoId: 'artigoId',
  revisorId: 'revisorId'
} as const

export type AtribuicaoScalarFieldEnum = (typeof AtribuicaoScalarFieldEnum)[keyof typeof AtribuicaoScalarFieldEnum]


export const PerguntaScalarFieldEnum = {
  id: 'id',
  texto: 'texto',
  tipo: 'tipo',
  artigoId: 'artigoId'
} as const

export type PerguntaScalarFieldEnum = (typeof PerguntaScalarFieldEnum)[keyof typeof PerguntaScalarFieldEnum]


export const RespostaScalarFieldEnum = {
  id: 'id',
  resposta: 'resposta',
  atribuicaoId: 'atribuicaoId',
  perguntaId: 'perguntaId'
} as const

export type RespostaScalarFieldEnum = (typeof RespostaScalarFieldEnum)[keyof typeof RespostaScalarFieldEnum]


export const SortOrder = {
  asc: 'asc',
  desc: 'desc'
} as const

export type SortOrder = (typeof SortOrder)[keyof typeof SortOrder]


export const JsonNullValueInput = {
  JsonNull: 'JsonNull'
} as const

export type JsonNullValueInput = (typeof JsonNullValueInput)[keyof typeof JsonNullValueInput]


export const JsonNullValueFilter = {
  DbNull: 'DbNull',
  JsonNull: 'JsonNull',
  AnyNull: 'AnyNull'
} as const

export type JsonNullValueFilter = (typeof JsonNullValueFilter)[keyof typeof JsonNullValueFilter]


export const QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
} as const

export type QueryMode = (typeof QueryMode)[keyof typeof QueryMode]


export const NullsOrder = {
  first: 'first',
  last: 'last'
} as const

export type NullsOrder = (typeof NullsOrder)[keyof typeof NullsOrder]
