// @prisma/client/browser.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file should be your main import to use Prisma-related types and utilities in a browser. 
 * Use it to get access to models, enums, and input types.
 * 
 * This file does not contain a `PrismaClient` class, nor several other helpers that are intended as server-side only.
 * See `client.ts` for the standard, server-side entry point.
 *
 * 🟢 You can import this file directly.
 */

import * as Prisma from './internal/prismaNamespaceBrowser'
export { Prisma }
export * as $Enums from './enums'
export * from './enums';
/**
 * Model Evento
 * 
 */
export type Evento = Prisma.EventoModel
/**
 * Model EdicaoEvento
 * 
 */
export type EdicaoEvento = Prisma.EdicaoEventoModel
/**
 * Model Artigo
 * 
 */
export type Artigo = Prisma.ArtigoModel
/**
 * Model Revisor
 * 
 */
export type Revisor = Prisma.RevisorModel
/**
 * Model Atribuicao
 * 
 */
export type Atribuicao = Prisma.AtribuicaoModel
/**
 * Model Pergunta
 * 
 */
export type Pergunta = Prisma.PerguntaModel
/**
 * Model Resposta
 * 
 */
export type Resposta = Prisma.RespostaModel
