// @prisma/client/models/Artigo.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Artigo` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Artigo
 * 
 */
export type ArtigoModel = runtime.Types.Result.DefaultSelection<Prisma.$ArtigoPayload>

export type AggregateArtigo = {
  _count: ArtigoCountAggregateOutputType | null
  _avg: ArtigoAvgAggregateOutputType | null
  _sum: ArtigoSumAggregateOutputType | null
  _min: ArtigoMinAggregateOutputType | null
  _max: ArtigoMaxAggregateOutputType | null
}

export type ArtigoAvgAggregateOutputType = {
  id: number | null
  edicaoId: number | null
}

export type ArtigoSumAggregateOutputType = {
  id: number | null
  edicaoId: number | null
}

export type ArtigoMinAggregateOutputType = {
  id: number | null
  titulo: string | null
  resumo: string | null
  pdfUrl: string | null
  edicaoId: number | null
}

export type ArtigoMaxAggregateOutputType = {
  id: number | null
  titulo: string | null
  resumo: string | null
  pdfUrl: string | null
  edicaoId: number | null
}

export type ArtigoCountAggregateOutputType = {
  id: number
  titulo: number
  autores: number
  resumo: number
  areas: number
  pdfUrl: number
  edicaoId: number
  _all: number
}


export type ArtigoAvgAggregateInputType = {
  id?: true
  edicaoId?: true
}

export type ArtigoSumAggregateInputType = {
  id?: true
  edicaoId?: true
}

export type ArtigoMinAggregateInputType = {
  id?: true
  titulo?: true
  resumo?: true
  pdfUrl?: true
  edicaoId?: true
}

export type ArtigoMaxAggregateInputType = {
  id?: true
  titulo?: true
  resumo?: true
  pdfUrl?: true
  edicaoId?: true
}

export type ArtigoCountAggregateInputType = {
  id?: true
  titulo?: true
  autores?: true
  resumo?: true
  areas?: true
  pdfUrl?: true
  edicaoId?: true
  _all?: true
}

export type ArtigoAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Artigo to aggregate.
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Artigos to fetch.
   */
  orderBy?: Prisma.ArtigoOrderByWithRelationInput | Prisma.ArtigoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.ArtigoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Artigos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Artigos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Artigos
  **/
  _count?: true | ArtigoCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: ArtigoAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: ArtigoSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: ArtigoMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: ArtigoMaxAggregateInputType
}

export type GetArtigoAggregateType<T extends ArtigoAggregateArgs> = {
      [P in keyof T & keyof AggregateArtigo]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateArtigo[P]>
    : Prisma.GetScalarType<T[P], AggregateArtigo[P]>
}




export type ArtigoGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.ArtigoWhereInput
  orderBy?: Prisma.ArtigoOrderByWithAggregationInput | Prisma.ArtigoOrderByWithAggregationInput[]
  by: Prisma.ArtigoScalarFieldEnum[] | Prisma.ArtigoScalarFieldEnum
  having?: Prisma.ArtigoScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: ArtigoCountAggregateInputType | true
  _avg?: ArtigoAvgAggregateInputType
  _sum?: ArtigoSumAggregateInputType
  _min?: ArtigoMinAggregateInputType
  _max?: ArtigoMaxAggregateInputType
}

export type ArtigoGroupByOutputType = {
  id: number
  titulo: string
  autores: runtime.JsonValue
  resumo: string
  areas: runtime.JsonValue
  pdfUrl: string | null
  edicaoId: number
  _count: ArtigoCountAggregateOutputType | null
  _avg: ArtigoAvgAggregateOutputType | null
  _sum: ArtigoSumAggregateOutputType | null
  _min: ArtigoMinAggregateOutputType | null
  _max: ArtigoMaxAggregateOutputType | null
}

type GetArtigoGroupByPayload<T extends ArtigoGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<ArtigoGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof ArtigoGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], ArtigoGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], ArtigoGroupByOutputType[P]>
      }
    >
  >



export type ArtigoWhereInput = {
  AND?: Prisma.ArtigoWhereInput | Prisma.ArtigoWhereInput[]
  OR?: Prisma.ArtigoWhereInput[]
  NOT?: Prisma.ArtigoWhereInput | Prisma.ArtigoWhereInput[]
  id?: Prisma.IntFilter<"Artigo"> | number
  titulo?: Prisma.StringFilter<"Artigo"> | string
  autores?: Prisma.JsonFilter<"Artigo">
  resumo?: Prisma.StringFilter<"Artigo"> | string
  areas?: Prisma.JsonFilter<"Artigo">
  pdfUrl?: Prisma.StringNullableFilter<"Artigo"> | string | null
  edicaoId?: Prisma.IntFilter<"Artigo"> | number
  edicao?: Prisma.XOR<Prisma.EdicaoEventoScalarRelationFilter, Prisma.EdicaoEventoWhereInput>
  atribuicoes?: Prisma.AtribuicaoListRelationFilter
  perguntas?: Prisma.PerguntaListRelationFilter
}

export type ArtigoOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  titulo?: Prisma.SortOrder
  autores?: Prisma.SortOrder
  resumo?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  pdfUrl?: Prisma.SortOrderInput | Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
  edicao?: Prisma.EdicaoEventoOrderByWithRelationInput
  atribuicoes?: Prisma.AtribuicaoOrderByRelationAggregateInput
  perguntas?: Prisma.PerguntaOrderByRelationAggregateInput
}

export type ArtigoWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  AND?: Prisma.ArtigoWhereInput | Prisma.ArtigoWhereInput[]
  OR?: Prisma.ArtigoWhereInput[]
  NOT?: Prisma.ArtigoWhereInput | Prisma.ArtigoWhereInput[]
  titulo?: Prisma.StringFilter<"Artigo"> | string
  autores?: Prisma.JsonFilter<"Artigo">
  resumo?: Prisma.StringFilter<"Artigo"> | string
  areas?: Prisma.JsonFilter<"Artigo">
  pdfUrl?: Prisma.StringNullableFilter<"Artigo"> | string | null
  edicaoId?: Prisma.IntFilter<"Artigo"> | number
  edicao?: Prisma.XOR<Prisma.EdicaoEventoScalarRelationFilter, Prisma.EdicaoEventoWhereInput>
  atribuicoes?: Prisma.AtribuicaoListRelationFilter
  perguntas?: Prisma.PerguntaListRelationFilter
}, "id">

export type ArtigoOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  titulo?: Prisma.SortOrder
  autores?: Prisma.SortOrder
  resumo?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  pdfUrl?: Prisma.SortOrderInput | Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
  _count?: Prisma.ArtigoCountOrderByAggregateInput
  _avg?: Prisma.ArtigoAvgOrderByAggregateInput
  _max?: Prisma.ArtigoMaxOrderByAggregateInput
  _min?: Prisma.ArtigoMinOrderByAggregateInput
  _sum?: Prisma.ArtigoSumOrderByAggregateInput
}

export type ArtigoScalarWhereWithAggregatesInput = {
  AND?: Prisma.ArtigoScalarWhereWithAggregatesInput | Prisma.ArtigoScalarWhereWithAggregatesInput[]
  OR?: Prisma.ArtigoScalarWhereWithAggregatesInput[]
  NOT?: Prisma.ArtigoScalarWhereWithAggregatesInput | Prisma.ArtigoScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"Artigo"> | number
  titulo?: Prisma.StringWithAggregatesFilter<"Artigo"> | string
  autores?: Prisma.JsonWithAggregatesFilter<"Artigo">
  resumo?: Prisma.StringWithAggregatesFilter<"Artigo"> | string
  areas?: Prisma.JsonWithAggregatesFilter<"Artigo">
  pdfUrl?: Prisma.StringNullableWithAggregatesFilter<"Artigo"> | string | null
  edicaoId?: Prisma.IntWithAggregatesFilter<"Artigo"> | number
}

export type ArtigoCreateInput = {
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicao: Prisma.EdicaoEventoCreateNestedOneWithoutArtigosInput
  atribuicoes?: Prisma.AtribuicaoCreateNestedManyWithoutArtigoInput
  perguntas?: Prisma.PerguntaCreateNestedManyWithoutArtigoInput
}

export type ArtigoUncheckedCreateInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicaoId: number
  atribuicoes?: Prisma.AtribuicaoUncheckedCreateNestedManyWithoutArtigoInput
  perguntas?: Prisma.PerguntaUncheckedCreateNestedManyWithoutArtigoInput
}

export type ArtigoUpdateInput = {
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicao?: Prisma.EdicaoEventoUpdateOneRequiredWithoutArtigosNestedInput
  atribuicoes?: Prisma.AtribuicaoUpdateManyWithoutArtigoNestedInput
  perguntas?: Prisma.PerguntaUpdateManyWithoutArtigoNestedInput
}

export type ArtigoUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  atribuicoes?: Prisma.AtribuicaoUncheckedUpdateManyWithoutArtigoNestedInput
  perguntas?: Prisma.PerguntaUncheckedUpdateManyWithoutArtigoNestedInput
}

export type ArtigoCreateManyInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicaoId: number
}

export type ArtigoUpdateManyMutationInput = {
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type ArtigoUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type ArtigoListRelationFilter = {
  every?: Prisma.ArtigoWhereInput
  some?: Prisma.ArtigoWhereInput
  none?: Prisma.ArtigoWhereInput
}

export type ArtigoOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type ArtigoCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  titulo?: Prisma.SortOrder
  autores?: Prisma.SortOrder
  resumo?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  pdfUrl?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type ArtigoAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type ArtigoMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  titulo?: Prisma.SortOrder
  resumo?: Prisma.SortOrder
  pdfUrl?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type ArtigoMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  titulo?: Prisma.SortOrder
  resumo?: Prisma.SortOrder
  pdfUrl?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type ArtigoSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type ArtigoScalarRelationFilter = {
  is?: Prisma.ArtigoWhereInput
  isNot?: Prisma.ArtigoWhereInput
}

export type ArtigoCreateNestedManyWithoutEdicaoInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput> | Prisma.ArtigoCreateWithoutEdicaoInput[] | Prisma.ArtigoUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutEdicaoInput | Prisma.ArtigoCreateOrConnectWithoutEdicaoInput[]
  createMany?: Prisma.ArtigoCreateManyEdicaoInputEnvelope
  connect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
}

export type ArtigoUncheckedCreateNestedManyWithoutEdicaoInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput> | Prisma.ArtigoCreateWithoutEdicaoInput[] | Prisma.ArtigoUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutEdicaoInput | Prisma.ArtigoCreateOrConnectWithoutEdicaoInput[]
  createMany?: Prisma.ArtigoCreateManyEdicaoInputEnvelope
  connect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
}

export type ArtigoUpdateManyWithoutEdicaoNestedInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput> | Prisma.ArtigoCreateWithoutEdicaoInput[] | Prisma.ArtigoUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutEdicaoInput | Prisma.ArtigoCreateOrConnectWithoutEdicaoInput[]
  upsert?: Prisma.ArtigoUpsertWithWhereUniqueWithoutEdicaoInput | Prisma.ArtigoUpsertWithWhereUniqueWithoutEdicaoInput[]
  createMany?: Prisma.ArtigoCreateManyEdicaoInputEnvelope
  set?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  disconnect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  delete?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  connect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  update?: Prisma.ArtigoUpdateWithWhereUniqueWithoutEdicaoInput | Prisma.ArtigoUpdateWithWhereUniqueWithoutEdicaoInput[]
  updateMany?: Prisma.ArtigoUpdateManyWithWhereWithoutEdicaoInput | Prisma.ArtigoUpdateManyWithWhereWithoutEdicaoInput[]
  deleteMany?: Prisma.ArtigoScalarWhereInput | Prisma.ArtigoScalarWhereInput[]
}

export type ArtigoUncheckedUpdateManyWithoutEdicaoNestedInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput> | Prisma.ArtigoCreateWithoutEdicaoInput[] | Prisma.ArtigoUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutEdicaoInput | Prisma.ArtigoCreateOrConnectWithoutEdicaoInput[]
  upsert?: Prisma.ArtigoUpsertWithWhereUniqueWithoutEdicaoInput | Prisma.ArtigoUpsertWithWhereUniqueWithoutEdicaoInput[]
  createMany?: Prisma.ArtigoCreateManyEdicaoInputEnvelope
  set?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  disconnect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  delete?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  connect?: Prisma.ArtigoWhereUniqueInput | Prisma.ArtigoWhereUniqueInput[]
  update?: Prisma.ArtigoUpdateWithWhereUniqueWithoutEdicaoInput | Prisma.ArtigoUpdateWithWhereUniqueWithoutEdicaoInput[]
  updateMany?: Prisma.ArtigoUpdateManyWithWhereWithoutEdicaoInput | Prisma.ArtigoUpdateManyWithWhereWithoutEdicaoInput[]
  deleteMany?: Prisma.ArtigoScalarWhereInput | Prisma.ArtigoScalarWhereInput[]
}

export type NullableStringFieldUpdateOperationsInput = {
  set?: string | null
}

export type ArtigoCreateNestedOneWithoutAtribuicoesInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedCreateWithoutAtribuicoesInput>
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutAtribuicoesInput
  connect?: Prisma.ArtigoWhereUniqueInput
}

export type ArtigoUpdateOneRequiredWithoutAtribuicoesNestedInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedCreateWithoutAtribuicoesInput>
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutAtribuicoesInput
  upsert?: Prisma.ArtigoUpsertWithoutAtribuicoesInput
  connect?: Prisma.ArtigoWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.ArtigoUpdateToOneWithWhereWithoutAtribuicoesInput, Prisma.ArtigoUpdateWithoutAtribuicoesInput>, Prisma.ArtigoUncheckedUpdateWithoutAtribuicoesInput>
}

export type ArtigoCreateNestedOneWithoutPerguntasInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutPerguntasInput, Prisma.ArtigoUncheckedCreateWithoutPerguntasInput>
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutPerguntasInput
  connect?: Prisma.ArtigoWhereUniqueInput
}

export type ArtigoUpdateOneRequiredWithoutPerguntasNestedInput = {
  create?: Prisma.XOR<Prisma.ArtigoCreateWithoutPerguntasInput, Prisma.ArtigoUncheckedCreateWithoutPerguntasInput>
  connectOrCreate?: Prisma.ArtigoCreateOrConnectWithoutPerguntasInput
  upsert?: Prisma.ArtigoUpsertWithoutPerguntasInput
  connect?: Prisma.ArtigoWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.ArtigoUpdateToOneWithWhereWithoutPerguntasInput, Prisma.ArtigoUpdateWithoutPerguntasInput>, Prisma.ArtigoUncheckedUpdateWithoutPerguntasInput>
}

export type ArtigoCreateWithoutEdicaoInput = {
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  atribuicoes?: Prisma.AtribuicaoCreateNestedManyWithoutArtigoInput
  perguntas?: Prisma.PerguntaCreateNestedManyWithoutArtigoInput
}

export type ArtigoUncheckedCreateWithoutEdicaoInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  atribuicoes?: Prisma.AtribuicaoUncheckedCreateNestedManyWithoutArtigoInput
  perguntas?: Prisma.PerguntaUncheckedCreateNestedManyWithoutArtigoInput
}

export type ArtigoCreateOrConnectWithoutEdicaoInput = {
  where: Prisma.ArtigoWhereUniqueInput
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput>
}

export type ArtigoCreateManyEdicaoInputEnvelope = {
  data: Prisma.ArtigoCreateManyEdicaoInput | Prisma.ArtigoCreateManyEdicaoInput[]
}

export type ArtigoUpsertWithWhereUniqueWithoutEdicaoInput = {
  where: Prisma.ArtigoWhereUniqueInput
  update: Prisma.XOR<Prisma.ArtigoUpdateWithoutEdicaoInput, Prisma.ArtigoUncheckedUpdateWithoutEdicaoInput>
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutEdicaoInput, Prisma.ArtigoUncheckedCreateWithoutEdicaoInput>
}

export type ArtigoUpdateWithWhereUniqueWithoutEdicaoInput = {
  where: Prisma.ArtigoWhereUniqueInput
  data: Prisma.XOR<Prisma.ArtigoUpdateWithoutEdicaoInput, Prisma.ArtigoUncheckedUpdateWithoutEdicaoInput>
}

export type ArtigoUpdateManyWithWhereWithoutEdicaoInput = {
  where: Prisma.ArtigoScalarWhereInput
  data: Prisma.XOR<Prisma.ArtigoUpdateManyMutationInput, Prisma.ArtigoUncheckedUpdateManyWithoutEdicaoInput>
}

export type ArtigoScalarWhereInput = {
  AND?: Prisma.ArtigoScalarWhereInput | Prisma.ArtigoScalarWhereInput[]
  OR?: Prisma.ArtigoScalarWhereInput[]
  NOT?: Prisma.ArtigoScalarWhereInput | Prisma.ArtigoScalarWhereInput[]
  id?: Prisma.IntFilter<"Artigo"> | number
  titulo?: Prisma.StringFilter<"Artigo"> | string
  autores?: Prisma.JsonFilter<"Artigo">
  resumo?: Prisma.StringFilter<"Artigo"> | string
  areas?: Prisma.JsonFilter<"Artigo">
  pdfUrl?: Prisma.StringNullableFilter<"Artigo"> | string | null
  edicaoId?: Prisma.IntFilter<"Artigo"> | number
}

export type ArtigoCreateWithoutAtribuicoesInput = {
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicao: Prisma.EdicaoEventoCreateNestedOneWithoutArtigosInput
  perguntas?: Prisma.PerguntaCreateNestedManyWithoutArtigoInput
}

export type ArtigoUncheckedCreateWithoutAtribuicoesInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicaoId: number
  perguntas?: Prisma.PerguntaUncheckedCreateNestedManyWithoutArtigoInput
}

export type ArtigoCreateOrConnectWithoutAtribuicoesInput = {
  where: Prisma.ArtigoWhereUniqueInput
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedCreateWithoutAtribuicoesInput>
}

export type ArtigoUpsertWithoutAtribuicoesInput = {
  update: Prisma.XOR<Prisma.ArtigoUpdateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedUpdateWithoutAtribuicoesInput>
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedCreateWithoutAtribuicoesInput>
  where?: Prisma.ArtigoWhereInput
}

export type ArtigoUpdateToOneWithWhereWithoutAtribuicoesInput = {
  where?: Prisma.ArtigoWhereInput
  data: Prisma.XOR<Prisma.ArtigoUpdateWithoutAtribuicoesInput, Prisma.ArtigoUncheckedUpdateWithoutAtribuicoesInput>
}

export type ArtigoUpdateWithoutAtribuicoesInput = {
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicao?: Prisma.EdicaoEventoUpdateOneRequiredWithoutArtigosNestedInput
  perguntas?: Prisma.PerguntaUpdateManyWithoutArtigoNestedInput
}

export type ArtigoUncheckedUpdateWithoutAtribuicoesInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  perguntas?: Prisma.PerguntaUncheckedUpdateManyWithoutArtigoNestedInput
}

export type ArtigoCreateWithoutPerguntasInput = {
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicao: Prisma.EdicaoEventoCreateNestedOneWithoutArtigosInput
  atribuicoes?: Prisma.AtribuicaoCreateNestedManyWithoutArtigoInput
}

export type ArtigoUncheckedCreateWithoutPerguntasInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
  edicaoId: number
  atribuicoes?: Prisma.AtribuicaoUncheckedCreateNestedManyWithoutArtigoInput
}

export type ArtigoCreateOrConnectWithoutPerguntasInput = {
  where: Prisma.ArtigoWhereUniqueInput
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutPerguntasInput, Prisma.ArtigoUncheckedCreateWithoutPerguntasInput>
}

export type ArtigoUpsertWithoutPerguntasInput = {
  update: Prisma.XOR<Prisma.ArtigoUpdateWithoutPerguntasInput, Prisma.ArtigoUncheckedUpdateWithoutPerguntasInput>
  create: Prisma.XOR<Prisma.ArtigoCreateWithoutPerguntasInput, Prisma.ArtigoUncheckedCreateWithoutPerguntasInput>
  where?: Prisma.ArtigoWhereInput
}

export type ArtigoUpdateToOneWithWhereWithoutPerguntasInput = {
  where?: Prisma.ArtigoWhereInput
  data: Prisma.XOR<Prisma.ArtigoUpdateWithoutPerguntasInput, Prisma.ArtigoUncheckedUpdateWithoutPerguntasInput>
}

export type ArtigoUpdateWithoutPerguntasInput = {
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicao?: Prisma.EdicaoEventoUpdateOneRequiredWithoutArtigosNestedInput
  atribuicoes?: Prisma.AtribuicaoUpdateManyWithoutArtigoNestedInput
}

export type ArtigoUncheckedUpdateWithoutPerguntasInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  atribuicoes?: Prisma.AtribuicaoUncheckedUpdateManyWithoutArtigoNestedInput
}

export type ArtigoCreateManyEdicaoInput = {
  id?: number
  titulo: string
  autores: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: string | null
}

export type ArtigoUpdateWithoutEdicaoInput = {
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  atribuicoes?: Prisma.AtribuicaoUpdateManyWithoutArtigoNestedInput
  perguntas?: Prisma.PerguntaUpdateManyWithoutArtigoNestedInput
}

export type ArtigoUncheckedUpdateWithoutEdicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  atribuicoes?: Prisma.AtribuicaoUncheckedUpdateManyWithoutArtigoNestedInput
  perguntas?: Prisma.PerguntaUncheckedUpdateManyWithoutArtigoNestedInput
}

export type ArtigoUncheckedUpdateManyWithoutEdicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  titulo?: Prisma.StringFieldUpdateOperationsInput | string
  autores?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  resumo?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  pdfUrl?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}


/**
 * Count Type ArtigoCountOutputType
 */

export type ArtigoCountOutputType = {
  atribuicoes: number
  perguntas: number
}

export type ArtigoCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  atribuicoes?: boolean | ArtigoCountOutputTypeCountAtribuicoesArgs
  perguntas?: boolean | ArtigoCountOutputTypeCountPerguntasArgs
}

/**
 * ArtigoCountOutputType without action
 */
export type ArtigoCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the ArtigoCountOutputType
   */
  select?: Prisma.ArtigoCountOutputTypeSelect<ExtArgs> | null
}

/**
 * ArtigoCountOutputType without action
 */
export type ArtigoCountOutputTypeCountAtribuicoesArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.AtribuicaoWhereInput
}

/**
 * ArtigoCountOutputType without action
 */
export type ArtigoCountOutputTypeCountPerguntasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.PerguntaWhereInput
}


export type ArtigoSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  titulo?: boolean
  autores?: boolean
  resumo?: boolean
  areas?: boolean
  pdfUrl?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
  atribuicoes?: boolean | Prisma.Artigo$atribuicoesArgs<ExtArgs>
  perguntas?: boolean | Prisma.Artigo$perguntasArgs<ExtArgs>
  _count?: boolean | Prisma.ArtigoCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["artigo"]>

export type ArtigoSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  titulo?: boolean
  autores?: boolean
  resumo?: boolean
  areas?: boolean
  pdfUrl?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["artigo"]>

export type ArtigoSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  titulo?: boolean
  autores?: boolean
  resumo?: boolean
  areas?: boolean
  pdfUrl?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["artigo"]>

export type ArtigoSelectScalar = {
  id?: boolean
  titulo?: boolean
  autores?: boolean
  resumo?: boolean
  areas?: boolean
  pdfUrl?: boolean
  edicaoId?: boolean
}

export type ArtigoOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "titulo" | "autores" | "resumo" | "areas" | "pdfUrl" | "edicaoId", ExtArgs["result"]["artigo"]>
export type ArtigoInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
  atribuicoes?: boolean | Prisma.Artigo$atribuicoesArgs<ExtArgs>
  perguntas?: boolean | Prisma.Artigo$perguntasArgs<ExtArgs>
  _count?: boolean | Prisma.ArtigoCountOutputTypeDefaultArgs<ExtArgs>
}
export type ArtigoIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}
export type ArtigoIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}

export type $ArtigoPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Artigo"
  objects: {
    edicao: Prisma.$EdicaoEventoPayload<ExtArgs>
    atribuicoes: Prisma.$AtribuicaoPayload<ExtArgs>[]
    perguntas: Prisma.$PerguntaPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    titulo: string
    autores: runtime.JsonValue
    resumo: string
    areas: runtime.JsonValue
    pdfUrl: string | null
    edicaoId: number
  }, ExtArgs["result"]["artigo"]>
  composites: {}
}

export type ArtigoGetPayload<S extends boolean | null | undefined | ArtigoDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$ArtigoPayload, S>

export type ArtigoCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<ArtigoFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: ArtigoCountAggregateInputType | true
  }

export interface ArtigoDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Artigo'], meta: { name: 'Artigo' } }
  /**
   * Find zero or one Artigo that matches the filter.
   * @param {ArtigoFindUniqueArgs} args - Arguments to find a Artigo
   * @example
   * // Get one Artigo
   * const artigo = await prisma.artigo.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends ArtigoFindUniqueArgs>(args: Prisma.SelectSubset<T, ArtigoFindUniqueArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Artigo that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {ArtigoFindUniqueOrThrowArgs} args - Arguments to find a Artigo
   * @example
   * // Get one Artigo
   * const artigo = await prisma.artigo.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends ArtigoFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, ArtigoFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Artigo that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoFindFirstArgs} args - Arguments to find a Artigo
   * @example
   * // Get one Artigo
   * const artigo = await prisma.artigo.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends ArtigoFindFirstArgs>(args?: Prisma.SelectSubset<T, ArtigoFindFirstArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Artigo that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoFindFirstOrThrowArgs} args - Arguments to find a Artigo
   * @example
   * // Get one Artigo
   * const artigo = await prisma.artigo.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends ArtigoFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, ArtigoFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Artigos that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Artigos
   * const artigos = await prisma.artigo.findMany()
   * 
   * // Get first 10 Artigos
   * const artigos = await prisma.artigo.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const artigoWithIdOnly = await prisma.artigo.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends ArtigoFindManyArgs>(args?: Prisma.SelectSubset<T, ArtigoFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Artigo.
   * @param {ArtigoCreateArgs} args - Arguments to create a Artigo.
   * @example
   * // Create one Artigo
   * const Artigo = await prisma.artigo.create({
   *   data: {
   *     // ... data to create a Artigo
   *   }
   * })
   * 
   */
  create<T extends ArtigoCreateArgs>(args: Prisma.SelectSubset<T, ArtigoCreateArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Artigos.
   * @param {ArtigoCreateManyArgs} args - Arguments to create many Artigos.
   * @example
   * // Create many Artigos
   * const artigo = await prisma.artigo.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends ArtigoCreateManyArgs>(args?: Prisma.SelectSubset<T, ArtigoCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Artigos and returns the data saved in the database.
   * @param {ArtigoCreateManyAndReturnArgs} args - Arguments to create many Artigos.
   * @example
   * // Create many Artigos
   * const artigo = await prisma.artigo.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Artigos and only return the `id`
   * const artigoWithIdOnly = await prisma.artigo.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends ArtigoCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, ArtigoCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Artigo.
   * @param {ArtigoDeleteArgs} args - Arguments to delete one Artigo.
   * @example
   * // Delete one Artigo
   * const Artigo = await prisma.artigo.delete({
   *   where: {
   *     // ... filter to delete one Artigo
   *   }
   * })
   * 
   */
  delete<T extends ArtigoDeleteArgs>(args: Prisma.SelectSubset<T, ArtigoDeleteArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Artigo.
   * @param {ArtigoUpdateArgs} args - Arguments to update one Artigo.
   * @example
   * // Update one Artigo
   * const artigo = await prisma.artigo.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends ArtigoUpdateArgs>(args: Prisma.SelectSubset<T, ArtigoUpdateArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Artigos.
   * @param {ArtigoDeleteManyArgs} args - Arguments to filter Artigos to delete.
   * @example
   * // Delete a few Artigos
   * const { count } = await prisma.artigo.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends ArtigoDeleteManyArgs>(args?: Prisma.SelectSubset<T, ArtigoDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Artigos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Artigos
   * const artigo = await prisma.artigo.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends ArtigoUpdateManyArgs>(args: Prisma.SelectSubset<T, ArtigoUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Artigos and returns the data updated in the database.
   * @param {ArtigoUpdateManyAndReturnArgs} args - Arguments to update many Artigos.
   * @example
   * // Update many Artigos
   * const artigo = await prisma.artigo.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Artigos and only return the `id`
   * const artigoWithIdOnly = await prisma.artigo.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends ArtigoUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, ArtigoUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Artigo.
   * @param {ArtigoUpsertArgs} args - Arguments to update or create a Artigo.
   * @example
   * // Update or create a Artigo
   * const artigo = await prisma.artigo.upsert({
   *   create: {
   *     // ... data to create a Artigo
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Artigo we want to update
   *   }
   * })
   */
  upsert<T extends ArtigoUpsertArgs>(args: Prisma.SelectSubset<T, ArtigoUpsertArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Artigos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoCountArgs} args - Arguments to filter Artigos to count.
   * @example
   * // Count the number of Artigos
   * const count = await prisma.artigo.count({
   *   where: {
   *     // ... the filter for the Artigos we want to count
   *   }
   * })
  **/
  count<T extends ArtigoCountArgs>(
    args?: Prisma.Subset<T, ArtigoCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], ArtigoCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Artigo.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends ArtigoAggregateArgs>(args: Prisma.Subset<T, ArtigoAggregateArgs>): Prisma.PrismaPromise<GetArtigoAggregateType<T>>

  /**
   * Group by Artigo.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {ArtigoGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends ArtigoGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: ArtigoGroupByArgs['orderBy'] }
      : { orderBy?: ArtigoGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, ArtigoGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetArtigoGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Artigo model
 */
readonly fields: ArtigoFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Artigo.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__ArtigoClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  edicao<T extends Prisma.EdicaoEventoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.EdicaoEventoDefaultArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  atribuicoes<T extends Prisma.Artigo$atribuicoesArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.Artigo$atribuicoesArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  perguntas<T extends Prisma.Artigo$perguntasArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.Artigo$perguntasArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Artigo model
 */
export interface ArtigoFieldRefs {
  readonly id: Prisma.FieldRef<"Artigo", 'Int'>
  readonly titulo: Prisma.FieldRef<"Artigo", 'String'>
  readonly autores: Prisma.FieldRef<"Artigo", 'Json'>
  readonly resumo: Prisma.FieldRef<"Artigo", 'String'>
  readonly areas: Prisma.FieldRef<"Artigo", 'Json'>
  readonly pdfUrl: Prisma.FieldRef<"Artigo", 'String'>
  readonly edicaoId: Prisma.FieldRef<"Artigo", 'Int'>
}
    

// Custom InputTypes
/**
 * Artigo findUnique
 */
export type ArtigoFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter, which Artigo to fetch.
   */
  where: Prisma.ArtigoWhereUniqueInput
}

/**
 * Artigo findUniqueOrThrow
 */
export type ArtigoFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter, which Artigo to fetch.
   */
  where: Prisma.ArtigoWhereUniqueInput
}

/**
 * Artigo findFirst
 */
export type ArtigoFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter, which Artigo to fetch.
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Artigos to fetch.
   */
  orderBy?: Prisma.ArtigoOrderByWithRelationInput | Prisma.ArtigoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Artigos.
   */
  cursor?: Prisma.ArtigoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Artigos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Artigos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Artigos.
   */
  distinct?: Prisma.ArtigoScalarFieldEnum | Prisma.ArtigoScalarFieldEnum[]
}

/**
 * Artigo findFirstOrThrow
 */
export type ArtigoFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter, which Artigo to fetch.
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Artigos to fetch.
   */
  orderBy?: Prisma.ArtigoOrderByWithRelationInput | Prisma.ArtigoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Artigos.
   */
  cursor?: Prisma.ArtigoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Artigos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Artigos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Artigos.
   */
  distinct?: Prisma.ArtigoScalarFieldEnum | Prisma.ArtigoScalarFieldEnum[]
}

/**
 * Artigo findMany
 */
export type ArtigoFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter, which Artigos to fetch.
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Artigos to fetch.
   */
  orderBy?: Prisma.ArtigoOrderByWithRelationInput | Prisma.ArtigoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Artigos.
   */
  cursor?: Prisma.ArtigoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Artigos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Artigos.
   */
  skip?: number
  distinct?: Prisma.ArtigoScalarFieldEnum | Prisma.ArtigoScalarFieldEnum[]
}

/**
 * Artigo create
 */
export type ArtigoCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * The data needed to create a Artigo.
   */
  data: Prisma.XOR<Prisma.ArtigoCreateInput, Prisma.ArtigoUncheckedCreateInput>
}

/**
 * Artigo createMany
 */
export type ArtigoCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Artigos.
   */
  data: Prisma.ArtigoCreateManyInput | Prisma.ArtigoCreateManyInput[]
}

/**
 * Artigo createManyAndReturn
 */
export type ArtigoCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * The data used to create many Artigos.
   */
  data: Prisma.ArtigoCreateManyInput | Prisma.ArtigoCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Artigo update
 */
export type ArtigoUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * The data needed to update a Artigo.
   */
  data: Prisma.XOR<Prisma.ArtigoUpdateInput, Prisma.ArtigoUncheckedUpdateInput>
  /**
   * Choose, which Artigo to update.
   */
  where: Prisma.ArtigoWhereUniqueInput
}

/**
 * Artigo updateMany
 */
export type ArtigoUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Artigos.
   */
  data: Prisma.XOR<Prisma.ArtigoUpdateManyMutationInput, Prisma.ArtigoUncheckedUpdateManyInput>
  /**
   * Filter which Artigos to update
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * Limit how many Artigos to update.
   */
  limit?: number
}

/**
 * Artigo updateManyAndReturn
 */
export type ArtigoUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * The data used to update Artigos.
   */
  data: Prisma.XOR<Prisma.ArtigoUpdateManyMutationInput, Prisma.ArtigoUncheckedUpdateManyInput>
  /**
   * Filter which Artigos to update
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * Limit how many Artigos to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Artigo upsert
 */
export type ArtigoUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * The filter to search for the Artigo to update in case it exists.
   */
  where: Prisma.ArtigoWhereUniqueInput
  /**
   * In case the Artigo found by the `where` argument doesn't exist, create a new Artigo with this data.
   */
  create: Prisma.XOR<Prisma.ArtigoCreateInput, Prisma.ArtigoUncheckedCreateInput>
  /**
   * In case the Artigo was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.ArtigoUpdateInput, Prisma.ArtigoUncheckedUpdateInput>
}

/**
 * Artigo delete
 */
export type ArtigoDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  /**
   * Filter which Artigo to delete.
   */
  where: Prisma.ArtigoWhereUniqueInput
}

/**
 * Artigo deleteMany
 */
export type ArtigoDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Artigos to delete
   */
  where?: Prisma.ArtigoWhereInput
  /**
   * Limit how many Artigos to delete.
   */
  limit?: number
}

/**
 * Artigo.atribuicoes
 */
export type Artigo$atribuicoesArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  where?: Prisma.AtribuicaoWhereInput
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.AtribuicaoScalarFieldEnum | Prisma.AtribuicaoScalarFieldEnum[]
}

/**
 * Artigo.perguntas
 */
export type Artigo$perguntasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  where?: Prisma.PerguntaWhereInput
  orderBy?: Prisma.PerguntaOrderByWithRelationInput | Prisma.PerguntaOrderByWithRelationInput[]
  cursor?: Prisma.PerguntaWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.PerguntaScalarFieldEnum | Prisma.PerguntaScalarFieldEnum[]
}

/**
 * Artigo without action
 */
export type ArtigoDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
}
