// @prisma/client/models/Atribuicao.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Atribuicao` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Atribuicao
 * 
 */
export type AtribuicaoModel = runtime.Types.Result.DefaultSelection<Prisma.$AtribuicaoPayload>

export type AggregateAtribuicao = {
  _count: AtribuicaoCountAggregateOutputType | null
  _avg: AtribuicaoAvgAggregateOutputType | null
  _sum: AtribuicaoSumAggregateOutputType | null
  _min: AtribuicaoMinAggregateOutputType | null
  _max: AtribuicaoMaxAggregateOutputType | null
}

export type AtribuicaoAvgAggregateOutputType = {
  id: number | null
  artigoId: number | null
  revisorId: number | null
}

export type AtribuicaoSumAggregateOutputType = {
  id: number | null
  artigoId: number | null
  revisorId: number | null
}

export type AtribuicaoMinAggregateOutputType = {
  id: number | null
  artigoId: number | null
  revisorId: number | null
}

export type AtribuicaoMaxAggregateOutputType = {
  id: number | null
  artigoId: number | null
  revisorId: number | null
}

export type AtribuicaoCountAggregateOutputType = {
  id: number
  artigoId: number
  revisorId: number
  _all: number
}


export type AtribuicaoAvgAggregateInputType = {
  id?: true
  artigoId?: true
  revisorId?: true
}

export type AtribuicaoSumAggregateInputType = {
  id?: true
  artigoId?: true
  revisorId?: true
}

export type AtribuicaoMinAggregateInputType = {
  id?: true
  artigoId?: true
  revisorId?: true
}

export type AtribuicaoMaxAggregateInputType = {
  id?: true
  artigoId?: true
  revisorId?: true
}

export type AtribuicaoCountAggregateInputType = {
  id?: true
  artigoId?: true
  revisorId?: true
  _all?: true
}

export type AtribuicaoAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Atribuicao to aggregate.
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Atribuicaos to fetch.
   */
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Atribuicaos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Atribuicaos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Atribuicaos
  **/
  _count?: true | AtribuicaoCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: AtribuicaoAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: AtribuicaoSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: AtribuicaoMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: AtribuicaoMaxAggregateInputType
}

export type GetAtribuicaoAggregateType<T extends AtribuicaoAggregateArgs> = {
      [P in keyof T & keyof AggregateAtribuicao]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateAtribuicao[P]>
    : Prisma.GetScalarType<T[P], AggregateAtribuicao[P]>
}




export type AtribuicaoGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.AtribuicaoWhereInput
  orderBy?: Prisma.AtribuicaoOrderByWithAggregationInput | Prisma.AtribuicaoOrderByWithAggregationInput[]
  by: Prisma.AtribuicaoScalarFieldEnum[] | Prisma.AtribuicaoScalarFieldEnum
  having?: Prisma.AtribuicaoScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: AtribuicaoCountAggregateInputType | true
  _avg?: AtribuicaoAvgAggregateInputType
  _sum?: AtribuicaoSumAggregateInputType
  _min?: AtribuicaoMinAggregateInputType
  _max?: AtribuicaoMaxAggregateInputType
}

export type AtribuicaoGroupByOutputType = {
  id: number
  artigoId: number
  revisorId: number
  _count: AtribuicaoCountAggregateOutputType | null
  _avg: AtribuicaoAvgAggregateOutputType | null
  _sum: AtribuicaoSumAggregateOutputType | null
  _min: AtribuicaoMinAggregateOutputType | null
  _max: AtribuicaoMaxAggregateOutputType | null
}

type GetAtribuicaoGroupByPayload<T extends AtribuicaoGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<AtribuicaoGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof AtribuicaoGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], AtribuicaoGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], AtribuicaoGroupByOutputType[P]>
      }
    >
  >



export type AtribuicaoWhereInput = {
  AND?: Prisma.AtribuicaoWhereInput | Prisma.AtribuicaoWhereInput[]
  OR?: Prisma.AtribuicaoWhereInput[]
  NOT?: Prisma.AtribuicaoWhereInput | Prisma.AtribuicaoWhereInput[]
  id?: Prisma.IntFilter<"Atribuicao"> | number
  artigoId?: Prisma.IntFilter<"Atribuicao"> | number
  revisorId?: Prisma.IntFilter<"Atribuicao"> | number
  artigo?: Prisma.XOR<Prisma.ArtigoScalarRelationFilter, Prisma.ArtigoWhereInput>
  revisor?: Prisma.XOR<Prisma.RevisorScalarRelationFilter, Prisma.RevisorWhereInput>
  respostas?: Prisma.RespostaListRelationFilter
}

export type AtribuicaoOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
  artigo?: Prisma.ArtigoOrderByWithRelationInput
  revisor?: Prisma.RevisorOrderByWithRelationInput
  respostas?: Prisma.RespostaOrderByRelationAggregateInput
}

export type AtribuicaoWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  artigoId_revisorId?: Prisma.AtribuicaoArtigoIdRevisorIdCompoundUniqueInput
  AND?: Prisma.AtribuicaoWhereInput | Prisma.AtribuicaoWhereInput[]
  OR?: Prisma.AtribuicaoWhereInput[]
  NOT?: Prisma.AtribuicaoWhereInput | Prisma.AtribuicaoWhereInput[]
  artigoId?: Prisma.IntFilter<"Atribuicao"> | number
  revisorId?: Prisma.IntFilter<"Atribuicao"> | number
  artigo?: Prisma.XOR<Prisma.ArtigoScalarRelationFilter, Prisma.ArtigoWhereInput>
  revisor?: Prisma.XOR<Prisma.RevisorScalarRelationFilter, Prisma.RevisorWhereInput>
  respostas?: Prisma.RespostaListRelationFilter
}, "id" | "artigoId_revisorId">

export type AtribuicaoOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
  _count?: Prisma.AtribuicaoCountOrderByAggregateInput
  _avg?: Prisma.AtribuicaoAvgOrderByAggregateInput
  _max?: Prisma.AtribuicaoMaxOrderByAggregateInput
  _min?: Prisma.AtribuicaoMinOrderByAggregateInput
  _sum?: Prisma.AtribuicaoSumOrderByAggregateInput
}

export type AtribuicaoScalarWhereWithAggregatesInput = {
  AND?: Prisma.AtribuicaoScalarWhereWithAggregatesInput | Prisma.AtribuicaoScalarWhereWithAggregatesInput[]
  OR?: Prisma.AtribuicaoScalarWhereWithAggregatesInput[]
  NOT?: Prisma.AtribuicaoScalarWhereWithAggregatesInput | Prisma.AtribuicaoScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"Atribuicao"> | number
  artigoId?: Prisma.IntWithAggregatesFilter<"Atribuicao"> | number
  revisorId?: Prisma.IntWithAggregatesFilter<"Atribuicao"> | number
}

export type AtribuicaoCreateInput = {
  artigo: Prisma.ArtigoCreateNestedOneWithoutAtribuicoesInput
  revisor: Prisma.RevisorCreateNestedOneWithoutAtribuicoesInput
  respostas?: Prisma.RespostaCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoUncheckedCreateInput = {
  id?: number
  artigoId: number
  revisorId: number
  respostas?: Prisma.RespostaUncheckedCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoUpdateInput = {
  artigo?: Prisma.ArtigoUpdateOneRequiredWithoutAtribuicoesNestedInput
  revisor?: Prisma.RevisorUpdateOneRequiredWithoutAtribuicoesNestedInput
  respostas?: Prisma.RespostaUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
  revisorId?: Prisma.IntFieldUpdateOperationsInput | number
  respostas?: Prisma.RespostaUncheckedUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoCreateManyInput = {
  id?: number
  artigoId: number
  revisorId: number
}

export type AtribuicaoUpdateManyMutationInput = {

}

export type AtribuicaoUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
  revisorId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type AtribuicaoListRelationFilter = {
  every?: Prisma.AtribuicaoWhereInput
  some?: Prisma.AtribuicaoWhereInput
  none?: Prisma.AtribuicaoWhereInput
}

export type AtribuicaoOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type AtribuicaoArtigoIdRevisorIdCompoundUniqueInput = {
  artigoId: number
  revisorId: number
}

export type AtribuicaoCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
}

export type AtribuicaoAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
}

export type AtribuicaoMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
}

export type AtribuicaoMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
}

export type AtribuicaoSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  revisorId?: Prisma.SortOrder
}

export type AtribuicaoScalarRelationFilter = {
  is?: Prisma.AtribuicaoWhereInput
  isNot?: Prisma.AtribuicaoWhereInput
}

export type AtribuicaoCreateNestedManyWithoutArtigoInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput> | Prisma.AtribuicaoCreateWithoutArtigoInput[] | Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput | Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput[]
  createMany?: Prisma.AtribuicaoCreateManyArtigoInputEnvelope
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
}

export type AtribuicaoUncheckedCreateNestedManyWithoutArtigoInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput> | Prisma.AtribuicaoCreateWithoutArtigoInput[] | Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput | Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput[]
  createMany?: Prisma.AtribuicaoCreateManyArtigoInputEnvelope
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
}

export type AtribuicaoUpdateManyWithoutArtigoNestedInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput> | Prisma.AtribuicaoCreateWithoutArtigoInput[] | Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput | Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput[]
  upsert?: Prisma.AtribuicaoUpsertWithWhereUniqueWithoutArtigoInput | Prisma.AtribuicaoUpsertWithWhereUniqueWithoutArtigoInput[]
  createMany?: Prisma.AtribuicaoCreateManyArtigoInputEnvelope
  set?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  disconnect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  delete?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  update?: Prisma.AtribuicaoUpdateWithWhereUniqueWithoutArtigoInput | Prisma.AtribuicaoUpdateWithWhereUniqueWithoutArtigoInput[]
  updateMany?: Prisma.AtribuicaoUpdateManyWithWhereWithoutArtigoInput | Prisma.AtribuicaoUpdateManyWithWhereWithoutArtigoInput[]
  deleteMany?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
}

export type AtribuicaoUncheckedUpdateManyWithoutArtigoNestedInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput> | Prisma.AtribuicaoCreateWithoutArtigoInput[] | Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput | Prisma.AtribuicaoCreateOrConnectWithoutArtigoInput[]
  upsert?: Prisma.AtribuicaoUpsertWithWhereUniqueWithoutArtigoInput | Prisma.AtribuicaoUpsertWithWhereUniqueWithoutArtigoInput[]
  createMany?: Prisma.AtribuicaoCreateManyArtigoInputEnvelope
  set?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  disconnect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  delete?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  update?: Prisma.AtribuicaoUpdateWithWhereUniqueWithoutArtigoInput | Prisma.AtribuicaoUpdateWithWhereUniqueWithoutArtigoInput[]
  updateMany?: Prisma.AtribuicaoUpdateManyWithWhereWithoutArtigoInput | Prisma.AtribuicaoUpdateManyWithWhereWithoutArtigoInput[]
  deleteMany?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
}

export type AtribuicaoCreateNestedManyWithoutRevisorInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput> | Prisma.AtribuicaoCreateWithoutRevisorInput[] | Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput | Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput[]
  createMany?: Prisma.AtribuicaoCreateManyRevisorInputEnvelope
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
}

export type AtribuicaoUncheckedCreateNestedManyWithoutRevisorInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput> | Prisma.AtribuicaoCreateWithoutRevisorInput[] | Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput | Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput[]
  createMany?: Prisma.AtribuicaoCreateManyRevisorInputEnvelope
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
}

export type AtribuicaoUpdateManyWithoutRevisorNestedInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput> | Prisma.AtribuicaoCreateWithoutRevisorInput[] | Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput | Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput[]
  upsert?: Prisma.AtribuicaoUpsertWithWhereUniqueWithoutRevisorInput | Prisma.AtribuicaoUpsertWithWhereUniqueWithoutRevisorInput[]
  createMany?: Prisma.AtribuicaoCreateManyRevisorInputEnvelope
  set?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  disconnect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  delete?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  update?: Prisma.AtribuicaoUpdateWithWhereUniqueWithoutRevisorInput | Prisma.AtribuicaoUpdateWithWhereUniqueWithoutRevisorInput[]
  updateMany?: Prisma.AtribuicaoUpdateManyWithWhereWithoutRevisorInput | Prisma.AtribuicaoUpdateManyWithWhereWithoutRevisorInput[]
  deleteMany?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
}

export type AtribuicaoUncheckedUpdateManyWithoutRevisorNestedInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput> | Prisma.AtribuicaoCreateWithoutRevisorInput[] | Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput[]
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput | Prisma.AtribuicaoCreateOrConnectWithoutRevisorInput[]
  upsert?: Prisma.AtribuicaoUpsertWithWhereUniqueWithoutRevisorInput | Prisma.AtribuicaoUpsertWithWhereUniqueWithoutRevisorInput[]
  createMany?: Prisma.AtribuicaoCreateManyRevisorInputEnvelope
  set?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  disconnect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  delete?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  connect?: Prisma.AtribuicaoWhereUniqueInput | Prisma.AtribuicaoWhereUniqueInput[]
  update?: Prisma.AtribuicaoUpdateWithWhereUniqueWithoutRevisorInput | Prisma.AtribuicaoUpdateWithWhereUniqueWithoutRevisorInput[]
  updateMany?: Prisma.AtribuicaoUpdateManyWithWhereWithoutRevisorInput | Prisma.AtribuicaoUpdateManyWithWhereWithoutRevisorInput[]
  deleteMany?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
}

export type AtribuicaoCreateNestedOneWithoutRespostasInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRespostasInput, Prisma.AtribuicaoUncheckedCreateWithoutRespostasInput>
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRespostasInput
  connect?: Prisma.AtribuicaoWhereUniqueInput
}

export type AtribuicaoUpdateOneRequiredWithoutRespostasNestedInput = {
  create?: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRespostasInput, Prisma.AtribuicaoUncheckedCreateWithoutRespostasInput>
  connectOrCreate?: Prisma.AtribuicaoCreateOrConnectWithoutRespostasInput
  upsert?: Prisma.AtribuicaoUpsertWithoutRespostasInput
  connect?: Prisma.AtribuicaoWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.AtribuicaoUpdateToOneWithWhereWithoutRespostasInput, Prisma.AtribuicaoUpdateWithoutRespostasInput>, Prisma.AtribuicaoUncheckedUpdateWithoutRespostasInput>
}

export type AtribuicaoCreateWithoutArtigoInput = {
  revisor: Prisma.RevisorCreateNestedOneWithoutAtribuicoesInput
  respostas?: Prisma.RespostaCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoUncheckedCreateWithoutArtigoInput = {
  id?: number
  revisorId: number
  respostas?: Prisma.RespostaUncheckedCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoCreateOrConnectWithoutArtigoInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput>
}

export type AtribuicaoCreateManyArtigoInputEnvelope = {
  data: Prisma.AtribuicaoCreateManyArtigoInput | Prisma.AtribuicaoCreateManyArtigoInput[]
}

export type AtribuicaoUpsertWithWhereUniqueWithoutArtigoInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  update: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutArtigoInput, Prisma.AtribuicaoUncheckedUpdateWithoutArtigoInput>
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutArtigoInput, Prisma.AtribuicaoUncheckedCreateWithoutArtigoInput>
}

export type AtribuicaoUpdateWithWhereUniqueWithoutArtigoInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  data: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutArtigoInput, Prisma.AtribuicaoUncheckedUpdateWithoutArtigoInput>
}

export type AtribuicaoUpdateManyWithWhereWithoutArtigoInput = {
  where: Prisma.AtribuicaoScalarWhereInput
  data: Prisma.XOR<Prisma.AtribuicaoUpdateManyMutationInput, Prisma.AtribuicaoUncheckedUpdateManyWithoutArtigoInput>
}

export type AtribuicaoScalarWhereInput = {
  AND?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
  OR?: Prisma.AtribuicaoScalarWhereInput[]
  NOT?: Prisma.AtribuicaoScalarWhereInput | Prisma.AtribuicaoScalarWhereInput[]
  id?: Prisma.IntFilter<"Atribuicao"> | number
  artigoId?: Prisma.IntFilter<"Atribuicao"> | number
  revisorId?: Prisma.IntFilter<"Atribuicao"> | number
}

export type AtribuicaoCreateWithoutRevisorInput = {
  artigo: Prisma.ArtigoCreateNestedOneWithoutAtribuicoesInput
  respostas?: Prisma.RespostaCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoUncheckedCreateWithoutRevisorInput = {
  id?: number
  artigoId: number
  respostas?: Prisma.RespostaUncheckedCreateNestedManyWithoutAtribuicaoInput
}

export type AtribuicaoCreateOrConnectWithoutRevisorInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput>
}

export type AtribuicaoCreateManyRevisorInputEnvelope = {
  data: Prisma.AtribuicaoCreateManyRevisorInput | Prisma.AtribuicaoCreateManyRevisorInput[]
}

export type AtribuicaoUpsertWithWhereUniqueWithoutRevisorInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  update: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutRevisorInput, Prisma.AtribuicaoUncheckedUpdateWithoutRevisorInput>
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRevisorInput, Prisma.AtribuicaoUncheckedCreateWithoutRevisorInput>
}

export type AtribuicaoUpdateWithWhereUniqueWithoutRevisorInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  data: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutRevisorInput, Prisma.AtribuicaoUncheckedUpdateWithoutRevisorInput>
}

export type AtribuicaoUpdateManyWithWhereWithoutRevisorInput = {
  where: Prisma.AtribuicaoScalarWhereInput
  data: Prisma.XOR<Prisma.AtribuicaoUpdateManyMutationInput, Prisma.AtribuicaoUncheckedUpdateManyWithoutRevisorInput>
}

export type AtribuicaoCreateWithoutRespostasInput = {
  artigo: Prisma.ArtigoCreateNestedOneWithoutAtribuicoesInput
  revisor: Prisma.RevisorCreateNestedOneWithoutAtribuicoesInput
}

export type AtribuicaoUncheckedCreateWithoutRespostasInput = {
  id?: number
  artigoId: number
  revisorId: number
}

export type AtribuicaoCreateOrConnectWithoutRespostasInput = {
  where: Prisma.AtribuicaoWhereUniqueInput
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRespostasInput, Prisma.AtribuicaoUncheckedCreateWithoutRespostasInput>
}

export type AtribuicaoUpsertWithoutRespostasInput = {
  update: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutRespostasInput, Prisma.AtribuicaoUncheckedUpdateWithoutRespostasInput>
  create: Prisma.XOR<Prisma.AtribuicaoCreateWithoutRespostasInput, Prisma.AtribuicaoUncheckedCreateWithoutRespostasInput>
  where?: Prisma.AtribuicaoWhereInput
}

export type AtribuicaoUpdateToOneWithWhereWithoutRespostasInput = {
  where?: Prisma.AtribuicaoWhereInput
  data: Prisma.XOR<Prisma.AtribuicaoUpdateWithoutRespostasInput, Prisma.AtribuicaoUncheckedUpdateWithoutRespostasInput>
}

export type AtribuicaoUpdateWithoutRespostasInput = {
  artigo?: Prisma.ArtigoUpdateOneRequiredWithoutAtribuicoesNestedInput
  revisor?: Prisma.RevisorUpdateOneRequiredWithoutAtribuicoesNestedInput
}

export type AtribuicaoUncheckedUpdateWithoutRespostasInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
  revisorId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type AtribuicaoCreateManyArtigoInput = {
  id?: number
  revisorId: number
}

export type AtribuicaoUpdateWithoutArtigoInput = {
  revisor?: Prisma.RevisorUpdateOneRequiredWithoutAtribuicoesNestedInput
  respostas?: Prisma.RespostaUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoUncheckedUpdateWithoutArtigoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  revisorId?: Prisma.IntFieldUpdateOperationsInput | number
  respostas?: Prisma.RespostaUncheckedUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoUncheckedUpdateManyWithoutArtigoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  revisorId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type AtribuicaoCreateManyRevisorInput = {
  id?: number
  artigoId: number
}

export type AtribuicaoUpdateWithoutRevisorInput = {
  artigo?: Prisma.ArtigoUpdateOneRequiredWithoutAtribuicoesNestedInput
  respostas?: Prisma.RespostaUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoUncheckedUpdateWithoutRevisorInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
  respostas?: Prisma.RespostaUncheckedUpdateManyWithoutAtribuicaoNestedInput
}

export type AtribuicaoUncheckedUpdateManyWithoutRevisorInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
}


/**
 * Count Type AtribuicaoCountOutputType
 */

export type AtribuicaoCountOutputType = {
  respostas: number
}

export type AtribuicaoCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  respostas?: boolean | AtribuicaoCountOutputTypeCountRespostasArgs
}

/**
 * AtribuicaoCountOutputType without action
 */
export type AtribuicaoCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the AtribuicaoCountOutputType
   */
  select?: Prisma.AtribuicaoCountOutputTypeSelect<ExtArgs> | null
}

/**
 * AtribuicaoCountOutputType without action
 */
export type AtribuicaoCountOutputTypeCountRespostasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.RespostaWhereInput
}


export type AtribuicaoSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  artigoId?: boolean
  revisorId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
  respostas?: boolean | Prisma.Atribuicao$respostasArgs<ExtArgs>
  _count?: boolean | Prisma.AtribuicaoCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["atribuicao"]>

export type AtribuicaoSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  artigoId?: boolean
  revisorId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
}, ExtArgs["result"]["atribuicao"]>

export type AtribuicaoSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  artigoId?: boolean
  revisorId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
}, ExtArgs["result"]["atribuicao"]>

export type AtribuicaoSelectScalar = {
  id?: boolean
  artigoId?: boolean
  revisorId?: boolean
}

export type AtribuicaoOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "artigoId" | "revisorId", ExtArgs["result"]["atribuicao"]>
export type AtribuicaoInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
  respostas?: boolean | Prisma.Atribuicao$respostasArgs<ExtArgs>
  _count?: boolean | Prisma.AtribuicaoCountOutputTypeDefaultArgs<ExtArgs>
}
export type AtribuicaoIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
}
export type AtribuicaoIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  revisor?: boolean | Prisma.RevisorDefaultArgs<ExtArgs>
}

export type $AtribuicaoPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Atribuicao"
  objects: {
    artigo: Prisma.$ArtigoPayload<ExtArgs>
    revisor: Prisma.$RevisorPayload<ExtArgs>
    respostas: Prisma.$RespostaPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    artigoId: number
    revisorId: number
  }, ExtArgs["result"]["atribuicao"]>
  composites: {}
}

export type AtribuicaoGetPayload<S extends boolean | null | undefined | AtribuicaoDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload, S>

export type AtribuicaoCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<AtribuicaoFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: AtribuicaoCountAggregateInputType | true
  }

export interface AtribuicaoDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Atribuicao'], meta: { name: 'Atribuicao' } }
  /**
   * Find zero or one Atribuicao that matches the filter.
   * @param {AtribuicaoFindUniqueArgs} args - Arguments to find a Atribuicao
   * @example
   * // Get one Atribuicao
   * const atribuicao = await prisma.atribuicao.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends AtribuicaoFindUniqueArgs>(args: Prisma.SelectSubset<T, AtribuicaoFindUniqueArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Atribuicao that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {AtribuicaoFindUniqueOrThrowArgs} args - Arguments to find a Atribuicao
   * @example
   * // Get one Atribuicao
   * const atribuicao = await prisma.atribuicao.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends AtribuicaoFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, AtribuicaoFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Atribuicao that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoFindFirstArgs} args - Arguments to find a Atribuicao
   * @example
   * // Get one Atribuicao
   * const atribuicao = await prisma.atribuicao.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends AtribuicaoFindFirstArgs>(args?: Prisma.SelectSubset<T, AtribuicaoFindFirstArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Atribuicao that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoFindFirstOrThrowArgs} args - Arguments to find a Atribuicao
   * @example
   * // Get one Atribuicao
   * const atribuicao = await prisma.atribuicao.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends AtribuicaoFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, AtribuicaoFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Atribuicaos that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Atribuicaos
   * const atribuicaos = await prisma.atribuicao.findMany()
   * 
   * // Get first 10 Atribuicaos
   * const atribuicaos = await prisma.atribuicao.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const atribuicaoWithIdOnly = await prisma.atribuicao.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends AtribuicaoFindManyArgs>(args?: Prisma.SelectSubset<T, AtribuicaoFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Atribuicao.
   * @param {AtribuicaoCreateArgs} args - Arguments to create a Atribuicao.
   * @example
   * // Create one Atribuicao
   * const Atribuicao = await prisma.atribuicao.create({
   *   data: {
   *     // ... data to create a Atribuicao
   *   }
   * })
   * 
   */
  create<T extends AtribuicaoCreateArgs>(args: Prisma.SelectSubset<T, AtribuicaoCreateArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Atribuicaos.
   * @param {AtribuicaoCreateManyArgs} args - Arguments to create many Atribuicaos.
   * @example
   * // Create many Atribuicaos
   * const atribuicao = await prisma.atribuicao.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends AtribuicaoCreateManyArgs>(args?: Prisma.SelectSubset<T, AtribuicaoCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Atribuicaos and returns the data saved in the database.
   * @param {AtribuicaoCreateManyAndReturnArgs} args - Arguments to create many Atribuicaos.
   * @example
   * // Create many Atribuicaos
   * const atribuicao = await prisma.atribuicao.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Atribuicaos and only return the `id`
   * const atribuicaoWithIdOnly = await prisma.atribuicao.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends AtribuicaoCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, AtribuicaoCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Atribuicao.
   * @param {AtribuicaoDeleteArgs} args - Arguments to delete one Atribuicao.
   * @example
   * // Delete one Atribuicao
   * const Atribuicao = await prisma.atribuicao.delete({
   *   where: {
   *     // ... filter to delete one Atribuicao
   *   }
   * })
   * 
   */
  delete<T extends AtribuicaoDeleteArgs>(args: Prisma.SelectSubset<T, AtribuicaoDeleteArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Atribuicao.
   * @param {AtribuicaoUpdateArgs} args - Arguments to update one Atribuicao.
   * @example
   * // Update one Atribuicao
   * const atribuicao = await prisma.atribuicao.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends AtribuicaoUpdateArgs>(args: Prisma.SelectSubset<T, AtribuicaoUpdateArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Atribuicaos.
   * @param {AtribuicaoDeleteManyArgs} args - Arguments to filter Atribuicaos to delete.
   * @example
   * // Delete a few Atribuicaos
   * const { count } = await prisma.atribuicao.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends AtribuicaoDeleteManyArgs>(args?: Prisma.SelectSubset<T, AtribuicaoDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Atribuicaos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Atribuicaos
   * const atribuicao = await prisma.atribuicao.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends AtribuicaoUpdateManyArgs>(args: Prisma.SelectSubset<T, AtribuicaoUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Atribuicaos and returns the data updated in the database.
   * @param {AtribuicaoUpdateManyAndReturnArgs} args - Arguments to update many Atribuicaos.
   * @example
   * // Update many Atribuicaos
   * const atribuicao = await prisma.atribuicao.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Atribuicaos and only return the `id`
   * const atribuicaoWithIdOnly = await prisma.atribuicao.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends AtribuicaoUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, AtribuicaoUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Atribuicao.
   * @param {AtribuicaoUpsertArgs} args - Arguments to update or create a Atribuicao.
   * @example
   * // Update or create a Atribuicao
   * const atribuicao = await prisma.atribuicao.upsert({
   *   create: {
   *     // ... data to create a Atribuicao
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Atribuicao we want to update
   *   }
   * })
   */
  upsert<T extends AtribuicaoUpsertArgs>(args: Prisma.SelectSubset<T, AtribuicaoUpsertArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Atribuicaos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoCountArgs} args - Arguments to filter Atribuicaos to count.
   * @example
   * // Count the number of Atribuicaos
   * const count = await prisma.atribuicao.count({
   *   where: {
   *     // ... the filter for the Atribuicaos we want to count
   *   }
   * })
  **/
  count<T extends AtribuicaoCountArgs>(
    args?: Prisma.Subset<T, AtribuicaoCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], AtribuicaoCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Atribuicao.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends AtribuicaoAggregateArgs>(args: Prisma.Subset<T, AtribuicaoAggregateArgs>): Prisma.PrismaPromise<GetAtribuicaoAggregateType<T>>

  /**
   * Group by Atribuicao.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {AtribuicaoGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends AtribuicaoGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: AtribuicaoGroupByArgs['orderBy'] }
      : { orderBy?: AtribuicaoGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, AtribuicaoGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetAtribuicaoGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Atribuicao model
 */
readonly fields: AtribuicaoFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Atribuicao.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__AtribuicaoClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  artigo<T extends Prisma.ArtigoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.ArtigoDefaultArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  revisor<T extends Prisma.RevisorDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.RevisorDefaultArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  respostas<T extends Prisma.Atribuicao$respostasArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.Atribuicao$respostasArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Atribuicao model
 */
export interface AtribuicaoFieldRefs {
  readonly id: Prisma.FieldRef<"Atribuicao", 'Int'>
  readonly artigoId: Prisma.FieldRef<"Atribuicao", 'Int'>
  readonly revisorId: Prisma.FieldRef<"Atribuicao", 'Int'>
}
    

// Custom InputTypes
/**
 * Atribuicao findUnique
 */
export type AtribuicaoFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter, which Atribuicao to fetch.
   */
  where: Prisma.AtribuicaoWhereUniqueInput
}

/**
 * Atribuicao findUniqueOrThrow
 */
export type AtribuicaoFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter, which Atribuicao to fetch.
   */
  where: Prisma.AtribuicaoWhereUniqueInput
}

/**
 * Atribuicao findFirst
 */
export type AtribuicaoFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter, which Atribuicao to fetch.
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Atribuicaos to fetch.
   */
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Atribuicaos.
   */
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Atribuicaos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Atribuicaos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Atribuicaos.
   */
  distinct?: Prisma.AtribuicaoScalarFieldEnum | Prisma.AtribuicaoScalarFieldEnum[]
}

/**
 * Atribuicao findFirstOrThrow
 */
export type AtribuicaoFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter, which Atribuicao to fetch.
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Atribuicaos to fetch.
   */
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Atribuicaos.
   */
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Atribuicaos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Atribuicaos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Atribuicaos.
   */
  distinct?: Prisma.AtribuicaoScalarFieldEnum | Prisma.AtribuicaoScalarFieldEnum[]
}

/**
 * Atribuicao findMany
 */
export type AtribuicaoFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter, which Atribuicaos to fetch.
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Atribuicaos to fetch.
   */
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Atribuicaos.
   */
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Atribuicaos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Atribuicaos.
   */
  skip?: number
  distinct?: Prisma.AtribuicaoScalarFieldEnum | Prisma.AtribuicaoScalarFieldEnum[]
}

/**
 * Atribuicao create
 */
export type AtribuicaoCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * The data needed to create a Atribuicao.
   */
  data: Prisma.XOR<Prisma.AtribuicaoCreateInput, Prisma.AtribuicaoUncheckedCreateInput>
}

/**
 * Atribuicao createMany
 */
export type AtribuicaoCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Atribuicaos.
   */
  data: Prisma.AtribuicaoCreateManyInput | Prisma.AtribuicaoCreateManyInput[]
}

/**
 * Atribuicao createManyAndReturn
 */
export type AtribuicaoCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * The data used to create many Atribuicaos.
   */
  data: Prisma.AtribuicaoCreateManyInput | Prisma.AtribuicaoCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Atribuicao update
 */
export type AtribuicaoUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * The data needed to update a Atribuicao.
   */
  data: Prisma.XOR<Prisma.AtribuicaoUpdateInput, Prisma.AtribuicaoUncheckedUpdateInput>
  /**
   * Choose, which Atribuicao to update.
   */
  where: Prisma.AtribuicaoWhereUniqueInput
}

/**
 * Atribuicao updateMany
 */
export type AtribuicaoUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Atribuicaos.
   */
  data: Prisma.XOR<Prisma.AtribuicaoUpdateManyMutationInput, Prisma.AtribuicaoUncheckedUpdateManyInput>
  /**
   * Filter which Atribuicaos to update
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * Limit how many Atribuicaos to update.
   */
  limit?: number
}

/**
 * Atribuicao updateManyAndReturn
 */
export type AtribuicaoUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * The data used to update Atribuicaos.
   */
  data: Prisma.XOR<Prisma.AtribuicaoUpdateManyMutationInput, Prisma.AtribuicaoUncheckedUpdateManyInput>
  /**
   * Filter which Atribuicaos to update
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * Limit how many Atribuicaos to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Atribuicao upsert
 */
export type AtribuicaoUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * The filter to search for the Atribuicao to update in case it exists.
   */
  where: Prisma.AtribuicaoWhereUniqueInput
  /**
   * In case the Atribuicao found by the `where` argument doesn't exist, create a new Atribuicao with this data.
   */
  create: Prisma.XOR<Prisma.AtribuicaoCreateInput, Prisma.AtribuicaoUncheckedCreateInput>
  /**
   * In case the Atribuicao was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.AtribuicaoUpdateInput, Prisma.AtribuicaoUncheckedUpdateInput>
}

/**
 * Atribuicao delete
 */
export type AtribuicaoDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  /**
   * Filter which Atribuicao to delete.
   */
  where: Prisma.AtribuicaoWhereUniqueInput
}

/**
 * Atribuicao deleteMany
 */
export type AtribuicaoDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Atribuicaos to delete
   */
  where?: Prisma.AtribuicaoWhereInput
  /**
   * Limit how many Atribuicaos to delete.
   */
  limit?: number
}

/**
 * Atribuicao.respostas
 */
export type Atribuicao$respostasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  where?: Prisma.RespostaWhereInput
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  cursor?: Prisma.RespostaWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.RespostaScalarFieldEnum | Prisma.RespostaScalarFieldEnum[]
}

/**
 * Atribuicao without action
 */
export type AtribuicaoDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
}
