// @prisma/client/models/EdicaoEvento.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `EdicaoEvento` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model EdicaoEvento
 * 
 */
export type EdicaoEventoModel = runtime.Types.Result.DefaultSelection<Prisma.$EdicaoEventoPayload>

export type AggregateEdicaoEvento = {
  _count: EdicaoEventoCountAggregateOutputType | null
  _avg: EdicaoEventoAvgAggregateOutputType | null
  _sum: EdicaoEventoSumAggregateOutputType | null
  _min: EdicaoEventoMinAggregateOutputType | null
  _max: EdicaoEventoMaxAggregateOutputType | null
}

export type EdicaoEventoAvgAggregateOutputType = {
  id: number | null
  ano: number | null
  eventoId: number | null
}

export type EdicaoEventoSumAggregateOutputType = {
  id: number | null
  ano: number | null
  eventoId: number | null
}

export type EdicaoEventoMinAggregateOutputType = {
  id: number | null
  ano: number | null
  eventoId: number | null
}

export type EdicaoEventoMaxAggregateOutputType = {
  id: number | null
  ano: number | null
  eventoId: number | null
}

export type EdicaoEventoCountAggregateOutputType = {
  id: number
  ano: number
  eventoId: number
  _all: number
}


export type EdicaoEventoAvgAggregateInputType = {
  id?: true
  ano?: true
  eventoId?: true
}

export type EdicaoEventoSumAggregateInputType = {
  id?: true
  ano?: true
  eventoId?: true
}

export type EdicaoEventoMinAggregateInputType = {
  id?: true
  ano?: true
  eventoId?: true
}

export type EdicaoEventoMaxAggregateInputType = {
  id?: true
  ano?: true
  eventoId?: true
}

export type EdicaoEventoCountAggregateInputType = {
  id?: true
  ano?: true
  eventoId?: true
  _all?: true
}

export type EdicaoEventoAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which EdicaoEvento to aggregate.
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of EdicaoEventos to fetch.
   */
  orderBy?: Prisma.EdicaoEventoOrderByWithRelationInput | Prisma.EdicaoEventoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.EdicaoEventoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` EdicaoEventos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` EdicaoEventos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned EdicaoEventos
  **/
  _count?: true | EdicaoEventoCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: EdicaoEventoAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: EdicaoEventoSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: EdicaoEventoMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: EdicaoEventoMaxAggregateInputType
}

export type GetEdicaoEventoAggregateType<T extends EdicaoEventoAggregateArgs> = {
      [P in keyof T & keyof AggregateEdicaoEvento]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateEdicaoEvento[P]>
    : Prisma.GetScalarType<T[P], AggregateEdicaoEvento[P]>
}




export type EdicaoEventoGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.EdicaoEventoWhereInput
  orderBy?: Prisma.EdicaoEventoOrderByWithAggregationInput | Prisma.EdicaoEventoOrderByWithAggregationInput[]
  by: Prisma.EdicaoEventoScalarFieldEnum[] | Prisma.EdicaoEventoScalarFieldEnum
  having?: Prisma.EdicaoEventoScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: EdicaoEventoCountAggregateInputType | true
  _avg?: EdicaoEventoAvgAggregateInputType
  _sum?: EdicaoEventoSumAggregateInputType
  _min?: EdicaoEventoMinAggregateInputType
  _max?: EdicaoEventoMaxAggregateInputType
}

export type EdicaoEventoGroupByOutputType = {
  id: number
  ano: number
  eventoId: number
  _count: EdicaoEventoCountAggregateOutputType | null
  _avg: EdicaoEventoAvgAggregateOutputType | null
  _sum: EdicaoEventoSumAggregateOutputType | null
  _min: EdicaoEventoMinAggregateOutputType | null
  _max: EdicaoEventoMaxAggregateOutputType | null
}

type GetEdicaoEventoGroupByPayload<T extends EdicaoEventoGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<EdicaoEventoGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof EdicaoEventoGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], EdicaoEventoGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], EdicaoEventoGroupByOutputType[P]>
      }
    >
  >



export type EdicaoEventoWhereInput = {
  AND?: Prisma.EdicaoEventoWhereInput | Prisma.EdicaoEventoWhereInput[]
  OR?: Prisma.EdicaoEventoWhereInput[]
  NOT?: Prisma.EdicaoEventoWhereInput | Prisma.EdicaoEventoWhereInput[]
  id?: Prisma.IntFilter<"EdicaoEvento"> | number
  ano?: Prisma.IntFilter<"EdicaoEvento"> | number
  eventoId?: Prisma.IntFilter<"EdicaoEvento"> | number
  evento?: Prisma.XOR<Prisma.EventoScalarRelationFilter, Prisma.EventoWhereInput>
  artigos?: Prisma.ArtigoListRelationFilter
  revisores?: Prisma.RevisorListRelationFilter
}

export type EdicaoEventoOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
  evento?: Prisma.EventoOrderByWithRelationInput
  artigos?: Prisma.ArtigoOrderByRelationAggregateInput
  revisores?: Prisma.RevisorOrderByRelationAggregateInput
}

export type EdicaoEventoWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  eventoId_ano?: Prisma.EdicaoEventoEventoIdAnoCompoundUniqueInput
  AND?: Prisma.EdicaoEventoWhereInput | Prisma.EdicaoEventoWhereInput[]
  OR?: Prisma.EdicaoEventoWhereInput[]
  NOT?: Prisma.EdicaoEventoWhereInput | Prisma.EdicaoEventoWhereInput[]
  ano?: Prisma.IntFilter<"EdicaoEvento"> | number
  eventoId?: Prisma.IntFilter<"EdicaoEvento"> | number
  evento?: Prisma.XOR<Prisma.EventoScalarRelationFilter, Prisma.EventoWhereInput>
  artigos?: Prisma.ArtigoListRelationFilter
  revisores?: Prisma.RevisorListRelationFilter
}, "id" | "eventoId_ano">

export type EdicaoEventoOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
  _count?: Prisma.EdicaoEventoCountOrderByAggregateInput
  _avg?: Prisma.EdicaoEventoAvgOrderByAggregateInput
  _max?: Prisma.EdicaoEventoMaxOrderByAggregateInput
  _min?: Prisma.EdicaoEventoMinOrderByAggregateInput
  _sum?: Prisma.EdicaoEventoSumOrderByAggregateInput
}

export type EdicaoEventoScalarWhereWithAggregatesInput = {
  AND?: Prisma.EdicaoEventoScalarWhereWithAggregatesInput | Prisma.EdicaoEventoScalarWhereWithAggregatesInput[]
  OR?: Prisma.EdicaoEventoScalarWhereWithAggregatesInput[]
  NOT?: Prisma.EdicaoEventoScalarWhereWithAggregatesInput | Prisma.EdicaoEventoScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"EdicaoEvento"> | number
  ano?: Prisma.IntWithAggregatesFilter<"EdicaoEvento"> | number
  eventoId?: Prisma.IntWithAggregatesFilter<"EdicaoEvento"> | number
}

export type EdicaoEventoCreateInput = {
  ano: number
  evento: Prisma.EventoCreateNestedOneWithoutEdicoesInput
  artigos?: Prisma.ArtigoCreateNestedManyWithoutEdicaoInput
  revisores?: Prisma.RevisorCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoUncheckedCreateInput = {
  id?: number
  ano: number
  eventoId: number
  artigos?: Prisma.ArtigoUncheckedCreateNestedManyWithoutEdicaoInput
  revisores?: Prisma.RevisorUncheckedCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoUpdateInput = {
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  evento?: Prisma.EventoUpdateOneRequiredWithoutEdicoesNestedInput
  artigos?: Prisma.ArtigoUpdateManyWithoutEdicaoNestedInput
  revisores?: Prisma.RevisorUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  eventoId?: Prisma.IntFieldUpdateOperationsInput | number
  artigos?: Prisma.ArtigoUncheckedUpdateManyWithoutEdicaoNestedInput
  revisores?: Prisma.RevisorUncheckedUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoCreateManyInput = {
  id?: number
  ano: number
  eventoId: number
}

export type EdicaoEventoUpdateManyMutationInput = {
  ano?: Prisma.IntFieldUpdateOperationsInput | number
}

export type EdicaoEventoUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  eventoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type EdicaoEventoListRelationFilter = {
  every?: Prisma.EdicaoEventoWhereInput
  some?: Prisma.EdicaoEventoWhereInput
  none?: Prisma.EdicaoEventoWhereInput
}

export type EdicaoEventoOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type EdicaoEventoEventoIdAnoCompoundUniqueInput = {
  eventoId: number
  ano: number
}

export type EdicaoEventoCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
}

export type EdicaoEventoAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
}

export type EdicaoEventoMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
}

export type EdicaoEventoMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
}

export type EdicaoEventoSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  ano?: Prisma.SortOrder
  eventoId?: Prisma.SortOrder
}

export type EdicaoEventoScalarRelationFilter = {
  is?: Prisma.EdicaoEventoWhereInput
  isNot?: Prisma.EdicaoEventoWhereInput
}

export type EdicaoEventoCreateNestedManyWithoutEventoInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput> | Prisma.EdicaoEventoCreateWithoutEventoInput[] | Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput[]
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput | Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput[]
  createMany?: Prisma.EdicaoEventoCreateManyEventoInputEnvelope
  connect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
}

export type EdicaoEventoUncheckedCreateNestedManyWithoutEventoInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput> | Prisma.EdicaoEventoCreateWithoutEventoInput[] | Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput[]
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput | Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput[]
  createMany?: Prisma.EdicaoEventoCreateManyEventoInputEnvelope
  connect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
}

export type EdicaoEventoUpdateManyWithoutEventoNestedInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput> | Prisma.EdicaoEventoCreateWithoutEventoInput[] | Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput[]
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput | Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput[]
  upsert?: Prisma.EdicaoEventoUpsertWithWhereUniqueWithoutEventoInput | Prisma.EdicaoEventoUpsertWithWhereUniqueWithoutEventoInput[]
  createMany?: Prisma.EdicaoEventoCreateManyEventoInputEnvelope
  set?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  disconnect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  delete?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  connect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  update?: Prisma.EdicaoEventoUpdateWithWhereUniqueWithoutEventoInput | Prisma.EdicaoEventoUpdateWithWhereUniqueWithoutEventoInput[]
  updateMany?: Prisma.EdicaoEventoUpdateManyWithWhereWithoutEventoInput | Prisma.EdicaoEventoUpdateManyWithWhereWithoutEventoInput[]
  deleteMany?: Prisma.EdicaoEventoScalarWhereInput | Prisma.EdicaoEventoScalarWhereInput[]
}

export type EdicaoEventoUncheckedUpdateManyWithoutEventoNestedInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput> | Prisma.EdicaoEventoCreateWithoutEventoInput[] | Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput[]
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput | Prisma.EdicaoEventoCreateOrConnectWithoutEventoInput[]
  upsert?: Prisma.EdicaoEventoUpsertWithWhereUniqueWithoutEventoInput | Prisma.EdicaoEventoUpsertWithWhereUniqueWithoutEventoInput[]
  createMany?: Prisma.EdicaoEventoCreateManyEventoInputEnvelope
  set?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  disconnect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  delete?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  connect?: Prisma.EdicaoEventoWhereUniqueInput | Prisma.EdicaoEventoWhereUniqueInput[]
  update?: Prisma.EdicaoEventoUpdateWithWhereUniqueWithoutEventoInput | Prisma.EdicaoEventoUpdateWithWhereUniqueWithoutEventoInput[]
  updateMany?: Prisma.EdicaoEventoUpdateManyWithWhereWithoutEventoInput | Prisma.EdicaoEventoUpdateManyWithWhereWithoutEventoInput[]
  deleteMany?: Prisma.EdicaoEventoScalarWhereInput | Prisma.EdicaoEventoScalarWhereInput[]
}

export type EdicaoEventoCreateNestedOneWithoutArtigosInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedCreateWithoutArtigosInput>
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutArtigosInput
  connect?: Prisma.EdicaoEventoWhereUniqueInput
}

export type EdicaoEventoUpdateOneRequiredWithoutArtigosNestedInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedCreateWithoutArtigosInput>
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutArtigosInput
  upsert?: Prisma.EdicaoEventoUpsertWithoutArtigosInput
  connect?: Prisma.EdicaoEventoWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.EdicaoEventoUpdateToOneWithWhereWithoutArtigosInput, Prisma.EdicaoEventoUpdateWithoutArtigosInput>, Prisma.EdicaoEventoUncheckedUpdateWithoutArtigosInput>
}

export type EdicaoEventoCreateNestedOneWithoutRevisoresInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedCreateWithoutRevisoresInput>
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutRevisoresInput
  connect?: Prisma.EdicaoEventoWhereUniqueInput
}

export type EdicaoEventoUpdateOneRequiredWithoutRevisoresNestedInput = {
  create?: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedCreateWithoutRevisoresInput>
  connectOrCreate?: Prisma.EdicaoEventoCreateOrConnectWithoutRevisoresInput
  upsert?: Prisma.EdicaoEventoUpsertWithoutRevisoresInput
  connect?: Prisma.EdicaoEventoWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.EdicaoEventoUpdateToOneWithWhereWithoutRevisoresInput, Prisma.EdicaoEventoUpdateWithoutRevisoresInput>, Prisma.EdicaoEventoUncheckedUpdateWithoutRevisoresInput>
}

export type EdicaoEventoCreateWithoutEventoInput = {
  ano: number
  artigos?: Prisma.ArtigoCreateNestedManyWithoutEdicaoInput
  revisores?: Prisma.RevisorCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoUncheckedCreateWithoutEventoInput = {
  id?: number
  ano: number
  artigos?: Prisma.ArtigoUncheckedCreateNestedManyWithoutEdicaoInput
  revisores?: Prisma.RevisorUncheckedCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoCreateOrConnectWithoutEventoInput = {
  where: Prisma.EdicaoEventoWhereUniqueInput
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput>
}

export type EdicaoEventoCreateManyEventoInputEnvelope = {
  data: Prisma.EdicaoEventoCreateManyEventoInput | Prisma.EdicaoEventoCreateManyEventoInput[]
}

export type EdicaoEventoUpsertWithWhereUniqueWithoutEventoInput = {
  where: Prisma.EdicaoEventoWhereUniqueInput
  update: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutEventoInput, Prisma.EdicaoEventoUncheckedUpdateWithoutEventoInput>
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutEventoInput, Prisma.EdicaoEventoUncheckedCreateWithoutEventoInput>
}

export type EdicaoEventoUpdateWithWhereUniqueWithoutEventoInput = {
  where: Prisma.EdicaoEventoWhereUniqueInput
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutEventoInput, Prisma.EdicaoEventoUncheckedUpdateWithoutEventoInput>
}

export type EdicaoEventoUpdateManyWithWhereWithoutEventoInput = {
  where: Prisma.EdicaoEventoScalarWhereInput
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateManyMutationInput, Prisma.EdicaoEventoUncheckedUpdateManyWithoutEventoInput>
}

export type EdicaoEventoScalarWhereInput = {
  AND?: Prisma.EdicaoEventoScalarWhereInput | Prisma.EdicaoEventoScalarWhereInput[]
  OR?: Prisma.EdicaoEventoScalarWhereInput[]
  NOT?: Prisma.EdicaoEventoScalarWhereInput | Prisma.EdicaoEventoScalarWhereInput[]
  id?: Prisma.IntFilter<"EdicaoEvento"> | number
  ano?: Prisma.IntFilter<"EdicaoEvento"> | number
  eventoId?: Prisma.IntFilter<"EdicaoEvento"> | number
}

export type EdicaoEventoCreateWithoutArtigosInput = {
  ano: number
  evento: Prisma.EventoCreateNestedOneWithoutEdicoesInput
  revisores?: Prisma.RevisorCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoUncheckedCreateWithoutArtigosInput = {
  id?: number
  ano: number
  eventoId: number
  revisores?: Prisma.RevisorUncheckedCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoCreateOrConnectWithoutArtigosInput = {
  where: Prisma.EdicaoEventoWhereUniqueInput
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedCreateWithoutArtigosInput>
}

export type EdicaoEventoUpsertWithoutArtigosInput = {
  update: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedUpdateWithoutArtigosInput>
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedCreateWithoutArtigosInput>
  where?: Prisma.EdicaoEventoWhereInput
}

export type EdicaoEventoUpdateToOneWithWhereWithoutArtigosInput = {
  where?: Prisma.EdicaoEventoWhereInput
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutArtigosInput, Prisma.EdicaoEventoUncheckedUpdateWithoutArtigosInput>
}

export type EdicaoEventoUpdateWithoutArtigosInput = {
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  evento?: Prisma.EventoUpdateOneRequiredWithoutEdicoesNestedInput
  revisores?: Prisma.RevisorUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoUncheckedUpdateWithoutArtigosInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  eventoId?: Prisma.IntFieldUpdateOperationsInput | number
  revisores?: Prisma.RevisorUncheckedUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoCreateWithoutRevisoresInput = {
  ano: number
  evento: Prisma.EventoCreateNestedOneWithoutEdicoesInput
  artigos?: Prisma.ArtigoCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoUncheckedCreateWithoutRevisoresInput = {
  id?: number
  ano: number
  eventoId: number
  artigos?: Prisma.ArtigoUncheckedCreateNestedManyWithoutEdicaoInput
}

export type EdicaoEventoCreateOrConnectWithoutRevisoresInput = {
  where: Prisma.EdicaoEventoWhereUniqueInput
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedCreateWithoutRevisoresInput>
}

export type EdicaoEventoUpsertWithoutRevisoresInput = {
  update: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedUpdateWithoutRevisoresInput>
  create: Prisma.XOR<Prisma.EdicaoEventoCreateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedCreateWithoutRevisoresInput>
  where?: Prisma.EdicaoEventoWhereInput
}

export type EdicaoEventoUpdateToOneWithWhereWithoutRevisoresInput = {
  where?: Prisma.EdicaoEventoWhereInput
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateWithoutRevisoresInput, Prisma.EdicaoEventoUncheckedUpdateWithoutRevisoresInput>
}

export type EdicaoEventoUpdateWithoutRevisoresInput = {
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  evento?: Prisma.EventoUpdateOneRequiredWithoutEdicoesNestedInput
  artigos?: Prisma.ArtigoUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoUncheckedUpdateWithoutRevisoresInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  eventoId?: Prisma.IntFieldUpdateOperationsInput | number
  artigos?: Prisma.ArtigoUncheckedUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoCreateManyEventoInput = {
  id?: number
  ano: number
}

export type EdicaoEventoUpdateWithoutEventoInput = {
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  artigos?: Prisma.ArtigoUpdateManyWithoutEdicaoNestedInput
  revisores?: Prisma.RevisorUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoUncheckedUpdateWithoutEventoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
  artigos?: Prisma.ArtigoUncheckedUpdateManyWithoutEdicaoNestedInput
  revisores?: Prisma.RevisorUncheckedUpdateManyWithoutEdicaoNestedInput
}

export type EdicaoEventoUncheckedUpdateManyWithoutEventoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  ano?: Prisma.IntFieldUpdateOperationsInput | number
}


/**
 * Count Type EdicaoEventoCountOutputType
 */

export type EdicaoEventoCountOutputType = {
  artigos: number
  revisores: number
}

export type EdicaoEventoCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigos?: boolean | EdicaoEventoCountOutputTypeCountArtigosArgs
  revisores?: boolean | EdicaoEventoCountOutputTypeCountRevisoresArgs
}

/**
 * EdicaoEventoCountOutputType without action
 */
export type EdicaoEventoCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEventoCountOutputType
   */
  select?: Prisma.EdicaoEventoCountOutputTypeSelect<ExtArgs> | null
}

/**
 * EdicaoEventoCountOutputType without action
 */
export type EdicaoEventoCountOutputTypeCountArtigosArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.ArtigoWhereInput
}

/**
 * EdicaoEventoCountOutputType without action
 */
export type EdicaoEventoCountOutputTypeCountRevisoresArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.RevisorWhereInput
}


export type EdicaoEventoSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  ano?: boolean
  eventoId?: boolean
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
  artigos?: boolean | Prisma.EdicaoEvento$artigosArgs<ExtArgs>
  revisores?: boolean | Prisma.EdicaoEvento$revisoresArgs<ExtArgs>
  _count?: boolean | Prisma.EdicaoEventoCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["edicaoEvento"]>

export type EdicaoEventoSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  ano?: boolean
  eventoId?: boolean
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["edicaoEvento"]>

export type EdicaoEventoSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  ano?: boolean
  eventoId?: boolean
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["edicaoEvento"]>

export type EdicaoEventoSelectScalar = {
  id?: boolean
  ano?: boolean
  eventoId?: boolean
}

export type EdicaoEventoOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "ano" | "eventoId", ExtArgs["result"]["edicaoEvento"]>
export type EdicaoEventoInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
  artigos?: boolean | Prisma.EdicaoEvento$artigosArgs<ExtArgs>
  revisores?: boolean | Prisma.EdicaoEvento$revisoresArgs<ExtArgs>
  _count?: boolean | Prisma.EdicaoEventoCountOutputTypeDefaultArgs<ExtArgs>
}
export type EdicaoEventoIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
}
export type EdicaoEventoIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  evento?: boolean | Prisma.EventoDefaultArgs<ExtArgs>
}

export type $EdicaoEventoPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "EdicaoEvento"
  objects: {
    evento: Prisma.$EventoPayload<ExtArgs>
    artigos: Prisma.$ArtigoPayload<ExtArgs>[]
    revisores: Prisma.$RevisorPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    ano: number
    eventoId: number
  }, ExtArgs["result"]["edicaoEvento"]>
  composites: {}
}

export type EdicaoEventoGetPayload<S extends boolean | null | undefined | EdicaoEventoDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload, S>

export type EdicaoEventoCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<EdicaoEventoFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: EdicaoEventoCountAggregateInputType | true
  }

export interface EdicaoEventoDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['EdicaoEvento'], meta: { name: 'EdicaoEvento' } }
  /**
   * Find zero or one EdicaoEvento that matches the filter.
   * @param {EdicaoEventoFindUniqueArgs} args - Arguments to find a EdicaoEvento
   * @example
   * // Get one EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends EdicaoEventoFindUniqueArgs>(args: Prisma.SelectSubset<T, EdicaoEventoFindUniqueArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one EdicaoEvento that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {EdicaoEventoFindUniqueOrThrowArgs} args - Arguments to find a EdicaoEvento
   * @example
   * // Get one EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends EdicaoEventoFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, EdicaoEventoFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first EdicaoEvento that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoFindFirstArgs} args - Arguments to find a EdicaoEvento
   * @example
   * // Get one EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends EdicaoEventoFindFirstArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoFindFirstArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first EdicaoEvento that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoFindFirstOrThrowArgs} args - Arguments to find a EdicaoEvento
   * @example
   * // Get one EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends EdicaoEventoFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more EdicaoEventos that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all EdicaoEventos
   * const edicaoEventos = await prisma.edicaoEvento.findMany()
   * 
   * // Get first 10 EdicaoEventos
   * const edicaoEventos = await prisma.edicaoEvento.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const edicaoEventoWithIdOnly = await prisma.edicaoEvento.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends EdicaoEventoFindManyArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a EdicaoEvento.
   * @param {EdicaoEventoCreateArgs} args - Arguments to create a EdicaoEvento.
   * @example
   * // Create one EdicaoEvento
   * const EdicaoEvento = await prisma.edicaoEvento.create({
   *   data: {
   *     // ... data to create a EdicaoEvento
   *   }
   * })
   * 
   */
  create<T extends EdicaoEventoCreateArgs>(args: Prisma.SelectSubset<T, EdicaoEventoCreateArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many EdicaoEventos.
   * @param {EdicaoEventoCreateManyArgs} args - Arguments to create many EdicaoEventos.
   * @example
   * // Create many EdicaoEventos
   * const edicaoEvento = await prisma.edicaoEvento.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends EdicaoEventoCreateManyArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many EdicaoEventos and returns the data saved in the database.
   * @param {EdicaoEventoCreateManyAndReturnArgs} args - Arguments to create many EdicaoEventos.
   * @example
   * // Create many EdicaoEventos
   * const edicaoEvento = await prisma.edicaoEvento.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many EdicaoEventos and only return the `id`
   * const edicaoEventoWithIdOnly = await prisma.edicaoEvento.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends EdicaoEventoCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a EdicaoEvento.
   * @param {EdicaoEventoDeleteArgs} args - Arguments to delete one EdicaoEvento.
   * @example
   * // Delete one EdicaoEvento
   * const EdicaoEvento = await prisma.edicaoEvento.delete({
   *   where: {
   *     // ... filter to delete one EdicaoEvento
   *   }
   * })
   * 
   */
  delete<T extends EdicaoEventoDeleteArgs>(args: Prisma.SelectSubset<T, EdicaoEventoDeleteArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one EdicaoEvento.
   * @param {EdicaoEventoUpdateArgs} args - Arguments to update one EdicaoEvento.
   * @example
   * // Update one EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends EdicaoEventoUpdateArgs>(args: Prisma.SelectSubset<T, EdicaoEventoUpdateArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more EdicaoEventos.
   * @param {EdicaoEventoDeleteManyArgs} args - Arguments to filter EdicaoEventos to delete.
   * @example
   * // Delete a few EdicaoEventos
   * const { count } = await prisma.edicaoEvento.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends EdicaoEventoDeleteManyArgs>(args?: Prisma.SelectSubset<T, EdicaoEventoDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more EdicaoEventos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many EdicaoEventos
   * const edicaoEvento = await prisma.edicaoEvento.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends EdicaoEventoUpdateManyArgs>(args: Prisma.SelectSubset<T, EdicaoEventoUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more EdicaoEventos and returns the data updated in the database.
   * @param {EdicaoEventoUpdateManyAndReturnArgs} args - Arguments to update many EdicaoEventos.
   * @example
   * // Update many EdicaoEventos
   * const edicaoEvento = await prisma.edicaoEvento.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more EdicaoEventos and only return the `id`
   * const edicaoEventoWithIdOnly = await prisma.edicaoEvento.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends EdicaoEventoUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, EdicaoEventoUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one EdicaoEvento.
   * @param {EdicaoEventoUpsertArgs} args - Arguments to update or create a EdicaoEvento.
   * @example
   * // Update or create a EdicaoEvento
   * const edicaoEvento = await prisma.edicaoEvento.upsert({
   *   create: {
   *     // ... data to create a EdicaoEvento
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the EdicaoEvento we want to update
   *   }
   * })
   */
  upsert<T extends EdicaoEventoUpsertArgs>(args: Prisma.SelectSubset<T, EdicaoEventoUpsertArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of EdicaoEventos.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoCountArgs} args - Arguments to filter EdicaoEventos to count.
   * @example
   * // Count the number of EdicaoEventos
   * const count = await prisma.edicaoEvento.count({
   *   where: {
   *     // ... the filter for the EdicaoEventos we want to count
   *   }
   * })
  **/
  count<T extends EdicaoEventoCountArgs>(
    args?: Prisma.Subset<T, EdicaoEventoCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], EdicaoEventoCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a EdicaoEvento.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends EdicaoEventoAggregateArgs>(args: Prisma.Subset<T, EdicaoEventoAggregateArgs>): Prisma.PrismaPromise<GetEdicaoEventoAggregateType<T>>

  /**
   * Group by EdicaoEvento.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {EdicaoEventoGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends EdicaoEventoGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: EdicaoEventoGroupByArgs['orderBy'] }
      : { orderBy?: EdicaoEventoGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, EdicaoEventoGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetEdicaoEventoGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the EdicaoEvento model
 */
readonly fields: EdicaoEventoFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for EdicaoEvento.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__EdicaoEventoClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  evento<T extends Prisma.EventoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.EventoDefaultArgs<ExtArgs>>): Prisma.Prisma__EventoClient<runtime.Types.Result.GetResult<Prisma.$EventoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  artigos<T extends Prisma.EdicaoEvento$artigosArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.EdicaoEvento$artigosArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  revisores<T extends Prisma.EdicaoEvento$revisoresArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.EdicaoEvento$revisoresArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the EdicaoEvento model
 */
export interface EdicaoEventoFieldRefs {
  readonly id: Prisma.FieldRef<"EdicaoEvento", 'Int'>
  readonly ano: Prisma.FieldRef<"EdicaoEvento", 'Int'>
  readonly eventoId: Prisma.FieldRef<"EdicaoEvento", 'Int'>
}
    

// Custom InputTypes
/**
 * EdicaoEvento findUnique
 */
export type EdicaoEventoFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter, which EdicaoEvento to fetch.
   */
  where: Prisma.EdicaoEventoWhereUniqueInput
}

/**
 * EdicaoEvento findUniqueOrThrow
 */
export type EdicaoEventoFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter, which EdicaoEvento to fetch.
   */
  where: Prisma.EdicaoEventoWhereUniqueInput
}

/**
 * EdicaoEvento findFirst
 */
export type EdicaoEventoFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter, which EdicaoEvento to fetch.
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of EdicaoEventos to fetch.
   */
  orderBy?: Prisma.EdicaoEventoOrderByWithRelationInput | Prisma.EdicaoEventoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for EdicaoEventos.
   */
  cursor?: Prisma.EdicaoEventoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` EdicaoEventos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` EdicaoEventos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of EdicaoEventos.
   */
  distinct?: Prisma.EdicaoEventoScalarFieldEnum | Prisma.EdicaoEventoScalarFieldEnum[]
}

/**
 * EdicaoEvento findFirstOrThrow
 */
export type EdicaoEventoFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter, which EdicaoEvento to fetch.
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of EdicaoEventos to fetch.
   */
  orderBy?: Prisma.EdicaoEventoOrderByWithRelationInput | Prisma.EdicaoEventoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for EdicaoEventos.
   */
  cursor?: Prisma.EdicaoEventoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` EdicaoEventos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` EdicaoEventos.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of EdicaoEventos.
   */
  distinct?: Prisma.EdicaoEventoScalarFieldEnum | Prisma.EdicaoEventoScalarFieldEnum[]
}

/**
 * EdicaoEvento findMany
 */
export type EdicaoEventoFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter, which EdicaoEventos to fetch.
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of EdicaoEventos to fetch.
   */
  orderBy?: Prisma.EdicaoEventoOrderByWithRelationInput | Prisma.EdicaoEventoOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing EdicaoEventos.
   */
  cursor?: Prisma.EdicaoEventoWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` EdicaoEventos from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` EdicaoEventos.
   */
  skip?: number
  distinct?: Prisma.EdicaoEventoScalarFieldEnum | Prisma.EdicaoEventoScalarFieldEnum[]
}

/**
 * EdicaoEvento create
 */
export type EdicaoEventoCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * The data needed to create a EdicaoEvento.
   */
  data: Prisma.XOR<Prisma.EdicaoEventoCreateInput, Prisma.EdicaoEventoUncheckedCreateInput>
}

/**
 * EdicaoEvento createMany
 */
export type EdicaoEventoCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many EdicaoEventos.
   */
  data: Prisma.EdicaoEventoCreateManyInput | Prisma.EdicaoEventoCreateManyInput[]
}

/**
 * EdicaoEvento createManyAndReturn
 */
export type EdicaoEventoCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * The data used to create many EdicaoEventos.
   */
  data: Prisma.EdicaoEventoCreateManyInput | Prisma.EdicaoEventoCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * EdicaoEvento update
 */
export type EdicaoEventoUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * The data needed to update a EdicaoEvento.
   */
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateInput, Prisma.EdicaoEventoUncheckedUpdateInput>
  /**
   * Choose, which EdicaoEvento to update.
   */
  where: Prisma.EdicaoEventoWhereUniqueInput
}

/**
 * EdicaoEvento updateMany
 */
export type EdicaoEventoUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update EdicaoEventos.
   */
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateManyMutationInput, Prisma.EdicaoEventoUncheckedUpdateManyInput>
  /**
   * Filter which EdicaoEventos to update
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * Limit how many EdicaoEventos to update.
   */
  limit?: number
}

/**
 * EdicaoEvento updateManyAndReturn
 */
export type EdicaoEventoUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * The data used to update EdicaoEventos.
   */
  data: Prisma.XOR<Prisma.EdicaoEventoUpdateManyMutationInput, Prisma.EdicaoEventoUncheckedUpdateManyInput>
  /**
   * Filter which EdicaoEventos to update
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * Limit how many EdicaoEventos to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * EdicaoEvento upsert
 */
export type EdicaoEventoUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * The filter to search for the EdicaoEvento to update in case it exists.
   */
  where: Prisma.EdicaoEventoWhereUniqueInput
  /**
   * In case the EdicaoEvento found by the `where` argument doesn't exist, create a new EdicaoEvento with this data.
   */
  create: Prisma.XOR<Prisma.EdicaoEventoCreateInput, Prisma.EdicaoEventoUncheckedCreateInput>
  /**
   * In case the EdicaoEvento was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.EdicaoEventoUpdateInput, Prisma.EdicaoEventoUncheckedUpdateInput>
}

/**
 * EdicaoEvento delete
 */
export type EdicaoEventoDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
  /**
   * Filter which EdicaoEvento to delete.
   */
  where: Prisma.EdicaoEventoWhereUniqueInput
}

/**
 * EdicaoEvento deleteMany
 */
export type EdicaoEventoDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which EdicaoEventos to delete
   */
  where?: Prisma.EdicaoEventoWhereInput
  /**
   * Limit how many EdicaoEventos to delete.
   */
  limit?: number
}

/**
 * EdicaoEvento.artigos
 */
export type EdicaoEvento$artigosArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Artigo
   */
  select?: Prisma.ArtigoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Artigo
   */
  omit?: Prisma.ArtigoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.ArtigoInclude<ExtArgs> | null
  where?: Prisma.ArtigoWhereInput
  orderBy?: Prisma.ArtigoOrderByWithRelationInput | Prisma.ArtigoOrderByWithRelationInput[]
  cursor?: Prisma.ArtigoWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.ArtigoScalarFieldEnum | Prisma.ArtigoScalarFieldEnum[]
}

/**
 * EdicaoEvento.revisores
 */
export type EdicaoEvento$revisoresArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  where?: Prisma.RevisorWhereInput
  orderBy?: Prisma.RevisorOrderByWithRelationInput | Prisma.RevisorOrderByWithRelationInput[]
  cursor?: Prisma.RevisorWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.RevisorScalarFieldEnum | Prisma.RevisorScalarFieldEnum[]
}

/**
 * EdicaoEvento without action
 */
export type EdicaoEventoDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the EdicaoEvento
   */
  select?: Prisma.EdicaoEventoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the EdicaoEvento
   */
  omit?: Prisma.EdicaoEventoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.EdicaoEventoInclude<ExtArgs> | null
}
