// @prisma/client/models/Pergunta.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Pergunta` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Pergunta
 * 
 */
export type PerguntaModel = runtime.Types.Result.DefaultSelection<Prisma.$PerguntaPayload>

export type AggregatePergunta = {
  _count: PerguntaCountAggregateOutputType | null
  _avg: PerguntaAvgAggregateOutputType | null
  _sum: PerguntaSumAggregateOutputType | null
  _min: PerguntaMinAggregateOutputType | null
  _max: PerguntaMaxAggregateOutputType | null
}

export type PerguntaAvgAggregateOutputType = {
  id: number | null
  artigoId: number | null
}

export type PerguntaSumAggregateOutputType = {
  id: number | null
  artigoId: number | null
}

export type PerguntaMinAggregateOutputType = {
  id: number | null
  texto: string | null
  tipo: string | null
  artigoId: number | null
}

export type PerguntaMaxAggregateOutputType = {
  id: number | null
  texto: string | null
  tipo: string | null
  artigoId: number | null
}

export type PerguntaCountAggregateOutputType = {
  id: number
  texto: number
  tipo: number
  artigoId: number
  _all: number
}


export type PerguntaAvgAggregateInputType = {
  id?: true
  artigoId?: true
}

export type PerguntaSumAggregateInputType = {
  id?: true
  artigoId?: true
}

export type PerguntaMinAggregateInputType = {
  id?: true
  texto?: true
  tipo?: true
  artigoId?: true
}

export type PerguntaMaxAggregateInputType = {
  id?: true
  texto?: true
  tipo?: true
  artigoId?: true
}

export type PerguntaCountAggregateInputType = {
  id?: true
  texto?: true
  tipo?: true
  artigoId?: true
  _all?: true
}

export type PerguntaAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Pergunta to aggregate.
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Perguntas to fetch.
   */
  orderBy?: Prisma.PerguntaOrderByWithRelationInput | Prisma.PerguntaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.PerguntaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Perguntas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Perguntas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Perguntas
  **/
  _count?: true | PerguntaCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: PerguntaAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: PerguntaSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: PerguntaMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: PerguntaMaxAggregateInputType
}

export type GetPerguntaAggregateType<T extends PerguntaAggregateArgs> = {
      [P in keyof T & keyof AggregatePergunta]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregatePergunta[P]>
    : Prisma.GetScalarType<T[P], AggregatePergunta[P]>
}




export type PerguntaGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.PerguntaWhereInput
  orderBy?: Prisma.PerguntaOrderByWithAggregationInput | Prisma.PerguntaOrderByWithAggregationInput[]
  by: Prisma.PerguntaScalarFieldEnum[] | Prisma.PerguntaScalarFieldEnum
  having?: Prisma.PerguntaScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: PerguntaCountAggregateInputType | true
  _avg?: PerguntaAvgAggregateInputType
  _sum?: PerguntaSumAggregateInputType
  _min?: PerguntaMinAggregateInputType
  _max?: PerguntaMaxAggregateInputType
}

export type PerguntaGroupByOutputType = {
  id: number
  texto: string
  tipo: string
  artigoId: number
  _count: PerguntaCountAggregateOutputType | null
  _avg: PerguntaAvgAggregateOutputType | null
  _sum: PerguntaSumAggregateOutputType | null
  _min: PerguntaMinAggregateOutputType | null
  _max: PerguntaMaxAggregateOutputType | null
}

type GetPerguntaGroupByPayload<T extends PerguntaGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<PerguntaGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof PerguntaGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], PerguntaGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], PerguntaGroupByOutputType[P]>
      }
    >
  >



export type PerguntaWhereInput = {
  AND?: Prisma.PerguntaWhereInput | Prisma.PerguntaWhereInput[]
  OR?: Prisma.PerguntaWhereInput[]
  NOT?: Prisma.PerguntaWhereInput | Prisma.PerguntaWhereInput[]
  id?: Prisma.IntFilter<"Pergunta"> | number
  texto?: Prisma.StringFilter<"Pergunta"> | string
  tipo?: Prisma.StringFilter<"Pergunta"> | string
  artigoId?: Prisma.IntFilter<"Pergunta"> | number
  artigo?: Prisma.XOR<Prisma.ArtigoScalarRelationFilter, Prisma.ArtigoWhereInput>
  respostas?: Prisma.RespostaListRelationFilter
}

export type PerguntaOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  texto?: Prisma.SortOrder
  tipo?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  artigo?: Prisma.ArtigoOrderByWithRelationInput
  respostas?: Prisma.RespostaOrderByRelationAggregateInput
}

export type PerguntaWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  AND?: Prisma.PerguntaWhereInput | Prisma.PerguntaWhereInput[]
  OR?: Prisma.PerguntaWhereInput[]
  NOT?: Prisma.PerguntaWhereInput | Prisma.PerguntaWhereInput[]
  texto?: Prisma.StringFilter<"Pergunta"> | string
  tipo?: Prisma.StringFilter<"Pergunta"> | string
  artigoId?: Prisma.IntFilter<"Pergunta"> | number
  artigo?: Prisma.XOR<Prisma.ArtigoScalarRelationFilter, Prisma.ArtigoWhereInput>
  respostas?: Prisma.RespostaListRelationFilter
}, "id">

export type PerguntaOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  texto?: Prisma.SortOrder
  tipo?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
  _count?: Prisma.PerguntaCountOrderByAggregateInput
  _avg?: Prisma.PerguntaAvgOrderByAggregateInput
  _max?: Prisma.PerguntaMaxOrderByAggregateInput
  _min?: Prisma.PerguntaMinOrderByAggregateInput
  _sum?: Prisma.PerguntaSumOrderByAggregateInput
}

export type PerguntaScalarWhereWithAggregatesInput = {
  AND?: Prisma.PerguntaScalarWhereWithAggregatesInput | Prisma.PerguntaScalarWhereWithAggregatesInput[]
  OR?: Prisma.PerguntaScalarWhereWithAggregatesInput[]
  NOT?: Prisma.PerguntaScalarWhereWithAggregatesInput | Prisma.PerguntaScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"Pergunta"> | number
  texto?: Prisma.StringWithAggregatesFilter<"Pergunta"> | string
  tipo?: Prisma.StringWithAggregatesFilter<"Pergunta"> | string
  artigoId?: Prisma.IntWithAggregatesFilter<"Pergunta"> | number
}

export type PerguntaCreateInput = {
  texto: string
  tipo: string
  artigo: Prisma.ArtigoCreateNestedOneWithoutPerguntasInput
  respostas?: Prisma.RespostaCreateNestedManyWithoutPerguntaInput
}

export type PerguntaUncheckedCreateInput = {
  id?: number
  texto: string
  tipo: string
  artigoId: number
  respostas?: Prisma.RespostaUncheckedCreateNestedManyWithoutPerguntaInput
}

export type PerguntaUpdateInput = {
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  artigo?: Prisma.ArtigoUpdateOneRequiredWithoutPerguntasNestedInput
  respostas?: Prisma.RespostaUpdateManyWithoutPerguntaNestedInput
}

export type PerguntaUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
  respostas?: Prisma.RespostaUncheckedUpdateManyWithoutPerguntaNestedInput
}

export type PerguntaCreateManyInput = {
  id?: number
  texto: string
  tipo: string
  artigoId: number
}

export type PerguntaUpdateManyMutationInput = {
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
}

export type PerguntaUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type PerguntaListRelationFilter = {
  every?: Prisma.PerguntaWhereInput
  some?: Prisma.PerguntaWhereInput
  none?: Prisma.PerguntaWhereInput
}

export type PerguntaOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type PerguntaCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  texto?: Prisma.SortOrder
  tipo?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
}

export type PerguntaAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
}

export type PerguntaMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  texto?: Prisma.SortOrder
  tipo?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
}

export type PerguntaMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  texto?: Prisma.SortOrder
  tipo?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
}

export type PerguntaSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  artigoId?: Prisma.SortOrder
}

export type PerguntaScalarRelationFilter = {
  is?: Prisma.PerguntaWhereInput
  isNot?: Prisma.PerguntaWhereInput
}

export type PerguntaCreateNestedManyWithoutArtigoInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput> | Prisma.PerguntaCreateWithoutArtigoInput[] | Prisma.PerguntaUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutArtigoInput | Prisma.PerguntaCreateOrConnectWithoutArtigoInput[]
  createMany?: Prisma.PerguntaCreateManyArtigoInputEnvelope
  connect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
}

export type PerguntaUncheckedCreateNestedManyWithoutArtigoInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput> | Prisma.PerguntaCreateWithoutArtigoInput[] | Prisma.PerguntaUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutArtigoInput | Prisma.PerguntaCreateOrConnectWithoutArtigoInput[]
  createMany?: Prisma.PerguntaCreateManyArtigoInputEnvelope
  connect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
}

export type PerguntaUpdateManyWithoutArtigoNestedInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput> | Prisma.PerguntaCreateWithoutArtigoInput[] | Prisma.PerguntaUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutArtigoInput | Prisma.PerguntaCreateOrConnectWithoutArtigoInput[]
  upsert?: Prisma.PerguntaUpsertWithWhereUniqueWithoutArtigoInput | Prisma.PerguntaUpsertWithWhereUniqueWithoutArtigoInput[]
  createMany?: Prisma.PerguntaCreateManyArtigoInputEnvelope
  set?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  disconnect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  delete?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  connect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  update?: Prisma.PerguntaUpdateWithWhereUniqueWithoutArtigoInput | Prisma.PerguntaUpdateWithWhereUniqueWithoutArtigoInput[]
  updateMany?: Prisma.PerguntaUpdateManyWithWhereWithoutArtigoInput | Prisma.PerguntaUpdateManyWithWhereWithoutArtigoInput[]
  deleteMany?: Prisma.PerguntaScalarWhereInput | Prisma.PerguntaScalarWhereInput[]
}

export type PerguntaUncheckedUpdateManyWithoutArtigoNestedInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput> | Prisma.PerguntaCreateWithoutArtigoInput[] | Prisma.PerguntaUncheckedCreateWithoutArtigoInput[]
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutArtigoInput | Prisma.PerguntaCreateOrConnectWithoutArtigoInput[]
  upsert?: Prisma.PerguntaUpsertWithWhereUniqueWithoutArtigoInput | Prisma.PerguntaUpsertWithWhereUniqueWithoutArtigoInput[]
  createMany?: Prisma.PerguntaCreateManyArtigoInputEnvelope
  set?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  disconnect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  delete?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  connect?: Prisma.PerguntaWhereUniqueInput | Prisma.PerguntaWhereUniqueInput[]
  update?: Prisma.PerguntaUpdateWithWhereUniqueWithoutArtigoInput | Prisma.PerguntaUpdateWithWhereUniqueWithoutArtigoInput[]
  updateMany?: Prisma.PerguntaUpdateManyWithWhereWithoutArtigoInput | Prisma.PerguntaUpdateManyWithWhereWithoutArtigoInput[]
  deleteMany?: Prisma.PerguntaScalarWhereInput | Prisma.PerguntaScalarWhereInput[]
}

export type PerguntaCreateNestedOneWithoutRespostasInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutRespostasInput, Prisma.PerguntaUncheckedCreateWithoutRespostasInput>
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutRespostasInput
  connect?: Prisma.PerguntaWhereUniqueInput
}

export type PerguntaUpdateOneRequiredWithoutRespostasNestedInput = {
  create?: Prisma.XOR<Prisma.PerguntaCreateWithoutRespostasInput, Prisma.PerguntaUncheckedCreateWithoutRespostasInput>
  connectOrCreate?: Prisma.PerguntaCreateOrConnectWithoutRespostasInput
  upsert?: Prisma.PerguntaUpsertWithoutRespostasInput
  connect?: Prisma.PerguntaWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.PerguntaUpdateToOneWithWhereWithoutRespostasInput, Prisma.PerguntaUpdateWithoutRespostasInput>, Prisma.PerguntaUncheckedUpdateWithoutRespostasInput>
}

export type PerguntaCreateWithoutArtigoInput = {
  texto: string
  tipo: string
  respostas?: Prisma.RespostaCreateNestedManyWithoutPerguntaInput
}

export type PerguntaUncheckedCreateWithoutArtigoInput = {
  id?: number
  texto: string
  tipo: string
  respostas?: Prisma.RespostaUncheckedCreateNestedManyWithoutPerguntaInput
}

export type PerguntaCreateOrConnectWithoutArtigoInput = {
  where: Prisma.PerguntaWhereUniqueInput
  create: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput>
}

export type PerguntaCreateManyArtigoInputEnvelope = {
  data: Prisma.PerguntaCreateManyArtigoInput | Prisma.PerguntaCreateManyArtigoInput[]
}

export type PerguntaUpsertWithWhereUniqueWithoutArtigoInput = {
  where: Prisma.PerguntaWhereUniqueInput
  update: Prisma.XOR<Prisma.PerguntaUpdateWithoutArtigoInput, Prisma.PerguntaUncheckedUpdateWithoutArtigoInput>
  create: Prisma.XOR<Prisma.PerguntaCreateWithoutArtigoInput, Prisma.PerguntaUncheckedCreateWithoutArtigoInput>
}

export type PerguntaUpdateWithWhereUniqueWithoutArtigoInput = {
  where: Prisma.PerguntaWhereUniqueInput
  data: Prisma.XOR<Prisma.PerguntaUpdateWithoutArtigoInput, Prisma.PerguntaUncheckedUpdateWithoutArtigoInput>
}

export type PerguntaUpdateManyWithWhereWithoutArtigoInput = {
  where: Prisma.PerguntaScalarWhereInput
  data: Prisma.XOR<Prisma.PerguntaUpdateManyMutationInput, Prisma.PerguntaUncheckedUpdateManyWithoutArtigoInput>
}

export type PerguntaScalarWhereInput = {
  AND?: Prisma.PerguntaScalarWhereInput | Prisma.PerguntaScalarWhereInput[]
  OR?: Prisma.PerguntaScalarWhereInput[]
  NOT?: Prisma.PerguntaScalarWhereInput | Prisma.PerguntaScalarWhereInput[]
  id?: Prisma.IntFilter<"Pergunta"> | number
  texto?: Prisma.StringFilter<"Pergunta"> | string
  tipo?: Prisma.StringFilter<"Pergunta"> | string
  artigoId?: Prisma.IntFilter<"Pergunta"> | number
}

export type PerguntaCreateWithoutRespostasInput = {
  texto: string
  tipo: string
  artigo: Prisma.ArtigoCreateNestedOneWithoutPerguntasInput
}

export type PerguntaUncheckedCreateWithoutRespostasInput = {
  id?: number
  texto: string
  tipo: string
  artigoId: number
}

export type PerguntaCreateOrConnectWithoutRespostasInput = {
  where: Prisma.PerguntaWhereUniqueInput
  create: Prisma.XOR<Prisma.PerguntaCreateWithoutRespostasInput, Prisma.PerguntaUncheckedCreateWithoutRespostasInput>
}

export type PerguntaUpsertWithoutRespostasInput = {
  update: Prisma.XOR<Prisma.PerguntaUpdateWithoutRespostasInput, Prisma.PerguntaUncheckedUpdateWithoutRespostasInput>
  create: Prisma.XOR<Prisma.PerguntaCreateWithoutRespostasInput, Prisma.PerguntaUncheckedCreateWithoutRespostasInput>
  where?: Prisma.PerguntaWhereInput
}

export type PerguntaUpdateToOneWithWhereWithoutRespostasInput = {
  where?: Prisma.PerguntaWhereInput
  data: Prisma.XOR<Prisma.PerguntaUpdateWithoutRespostasInput, Prisma.PerguntaUncheckedUpdateWithoutRespostasInput>
}

export type PerguntaUpdateWithoutRespostasInput = {
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  artigo?: Prisma.ArtigoUpdateOneRequiredWithoutPerguntasNestedInput
}

export type PerguntaUncheckedUpdateWithoutRespostasInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  artigoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type PerguntaCreateManyArtigoInput = {
  id?: number
  texto: string
  tipo: string
}

export type PerguntaUpdateWithoutArtigoInput = {
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  respostas?: Prisma.RespostaUpdateManyWithoutPerguntaNestedInput
}

export type PerguntaUncheckedUpdateWithoutArtigoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
  respostas?: Prisma.RespostaUncheckedUpdateManyWithoutPerguntaNestedInput
}

export type PerguntaUncheckedUpdateManyWithoutArtigoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  texto?: Prisma.StringFieldUpdateOperationsInput | string
  tipo?: Prisma.StringFieldUpdateOperationsInput | string
}


/**
 * Count Type PerguntaCountOutputType
 */

export type PerguntaCountOutputType = {
  respostas: number
}

export type PerguntaCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  respostas?: boolean | PerguntaCountOutputTypeCountRespostasArgs
}

/**
 * PerguntaCountOutputType without action
 */
export type PerguntaCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the PerguntaCountOutputType
   */
  select?: Prisma.PerguntaCountOutputTypeSelect<ExtArgs> | null
}

/**
 * PerguntaCountOutputType without action
 */
export type PerguntaCountOutputTypeCountRespostasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.RespostaWhereInput
}


export type PerguntaSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  texto?: boolean
  tipo?: boolean
  artigoId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  respostas?: boolean | Prisma.Pergunta$respostasArgs<ExtArgs>
  _count?: boolean | Prisma.PerguntaCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["pergunta"]>

export type PerguntaSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  texto?: boolean
  tipo?: boolean
  artigoId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["pergunta"]>

export type PerguntaSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  texto?: boolean
  tipo?: boolean
  artigoId?: boolean
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["pergunta"]>

export type PerguntaSelectScalar = {
  id?: boolean
  texto?: boolean
  tipo?: boolean
  artigoId?: boolean
}

export type PerguntaOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "texto" | "tipo" | "artigoId", ExtArgs["result"]["pergunta"]>
export type PerguntaInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
  respostas?: boolean | Prisma.Pergunta$respostasArgs<ExtArgs>
  _count?: boolean | Prisma.PerguntaCountOutputTypeDefaultArgs<ExtArgs>
}
export type PerguntaIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
}
export type PerguntaIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  artigo?: boolean | Prisma.ArtigoDefaultArgs<ExtArgs>
}

export type $PerguntaPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Pergunta"
  objects: {
    artigo: Prisma.$ArtigoPayload<ExtArgs>
    respostas: Prisma.$RespostaPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    texto: string
    tipo: string
    artigoId: number
  }, ExtArgs["result"]["pergunta"]>
  composites: {}
}

export type PerguntaGetPayload<S extends boolean | null | undefined | PerguntaDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$PerguntaPayload, S>

export type PerguntaCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<PerguntaFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: PerguntaCountAggregateInputType | true
  }

export interface PerguntaDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Pergunta'], meta: { name: 'Pergunta' } }
  /**
   * Find zero or one Pergunta that matches the filter.
   * @param {PerguntaFindUniqueArgs} args - Arguments to find a Pergunta
   * @example
   * // Get one Pergunta
   * const pergunta = await prisma.pergunta.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends PerguntaFindUniqueArgs>(args: Prisma.SelectSubset<T, PerguntaFindUniqueArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Pergunta that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {PerguntaFindUniqueOrThrowArgs} args - Arguments to find a Pergunta
   * @example
   * // Get one Pergunta
   * const pergunta = await prisma.pergunta.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends PerguntaFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, PerguntaFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Pergunta that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaFindFirstArgs} args - Arguments to find a Pergunta
   * @example
   * // Get one Pergunta
   * const pergunta = await prisma.pergunta.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends PerguntaFindFirstArgs>(args?: Prisma.SelectSubset<T, PerguntaFindFirstArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Pergunta that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaFindFirstOrThrowArgs} args - Arguments to find a Pergunta
   * @example
   * // Get one Pergunta
   * const pergunta = await prisma.pergunta.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends PerguntaFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, PerguntaFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Perguntas that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Perguntas
   * const perguntas = await prisma.pergunta.findMany()
   * 
   * // Get first 10 Perguntas
   * const perguntas = await prisma.pergunta.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const perguntaWithIdOnly = await prisma.pergunta.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends PerguntaFindManyArgs>(args?: Prisma.SelectSubset<T, PerguntaFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Pergunta.
   * @param {PerguntaCreateArgs} args - Arguments to create a Pergunta.
   * @example
   * // Create one Pergunta
   * const Pergunta = await prisma.pergunta.create({
   *   data: {
   *     // ... data to create a Pergunta
   *   }
   * })
   * 
   */
  create<T extends PerguntaCreateArgs>(args: Prisma.SelectSubset<T, PerguntaCreateArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Perguntas.
   * @param {PerguntaCreateManyArgs} args - Arguments to create many Perguntas.
   * @example
   * // Create many Perguntas
   * const pergunta = await prisma.pergunta.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends PerguntaCreateManyArgs>(args?: Prisma.SelectSubset<T, PerguntaCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Perguntas and returns the data saved in the database.
   * @param {PerguntaCreateManyAndReturnArgs} args - Arguments to create many Perguntas.
   * @example
   * // Create many Perguntas
   * const pergunta = await prisma.pergunta.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Perguntas and only return the `id`
   * const perguntaWithIdOnly = await prisma.pergunta.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends PerguntaCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, PerguntaCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Pergunta.
   * @param {PerguntaDeleteArgs} args - Arguments to delete one Pergunta.
   * @example
   * // Delete one Pergunta
   * const Pergunta = await prisma.pergunta.delete({
   *   where: {
   *     // ... filter to delete one Pergunta
   *   }
   * })
   * 
   */
  delete<T extends PerguntaDeleteArgs>(args: Prisma.SelectSubset<T, PerguntaDeleteArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Pergunta.
   * @param {PerguntaUpdateArgs} args - Arguments to update one Pergunta.
   * @example
   * // Update one Pergunta
   * const pergunta = await prisma.pergunta.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends PerguntaUpdateArgs>(args: Prisma.SelectSubset<T, PerguntaUpdateArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Perguntas.
   * @param {PerguntaDeleteManyArgs} args - Arguments to filter Perguntas to delete.
   * @example
   * // Delete a few Perguntas
   * const { count } = await prisma.pergunta.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends PerguntaDeleteManyArgs>(args?: Prisma.SelectSubset<T, PerguntaDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Perguntas.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Perguntas
   * const pergunta = await prisma.pergunta.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends PerguntaUpdateManyArgs>(args: Prisma.SelectSubset<T, PerguntaUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Perguntas and returns the data updated in the database.
   * @param {PerguntaUpdateManyAndReturnArgs} args - Arguments to update many Perguntas.
   * @example
   * // Update many Perguntas
   * const pergunta = await prisma.pergunta.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Perguntas and only return the `id`
   * const perguntaWithIdOnly = await prisma.pergunta.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends PerguntaUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, PerguntaUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Pergunta.
   * @param {PerguntaUpsertArgs} args - Arguments to update or create a Pergunta.
   * @example
   * // Update or create a Pergunta
   * const pergunta = await prisma.pergunta.upsert({
   *   create: {
   *     // ... data to create a Pergunta
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Pergunta we want to update
   *   }
   * })
   */
  upsert<T extends PerguntaUpsertArgs>(args: Prisma.SelectSubset<T, PerguntaUpsertArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Perguntas.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaCountArgs} args - Arguments to filter Perguntas to count.
   * @example
   * // Count the number of Perguntas
   * const count = await prisma.pergunta.count({
   *   where: {
   *     // ... the filter for the Perguntas we want to count
   *   }
   * })
  **/
  count<T extends PerguntaCountArgs>(
    args?: Prisma.Subset<T, PerguntaCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], PerguntaCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Pergunta.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends PerguntaAggregateArgs>(args: Prisma.Subset<T, PerguntaAggregateArgs>): Prisma.PrismaPromise<GetPerguntaAggregateType<T>>

  /**
   * Group by Pergunta.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {PerguntaGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends PerguntaGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: PerguntaGroupByArgs['orderBy'] }
      : { orderBy?: PerguntaGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, PerguntaGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetPerguntaGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Pergunta model
 */
readonly fields: PerguntaFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Pergunta.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__PerguntaClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  artigo<T extends Prisma.ArtigoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.ArtigoDefaultArgs<ExtArgs>>): Prisma.Prisma__ArtigoClient<runtime.Types.Result.GetResult<Prisma.$ArtigoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  respostas<T extends Prisma.Pergunta$respostasArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.Pergunta$respostasArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Pergunta model
 */
export interface PerguntaFieldRefs {
  readonly id: Prisma.FieldRef<"Pergunta", 'Int'>
  readonly texto: Prisma.FieldRef<"Pergunta", 'String'>
  readonly tipo: Prisma.FieldRef<"Pergunta", 'String'>
  readonly artigoId: Prisma.FieldRef<"Pergunta", 'Int'>
}
    

// Custom InputTypes
/**
 * Pergunta findUnique
 */
export type PerguntaFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter, which Pergunta to fetch.
   */
  where: Prisma.PerguntaWhereUniqueInput
}

/**
 * Pergunta findUniqueOrThrow
 */
export type PerguntaFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter, which Pergunta to fetch.
   */
  where: Prisma.PerguntaWhereUniqueInput
}

/**
 * Pergunta findFirst
 */
export type PerguntaFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter, which Pergunta to fetch.
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Perguntas to fetch.
   */
  orderBy?: Prisma.PerguntaOrderByWithRelationInput | Prisma.PerguntaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Perguntas.
   */
  cursor?: Prisma.PerguntaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Perguntas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Perguntas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Perguntas.
   */
  distinct?: Prisma.PerguntaScalarFieldEnum | Prisma.PerguntaScalarFieldEnum[]
}

/**
 * Pergunta findFirstOrThrow
 */
export type PerguntaFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter, which Pergunta to fetch.
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Perguntas to fetch.
   */
  orderBy?: Prisma.PerguntaOrderByWithRelationInput | Prisma.PerguntaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Perguntas.
   */
  cursor?: Prisma.PerguntaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Perguntas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Perguntas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Perguntas.
   */
  distinct?: Prisma.PerguntaScalarFieldEnum | Prisma.PerguntaScalarFieldEnum[]
}

/**
 * Pergunta findMany
 */
export type PerguntaFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter, which Perguntas to fetch.
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Perguntas to fetch.
   */
  orderBy?: Prisma.PerguntaOrderByWithRelationInput | Prisma.PerguntaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Perguntas.
   */
  cursor?: Prisma.PerguntaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Perguntas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Perguntas.
   */
  skip?: number
  distinct?: Prisma.PerguntaScalarFieldEnum | Prisma.PerguntaScalarFieldEnum[]
}

/**
 * Pergunta create
 */
export type PerguntaCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * The data needed to create a Pergunta.
   */
  data: Prisma.XOR<Prisma.PerguntaCreateInput, Prisma.PerguntaUncheckedCreateInput>
}

/**
 * Pergunta createMany
 */
export type PerguntaCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Perguntas.
   */
  data: Prisma.PerguntaCreateManyInput | Prisma.PerguntaCreateManyInput[]
}

/**
 * Pergunta createManyAndReturn
 */
export type PerguntaCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * The data used to create many Perguntas.
   */
  data: Prisma.PerguntaCreateManyInput | Prisma.PerguntaCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Pergunta update
 */
export type PerguntaUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * The data needed to update a Pergunta.
   */
  data: Prisma.XOR<Prisma.PerguntaUpdateInput, Prisma.PerguntaUncheckedUpdateInput>
  /**
   * Choose, which Pergunta to update.
   */
  where: Prisma.PerguntaWhereUniqueInput
}

/**
 * Pergunta updateMany
 */
export type PerguntaUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Perguntas.
   */
  data: Prisma.XOR<Prisma.PerguntaUpdateManyMutationInput, Prisma.PerguntaUncheckedUpdateManyInput>
  /**
   * Filter which Perguntas to update
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * Limit how many Perguntas to update.
   */
  limit?: number
}

/**
 * Pergunta updateManyAndReturn
 */
export type PerguntaUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * The data used to update Perguntas.
   */
  data: Prisma.XOR<Prisma.PerguntaUpdateManyMutationInput, Prisma.PerguntaUncheckedUpdateManyInput>
  /**
   * Filter which Perguntas to update
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * Limit how many Perguntas to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Pergunta upsert
 */
export type PerguntaUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * The filter to search for the Pergunta to update in case it exists.
   */
  where: Prisma.PerguntaWhereUniqueInput
  /**
   * In case the Pergunta found by the `where` argument doesn't exist, create a new Pergunta with this data.
   */
  create: Prisma.XOR<Prisma.PerguntaCreateInput, Prisma.PerguntaUncheckedCreateInput>
  /**
   * In case the Pergunta was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.PerguntaUpdateInput, Prisma.PerguntaUncheckedUpdateInput>
}

/**
 * Pergunta delete
 */
export type PerguntaDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
  /**
   * Filter which Pergunta to delete.
   */
  where: Prisma.PerguntaWhereUniqueInput
}

/**
 * Pergunta deleteMany
 */
export type PerguntaDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Perguntas to delete
   */
  where?: Prisma.PerguntaWhereInput
  /**
   * Limit how many Perguntas to delete.
   */
  limit?: number
}

/**
 * Pergunta.respostas
 */
export type Pergunta$respostasArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  where?: Prisma.RespostaWhereInput
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  cursor?: Prisma.RespostaWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.RespostaScalarFieldEnum | Prisma.RespostaScalarFieldEnum[]
}

/**
 * Pergunta without action
 */
export type PerguntaDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Pergunta
   */
  select?: Prisma.PerguntaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Pergunta
   */
  omit?: Prisma.PerguntaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.PerguntaInclude<ExtArgs> | null
}
