// @prisma/client/models/Resposta.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Resposta` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Resposta
 * 
 */
export type RespostaModel = runtime.Types.Result.DefaultSelection<Prisma.$RespostaPayload>

export type AggregateResposta = {
  _count: RespostaCountAggregateOutputType | null
  _avg: RespostaAvgAggregateOutputType | null
  _sum: RespostaSumAggregateOutputType | null
  _min: RespostaMinAggregateOutputType | null
  _max: RespostaMaxAggregateOutputType | null
}

export type RespostaAvgAggregateOutputType = {
  id: number | null
  atribuicaoId: number | null
  perguntaId: number | null
}

export type RespostaSumAggregateOutputType = {
  id: number | null
  atribuicaoId: number | null
  perguntaId: number | null
}

export type RespostaMinAggregateOutputType = {
  id: number | null
  resposta: string | null
  atribuicaoId: number | null
  perguntaId: number | null
}

export type RespostaMaxAggregateOutputType = {
  id: number | null
  resposta: string | null
  atribuicaoId: number | null
  perguntaId: number | null
}

export type RespostaCountAggregateOutputType = {
  id: number
  resposta: number
  atribuicaoId: number
  perguntaId: number
  _all: number
}


export type RespostaAvgAggregateInputType = {
  id?: true
  atribuicaoId?: true
  perguntaId?: true
}

export type RespostaSumAggregateInputType = {
  id?: true
  atribuicaoId?: true
  perguntaId?: true
}

export type RespostaMinAggregateInputType = {
  id?: true
  resposta?: true
  atribuicaoId?: true
  perguntaId?: true
}

export type RespostaMaxAggregateInputType = {
  id?: true
  resposta?: true
  atribuicaoId?: true
  perguntaId?: true
}

export type RespostaCountAggregateInputType = {
  id?: true
  resposta?: true
  atribuicaoId?: true
  perguntaId?: true
  _all?: true
}

export type RespostaAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Resposta to aggregate.
   */
  where?: Prisma.RespostaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Respostas to fetch.
   */
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.RespostaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Respostas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Respostas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Respostas
  **/
  _count?: true | RespostaCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: RespostaAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: RespostaSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: RespostaMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: RespostaMaxAggregateInputType
}

export type GetRespostaAggregateType<T extends RespostaAggregateArgs> = {
      [P in keyof T & keyof AggregateResposta]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateResposta[P]>
    : Prisma.GetScalarType<T[P], AggregateResposta[P]>
}




export type RespostaGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.RespostaWhereInput
  orderBy?: Prisma.RespostaOrderByWithAggregationInput | Prisma.RespostaOrderByWithAggregationInput[]
  by: Prisma.RespostaScalarFieldEnum[] | Prisma.RespostaScalarFieldEnum
  having?: Prisma.RespostaScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: RespostaCountAggregateInputType | true
  _avg?: RespostaAvgAggregateInputType
  _sum?: RespostaSumAggregateInputType
  _min?: RespostaMinAggregateInputType
  _max?: RespostaMaxAggregateInputType
}

export type RespostaGroupByOutputType = {
  id: number
  resposta: string
  atribuicaoId: number
  perguntaId: number
  _count: RespostaCountAggregateOutputType | null
  _avg: RespostaAvgAggregateOutputType | null
  _sum: RespostaSumAggregateOutputType | null
  _min: RespostaMinAggregateOutputType | null
  _max: RespostaMaxAggregateOutputType | null
}

type GetRespostaGroupByPayload<T extends RespostaGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<RespostaGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof RespostaGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], RespostaGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], RespostaGroupByOutputType[P]>
      }
    >
  >



export type RespostaWhereInput = {
  AND?: Prisma.RespostaWhereInput | Prisma.RespostaWhereInput[]
  OR?: Prisma.RespostaWhereInput[]
  NOT?: Prisma.RespostaWhereInput | Prisma.RespostaWhereInput[]
  id?: Prisma.IntFilter<"Resposta"> | number
  resposta?: Prisma.StringFilter<"Resposta"> | string
  atribuicaoId?: Prisma.IntFilter<"Resposta"> | number
  perguntaId?: Prisma.IntFilter<"Resposta"> | number
  atribuicao?: Prisma.XOR<Prisma.AtribuicaoScalarRelationFilter, Prisma.AtribuicaoWhereInput>
  pergunta?: Prisma.XOR<Prisma.PerguntaScalarRelationFilter, Prisma.PerguntaWhereInput>
}

export type RespostaOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  resposta?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
  atribuicao?: Prisma.AtribuicaoOrderByWithRelationInput
  pergunta?: Prisma.PerguntaOrderByWithRelationInput
}

export type RespostaWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  atribuicaoId_perguntaId?: Prisma.RespostaAtribuicaoIdPerguntaIdCompoundUniqueInput
  AND?: Prisma.RespostaWhereInput | Prisma.RespostaWhereInput[]
  OR?: Prisma.RespostaWhereInput[]
  NOT?: Prisma.RespostaWhereInput | Prisma.RespostaWhereInput[]
  resposta?: Prisma.StringFilter<"Resposta"> | string
  atribuicaoId?: Prisma.IntFilter<"Resposta"> | number
  perguntaId?: Prisma.IntFilter<"Resposta"> | number
  atribuicao?: Prisma.XOR<Prisma.AtribuicaoScalarRelationFilter, Prisma.AtribuicaoWhereInput>
  pergunta?: Prisma.XOR<Prisma.PerguntaScalarRelationFilter, Prisma.PerguntaWhereInput>
}, "id" | "atribuicaoId_perguntaId">

export type RespostaOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  resposta?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
  _count?: Prisma.RespostaCountOrderByAggregateInput
  _avg?: Prisma.RespostaAvgOrderByAggregateInput
  _max?: Prisma.RespostaMaxOrderByAggregateInput
  _min?: Prisma.RespostaMinOrderByAggregateInput
  _sum?: Prisma.RespostaSumOrderByAggregateInput
}

export type RespostaScalarWhereWithAggregatesInput = {
  AND?: Prisma.RespostaScalarWhereWithAggregatesInput | Prisma.RespostaScalarWhereWithAggregatesInput[]
  OR?: Prisma.RespostaScalarWhereWithAggregatesInput[]
  NOT?: Prisma.RespostaScalarWhereWithAggregatesInput | Prisma.RespostaScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"Resposta"> | number
  resposta?: Prisma.StringWithAggregatesFilter<"Resposta"> | string
  atribuicaoId?: Prisma.IntWithAggregatesFilter<"Resposta"> | number
  perguntaId?: Prisma.IntWithAggregatesFilter<"Resposta"> | number
}

export type RespostaCreateInput = {
  resposta: string
  atribuicao: Prisma.AtribuicaoCreateNestedOneWithoutRespostasInput
  pergunta: Prisma.PerguntaCreateNestedOneWithoutRespostasInput
}

export type RespostaUncheckedCreateInput = {
  id?: number
  resposta: string
  atribuicaoId: number
  perguntaId: number
}

export type RespostaUpdateInput = {
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicao?: Prisma.AtribuicaoUpdateOneRequiredWithoutRespostasNestedInput
  pergunta?: Prisma.PerguntaUpdateOneRequiredWithoutRespostasNestedInput
}

export type RespostaUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  perguntaId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RespostaCreateManyInput = {
  id?: number
  resposta: string
  atribuicaoId: number
  perguntaId: number
}

export type RespostaUpdateManyMutationInput = {
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
}

export type RespostaUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  perguntaId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RespostaListRelationFilter = {
  every?: Prisma.RespostaWhereInput
  some?: Prisma.RespostaWhereInput
  none?: Prisma.RespostaWhereInput
}

export type RespostaOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type RespostaAtribuicaoIdPerguntaIdCompoundUniqueInput = {
  atribuicaoId: number
  perguntaId: number
}

export type RespostaCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  resposta?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
}

export type RespostaAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
}

export type RespostaMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  resposta?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
}

export type RespostaMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  resposta?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
}

export type RespostaSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  atribuicaoId?: Prisma.SortOrder
  perguntaId?: Prisma.SortOrder
}

export type RespostaCreateNestedManyWithoutAtribuicaoInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput> | Prisma.RespostaCreateWithoutAtribuicaoInput[] | Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput | Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput[]
  createMany?: Prisma.RespostaCreateManyAtribuicaoInputEnvelope
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
}

export type RespostaUncheckedCreateNestedManyWithoutAtribuicaoInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput> | Prisma.RespostaCreateWithoutAtribuicaoInput[] | Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput | Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput[]
  createMany?: Prisma.RespostaCreateManyAtribuicaoInputEnvelope
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
}

export type RespostaUpdateManyWithoutAtribuicaoNestedInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput> | Prisma.RespostaCreateWithoutAtribuicaoInput[] | Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput | Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput[]
  upsert?: Prisma.RespostaUpsertWithWhereUniqueWithoutAtribuicaoInput | Prisma.RespostaUpsertWithWhereUniqueWithoutAtribuicaoInput[]
  createMany?: Prisma.RespostaCreateManyAtribuicaoInputEnvelope
  set?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  disconnect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  delete?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  update?: Prisma.RespostaUpdateWithWhereUniqueWithoutAtribuicaoInput | Prisma.RespostaUpdateWithWhereUniqueWithoutAtribuicaoInput[]
  updateMany?: Prisma.RespostaUpdateManyWithWhereWithoutAtribuicaoInput | Prisma.RespostaUpdateManyWithWhereWithoutAtribuicaoInput[]
  deleteMany?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
}

export type RespostaUncheckedUpdateManyWithoutAtribuicaoNestedInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput> | Prisma.RespostaCreateWithoutAtribuicaoInput[] | Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput | Prisma.RespostaCreateOrConnectWithoutAtribuicaoInput[]
  upsert?: Prisma.RespostaUpsertWithWhereUniqueWithoutAtribuicaoInput | Prisma.RespostaUpsertWithWhereUniqueWithoutAtribuicaoInput[]
  createMany?: Prisma.RespostaCreateManyAtribuicaoInputEnvelope
  set?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  disconnect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  delete?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  update?: Prisma.RespostaUpdateWithWhereUniqueWithoutAtribuicaoInput | Prisma.RespostaUpdateWithWhereUniqueWithoutAtribuicaoInput[]
  updateMany?: Prisma.RespostaUpdateManyWithWhereWithoutAtribuicaoInput | Prisma.RespostaUpdateManyWithWhereWithoutAtribuicaoInput[]
  deleteMany?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
}

export type RespostaCreateNestedManyWithoutPerguntaInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput> | Prisma.RespostaCreateWithoutPerguntaInput[] | Prisma.RespostaUncheckedCreateWithoutPerguntaInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutPerguntaInput | Prisma.RespostaCreateOrConnectWithoutPerguntaInput[]
  createMany?: Prisma.RespostaCreateManyPerguntaInputEnvelope
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
}

export type RespostaUncheckedCreateNestedManyWithoutPerguntaInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput> | Prisma.RespostaCreateWithoutPerguntaInput[] | Prisma.RespostaUncheckedCreateWithoutPerguntaInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutPerguntaInput | Prisma.RespostaCreateOrConnectWithoutPerguntaInput[]
  createMany?: Prisma.RespostaCreateManyPerguntaInputEnvelope
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
}

export type RespostaUpdateManyWithoutPerguntaNestedInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput> | Prisma.RespostaCreateWithoutPerguntaInput[] | Prisma.RespostaUncheckedCreateWithoutPerguntaInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutPerguntaInput | Prisma.RespostaCreateOrConnectWithoutPerguntaInput[]
  upsert?: Prisma.RespostaUpsertWithWhereUniqueWithoutPerguntaInput | Prisma.RespostaUpsertWithWhereUniqueWithoutPerguntaInput[]
  createMany?: Prisma.RespostaCreateManyPerguntaInputEnvelope
  set?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  disconnect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  delete?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  update?: Prisma.RespostaUpdateWithWhereUniqueWithoutPerguntaInput | Prisma.RespostaUpdateWithWhereUniqueWithoutPerguntaInput[]
  updateMany?: Prisma.RespostaUpdateManyWithWhereWithoutPerguntaInput | Prisma.RespostaUpdateManyWithWhereWithoutPerguntaInput[]
  deleteMany?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
}

export type RespostaUncheckedUpdateManyWithoutPerguntaNestedInput = {
  create?: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput> | Prisma.RespostaCreateWithoutPerguntaInput[] | Prisma.RespostaUncheckedCreateWithoutPerguntaInput[]
  connectOrCreate?: Prisma.RespostaCreateOrConnectWithoutPerguntaInput | Prisma.RespostaCreateOrConnectWithoutPerguntaInput[]
  upsert?: Prisma.RespostaUpsertWithWhereUniqueWithoutPerguntaInput | Prisma.RespostaUpsertWithWhereUniqueWithoutPerguntaInput[]
  createMany?: Prisma.RespostaCreateManyPerguntaInputEnvelope
  set?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  disconnect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  delete?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  connect?: Prisma.RespostaWhereUniqueInput | Prisma.RespostaWhereUniqueInput[]
  update?: Prisma.RespostaUpdateWithWhereUniqueWithoutPerguntaInput | Prisma.RespostaUpdateWithWhereUniqueWithoutPerguntaInput[]
  updateMany?: Prisma.RespostaUpdateManyWithWhereWithoutPerguntaInput | Prisma.RespostaUpdateManyWithWhereWithoutPerguntaInput[]
  deleteMany?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
}

export type RespostaCreateWithoutAtribuicaoInput = {
  resposta: string
  pergunta: Prisma.PerguntaCreateNestedOneWithoutRespostasInput
}

export type RespostaUncheckedCreateWithoutAtribuicaoInput = {
  id?: number
  resposta: string
  perguntaId: number
}

export type RespostaCreateOrConnectWithoutAtribuicaoInput = {
  where: Prisma.RespostaWhereUniqueInput
  create: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput>
}

export type RespostaCreateManyAtribuicaoInputEnvelope = {
  data: Prisma.RespostaCreateManyAtribuicaoInput | Prisma.RespostaCreateManyAtribuicaoInput[]
}

export type RespostaUpsertWithWhereUniqueWithoutAtribuicaoInput = {
  where: Prisma.RespostaWhereUniqueInput
  update: Prisma.XOR<Prisma.RespostaUpdateWithoutAtribuicaoInput, Prisma.RespostaUncheckedUpdateWithoutAtribuicaoInput>
  create: Prisma.XOR<Prisma.RespostaCreateWithoutAtribuicaoInput, Prisma.RespostaUncheckedCreateWithoutAtribuicaoInput>
}

export type RespostaUpdateWithWhereUniqueWithoutAtribuicaoInput = {
  where: Prisma.RespostaWhereUniqueInput
  data: Prisma.XOR<Prisma.RespostaUpdateWithoutAtribuicaoInput, Prisma.RespostaUncheckedUpdateWithoutAtribuicaoInput>
}

export type RespostaUpdateManyWithWhereWithoutAtribuicaoInput = {
  where: Prisma.RespostaScalarWhereInput
  data: Prisma.XOR<Prisma.RespostaUpdateManyMutationInput, Prisma.RespostaUncheckedUpdateManyWithoutAtribuicaoInput>
}

export type RespostaScalarWhereInput = {
  AND?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
  OR?: Prisma.RespostaScalarWhereInput[]
  NOT?: Prisma.RespostaScalarWhereInput | Prisma.RespostaScalarWhereInput[]
  id?: Prisma.IntFilter<"Resposta"> | number
  resposta?: Prisma.StringFilter<"Resposta"> | string
  atribuicaoId?: Prisma.IntFilter<"Resposta"> | number
  perguntaId?: Prisma.IntFilter<"Resposta"> | number
}

export type RespostaCreateWithoutPerguntaInput = {
  resposta: string
  atribuicao: Prisma.AtribuicaoCreateNestedOneWithoutRespostasInput
}

export type RespostaUncheckedCreateWithoutPerguntaInput = {
  id?: number
  resposta: string
  atribuicaoId: number
}

export type RespostaCreateOrConnectWithoutPerguntaInput = {
  where: Prisma.RespostaWhereUniqueInput
  create: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput>
}

export type RespostaCreateManyPerguntaInputEnvelope = {
  data: Prisma.RespostaCreateManyPerguntaInput | Prisma.RespostaCreateManyPerguntaInput[]
}

export type RespostaUpsertWithWhereUniqueWithoutPerguntaInput = {
  where: Prisma.RespostaWhereUniqueInput
  update: Prisma.XOR<Prisma.RespostaUpdateWithoutPerguntaInput, Prisma.RespostaUncheckedUpdateWithoutPerguntaInput>
  create: Prisma.XOR<Prisma.RespostaCreateWithoutPerguntaInput, Prisma.RespostaUncheckedCreateWithoutPerguntaInput>
}

export type RespostaUpdateWithWhereUniqueWithoutPerguntaInput = {
  where: Prisma.RespostaWhereUniqueInput
  data: Prisma.XOR<Prisma.RespostaUpdateWithoutPerguntaInput, Prisma.RespostaUncheckedUpdateWithoutPerguntaInput>
}

export type RespostaUpdateManyWithWhereWithoutPerguntaInput = {
  where: Prisma.RespostaScalarWhereInput
  data: Prisma.XOR<Prisma.RespostaUpdateManyMutationInput, Prisma.RespostaUncheckedUpdateManyWithoutPerguntaInput>
}

export type RespostaCreateManyAtribuicaoInput = {
  id?: number
  resposta: string
  perguntaId: number
}

export type RespostaUpdateWithoutAtribuicaoInput = {
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  pergunta?: Prisma.PerguntaUpdateOneRequiredWithoutRespostasNestedInput
}

export type RespostaUncheckedUpdateWithoutAtribuicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  perguntaId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RespostaUncheckedUpdateManyWithoutAtribuicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  perguntaId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RespostaCreateManyPerguntaInput = {
  id?: number
  resposta: string
  atribuicaoId: number
}

export type RespostaUpdateWithoutPerguntaInput = {
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicao?: Prisma.AtribuicaoUpdateOneRequiredWithoutRespostasNestedInput
}

export type RespostaUncheckedUpdateWithoutPerguntaInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicaoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RespostaUncheckedUpdateManyWithoutPerguntaInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  resposta?: Prisma.StringFieldUpdateOperationsInput | string
  atribuicaoId?: Prisma.IntFieldUpdateOperationsInput | number
}



export type RespostaSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  resposta?: boolean
  atribuicaoId?: boolean
  perguntaId?: boolean
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}, ExtArgs["result"]["resposta"]>

export type RespostaSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  resposta?: boolean
  atribuicaoId?: boolean
  perguntaId?: boolean
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}, ExtArgs["result"]["resposta"]>

export type RespostaSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  resposta?: boolean
  atribuicaoId?: boolean
  perguntaId?: boolean
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}, ExtArgs["result"]["resposta"]>

export type RespostaSelectScalar = {
  id?: boolean
  resposta?: boolean
  atribuicaoId?: boolean
  perguntaId?: boolean
}

export type RespostaOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "resposta" | "atribuicaoId" | "perguntaId", ExtArgs["result"]["resposta"]>
export type RespostaInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}
export type RespostaIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}
export type RespostaIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  atribuicao?: boolean | Prisma.AtribuicaoDefaultArgs<ExtArgs>
  pergunta?: boolean | Prisma.PerguntaDefaultArgs<ExtArgs>
}

export type $RespostaPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Resposta"
  objects: {
    atribuicao: Prisma.$AtribuicaoPayload<ExtArgs>
    pergunta: Prisma.$PerguntaPayload<ExtArgs>
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    resposta: string
    atribuicaoId: number
    perguntaId: number
  }, ExtArgs["result"]["resposta"]>
  composites: {}
}

export type RespostaGetPayload<S extends boolean | null | undefined | RespostaDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$RespostaPayload, S>

export type RespostaCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<RespostaFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: RespostaCountAggregateInputType | true
  }

export interface RespostaDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Resposta'], meta: { name: 'Resposta' } }
  /**
   * Find zero or one Resposta that matches the filter.
   * @param {RespostaFindUniqueArgs} args - Arguments to find a Resposta
   * @example
   * // Get one Resposta
   * const resposta = await prisma.resposta.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends RespostaFindUniqueArgs>(args: Prisma.SelectSubset<T, RespostaFindUniqueArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Resposta that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {RespostaFindUniqueOrThrowArgs} args - Arguments to find a Resposta
   * @example
   * // Get one Resposta
   * const resposta = await prisma.resposta.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends RespostaFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, RespostaFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Resposta that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaFindFirstArgs} args - Arguments to find a Resposta
   * @example
   * // Get one Resposta
   * const resposta = await prisma.resposta.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends RespostaFindFirstArgs>(args?: Prisma.SelectSubset<T, RespostaFindFirstArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Resposta that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaFindFirstOrThrowArgs} args - Arguments to find a Resposta
   * @example
   * // Get one Resposta
   * const resposta = await prisma.resposta.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends RespostaFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, RespostaFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Respostas that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Respostas
   * const respostas = await prisma.resposta.findMany()
   * 
   * // Get first 10 Respostas
   * const respostas = await prisma.resposta.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const respostaWithIdOnly = await prisma.resposta.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends RespostaFindManyArgs>(args?: Prisma.SelectSubset<T, RespostaFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Resposta.
   * @param {RespostaCreateArgs} args - Arguments to create a Resposta.
   * @example
   * // Create one Resposta
   * const Resposta = await prisma.resposta.create({
   *   data: {
   *     // ... data to create a Resposta
   *   }
   * })
   * 
   */
  create<T extends RespostaCreateArgs>(args: Prisma.SelectSubset<T, RespostaCreateArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Respostas.
   * @param {RespostaCreateManyArgs} args - Arguments to create many Respostas.
   * @example
   * // Create many Respostas
   * const resposta = await prisma.resposta.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends RespostaCreateManyArgs>(args?: Prisma.SelectSubset<T, RespostaCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Respostas and returns the data saved in the database.
   * @param {RespostaCreateManyAndReturnArgs} args - Arguments to create many Respostas.
   * @example
   * // Create many Respostas
   * const resposta = await prisma.resposta.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Respostas and only return the `id`
   * const respostaWithIdOnly = await prisma.resposta.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends RespostaCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, RespostaCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Resposta.
   * @param {RespostaDeleteArgs} args - Arguments to delete one Resposta.
   * @example
   * // Delete one Resposta
   * const Resposta = await prisma.resposta.delete({
   *   where: {
   *     // ... filter to delete one Resposta
   *   }
   * })
   * 
   */
  delete<T extends RespostaDeleteArgs>(args: Prisma.SelectSubset<T, RespostaDeleteArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Resposta.
   * @param {RespostaUpdateArgs} args - Arguments to update one Resposta.
   * @example
   * // Update one Resposta
   * const resposta = await prisma.resposta.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends RespostaUpdateArgs>(args: Prisma.SelectSubset<T, RespostaUpdateArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Respostas.
   * @param {RespostaDeleteManyArgs} args - Arguments to filter Respostas to delete.
   * @example
   * // Delete a few Respostas
   * const { count } = await prisma.resposta.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends RespostaDeleteManyArgs>(args?: Prisma.SelectSubset<T, RespostaDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Respostas.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Respostas
   * const resposta = await prisma.resposta.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends RespostaUpdateManyArgs>(args: Prisma.SelectSubset<T, RespostaUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Respostas and returns the data updated in the database.
   * @param {RespostaUpdateManyAndReturnArgs} args - Arguments to update many Respostas.
   * @example
   * // Update many Respostas
   * const resposta = await prisma.resposta.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Respostas and only return the `id`
   * const respostaWithIdOnly = await prisma.resposta.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends RespostaUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, RespostaUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Resposta.
   * @param {RespostaUpsertArgs} args - Arguments to update or create a Resposta.
   * @example
   * // Update or create a Resposta
   * const resposta = await prisma.resposta.upsert({
   *   create: {
   *     // ... data to create a Resposta
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Resposta we want to update
   *   }
   * })
   */
  upsert<T extends RespostaUpsertArgs>(args: Prisma.SelectSubset<T, RespostaUpsertArgs<ExtArgs>>): Prisma.Prisma__RespostaClient<runtime.Types.Result.GetResult<Prisma.$RespostaPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Respostas.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaCountArgs} args - Arguments to filter Respostas to count.
   * @example
   * // Count the number of Respostas
   * const count = await prisma.resposta.count({
   *   where: {
   *     // ... the filter for the Respostas we want to count
   *   }
   * })
  **/
  count<T extends RespostaCountArgs>(
    args?: Prisma.Subset<T, RespostaCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], RespostaCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Resposta.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends RespostaAggregateArgs>(args: Prisma.Subset<T, RespostaAggregateArgs>): Prisma.PrismaPromise<GetRespostaAggregateType<T>>

  /**
   * Group by Resposta.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RespostaGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends RespostaGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: RespostaGroupByArgs['orderBy'] }
      : { orderBy?: RespostaGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, RespostaGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetRespostaGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Resposta model
 */
readonly fields: RespostaFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Resposta.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__RespostaClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  atribuicao<T extends Prisma.AtribuicaoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.AtribuicaoDefaultArgs<ExtArgs>>): Prisma.Prisma__AtribuicaoClient<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  pergunta<T extends Prisma.PerguntaDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.PerguntaDefaultArgs<ExtArgs>>): Prisma.Prisma__PerguntaClient<runtime.Types.Result.GetResult<Prisma.$PerguntaPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Resposta model
 */
export interface RespostaFieldRefs {
  readonly id: Prisma.FieldRef<"Resposta", 'Int'>
  readonly resposta: Prisma.FieldRef<"Resposta", 'String'>
  readonly atribuicaoId: Prisma.FieldRef<"Resposta", 'Int'>
  readonly perguntaId: Prisma.FieldRef<"Resposta", 'Int'>
}
    

// Custom InputTypes
/**
 * Resposta findUnique
 */
export type RespostaFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter, which Resposta to fetch.
   */
  where: Prisma.RespostaWhereUniqueInput
}

/**
 * Resposta findUniqueOrThrow
 */
export type RespostaFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter, which Resposta to fetch.
   */
  where: Prisma.RespostaWhereUniqueInput
}

/**
 * Resposta findFirst
 */
export type RespostaFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter, which Resposta to fetch.
   */
  where?: Prisma.RespostaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Respostas to fetch.
   */
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Respostas.
   */
  cursor?: Prisma.RespostaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Respostas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Respostas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Respostas.
   */
  distinct?: Prisma.RespostaScalarFieldEnum | Prisma.RespostaScalarFieldEnum[]
}

/**
 * Resposta findFirstOrThrow
 */
export type RespostaFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter, which Resposta to fetch.
   */
  where?: Prisma.RespostaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Respostas to fetch.
   */
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Respostas.
   */
  cursor?: Prisma.RespostaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Respostas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Respostas.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Respostas.
   */
  distinct?: Prisma.RespostaScalarFieldEnum | Prisma.RespostaScalarFieldEnum[]
}

/**
 * Resposta findMany
 */
export type RespostaFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter, which Respostas to fetch.
   */
  where?: Prisma.RespostaWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Respostas to fetch.
   */
  orderBy?: Prisma.RespostaOrderByWithRelationInput | Prisma.RespostaOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Respostas.
   */
  cursor?: Prisma.RespostaWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Respostas from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Respostas.
   */
  skip?: number
  distinct?: Prisma.RespostaScalarFieldEnum | Prisma.RespostaScalarFieldEnum[]
}

/**
 * Resposta create
 */
export type RespostaCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * The data needed to create a Resposta.
   */
  data: Prisma.XOR<Prisma.RespostaCreateInput, Prisma.RespostaUncheckedCreateInput>
}

/**
 * Resposta createMany
 */
export type RespostaCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Respostas.
   */
  data: Prisma.RespostaCreateManyInput | Prisma.RespostaCreateManyInput[]
}

/**
 * Resposta createManyAndReturn
 */
export type RespostaCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * The data used to create many Respostas.
   */
  data: Prisma.RespostaCreateManyInput | Prisma.RespostaCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Resposta update
 */
export type RespostaUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * The data needed to update a Resposta.
   */
  data: Prisma.XOR<Prisma.RespostaUpdateInput, Prisma.RespostaUncheckedUpdateInput>
  /**
   * Choose, which Resposta to update.
   */
  where: Prisma.RespostaWhereUniqueInput
}

/**
 * Resposta updateMany
 */
export type RespostaUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Respostas.
   */
  data: Prisma.XOR<Prisma.RespostaUpdateManyMutationInput, Prisma.RespostaUncheckedUpdateManyInput>
  /**
   * Filter which Respostas to update
   */
  where?: Prisma.RespostaWhereInput
  /**
   * Limit how many Respostas to update.
   */
  limit?: number
}

/**
 * Resposta updateManyAndReturn
 */
export type RespostaUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * The data used to update Respostas.
   */
  data: Prisma.XOR<Prisma.RespostaUpdateManyMutationInput, Prisma.RespostaUncheckedUpdateManyInput>
  /**
   * Filter which Respostas to update
   */
  where?: Prisma.RespostaWhereInput
  /**
   * Limit how many Respostas to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Resposta upsert
 */
export type RespostaUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * The filter to search for the Resposta to update in case it exists.
   */
  where: Prisma.RespostaWhereUniqueInput
  /**
   * In case the Resposta found by the `where` argument doesn't exist, create a new Resposta with this data.
   */
  create: Prisma.XOR<Prisma.RespostaCreateInput, Prisma.RespostaUncheckedCreateInput>
  /**
   * In case the Resposta was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.RespostaUpdateInput, Prisma.RespostaUncheckedUpdateInput>
}

/**
 * Resposta delete
 */
export type RespostaDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
  /**
   * Filter which Resposta to delete.
   */
  where: Prisma.RespostaWhereUniqueInput
}

/**
 * Resposta deleteMany
 */
export type RespostaDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Respostas to delete
   */
  where?: Prisma.RespostaWhereInput
  /**
   * Limit how many Respostas to delete.
   */
  limit?: number
}

/**
 * Resposta without action
 */
export type RespostaDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Resposta
   */
  select?: Prisma.RespostaSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Resposta
   */
  omit?: Prisma.RespostaOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RespostaInclude<ExtArgs> | null
}
