// @prisma/client/models/Revisor.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Revisor` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Revisor
 * 
 */
export type RevisorModel = runtime.Types.Result.DefaultSelection<Prisma.$RevisorPayload>

export type AggregateRevisor = {
  _count: RevisorCountAggregateOutputType | null
  _avg: RevisorAvgAggregateOutputType | null
  _sum: RevisorSumAggregateOutputType | null
  _min: RevisorMinAggregateOutputType | null
  _max: RevisorMaxAggregateOutputType | null
}

export type RevisorAvgAggregateOutputType = {
  id: number | null
  edicaoId: number | null
}

export type RevisorSumAggregateOutputType = {
  id: number | null
  edicaoId: number | null
}

export type RevisorMinAggregateOutputType = {
  id: number | null
  nome: string | null
  edicaoId: number | null
}

export type RevisorMaxAggregateOutputType = {
  id: number | null
  nome: string | null
  edicaoId: number | null
}

export type RevisorCountAggregateOutputType = {
  id: number
  nome: number
  areas: number
  edicaoId: number
  _all: number
}


export type RevisorAvgAggregateInputType = {
  id?: true
  edicaoId?: true
}

export type RevisorSumAggregateInputType = {
  id?: true
  edicaoId?: true
}

export type RevisorMinAggregateInputType = {
  id?: true
  nome?: true
  edicaoId?: true
}

export type RevisorMaxAggregateInputType = {
  id?: true
  nome?: true
  edicaoId?: true
}

export type RevisorCountAggregateInputType = {
  id?: true
  nome?: true
  areas?: true
  edicaoId?: true
  _all?: true
}

export type RevisorAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Revisor to aggregate.
   */
  where?: Prisma.RevisorWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Revisors to fetch.
   */
  orderBy?: Prisma.RevisorOrderByWithRelationInput | Prisma.RevisorOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.RevisorWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Revisors from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Revisors.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Revisors
  **/
  _count?: true | RevisorCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to average
  **/
  _avg?: RevisorAvgAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to sum
  **/
  _sum?: RevisorSumAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: RevisorMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: RevisorMaxAggregateInputType
}

export type GetRevisorAggregateType<T extends RevisorAggregateArgs> = {
      [P in keyof T & keyof AggregateRevisor]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateRevisor[P]>
    : Prisma.GetScalarType<T[P], AggregateRevisor[P]>
}




export type RevisorGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.RevisorWhereInput
  orderBy?: Prisma.RevisorOrderByWithAggregationInput | Prisma.RevisorOrderByWithAggregationInput[]
  by: Prisma.RevisorScalarFieldEnum[] | Prisma.RevisorScalarFieldEnum
  having?: Prisma.RevisorScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: RevisorCountAggregateInputType | true
  _avg?: RevisorAvgAggregateInputType
  _sum?: RevisorSumAggregateInputType
  _min?: RevisorMinAggregateInputType
  _max?: RevisorMaxAggregateInputType
}

export type RevisorGroupByOutputType = {
  id: number
  nome: string
  areas: runtime.JsonValue
  edicaoId: number
  _count: RevisorCountAggregateOutputType | null
  _avg: RevisorAvgAggregateOutputType | null
  _sum: RevisorSumAggregateOutputType | null
  _min: RevisorMinAggregateOutputType | null
  _max: RevisorMaxAggregateOutputType | null
}

type GetRevisorGroupByPayload<T extends RevisorGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<RevisorGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof RevisorGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], RevisorGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], RevisorGroupByOutputType[P]>
      }
    >
  >



export type RevisorWhereInput = {
  AND?: Prisma.RevisorWhereInput | Prisma.RevisorWhereInput[]
  OR?: Prisma.RevisorWhereInput[]
  NOT?: Prisma.RevisorWhereInput | Prisma.RevisorWhereInput[]
  id?: Prisma.IntFilter<"Revisor"> | number
  nome?: Prisma.StringFilter<"Revisor"> | string
  areas?: Prisma.JsonFilter<"Revisor">
  edicaoId?: Prisma.IntFilter<"Revisor"> | number
  edicao?: Prisma.XOR<Prisma.EdicaoEventoScalarRelationFilter, Prisma.EdicaoEventoWhereInput>
  atribuicoes?: Prisma.AtribuicaoListRelationFilter
}

export type RevisorOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  nome?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
  edicao?: Prisma.EdicaoEventoOrderByWithRelationInput
  atribuicoes?: Prisma.AtribuicaoOrderByRelationAggregateInput
}

export type RevisorWhereUniqueInput = Prisma.AtLeast<{
  id?: number
  AND?: Prisma.RevisorWhereInput | Prisma.RevisorWhereInput[]
  OR?: Prisma.RevisorWhereInput[]
  NOT?: Prisma.RevisorWhereInput | Prisma.RevisorWhereInput[]
  nome?: Prisma.StringFilter<"Revisor"> | string
  areas?: Prisma.JsonFilter<"Revisor">
  edicaoId?: Prisma.IntFilter<"Revisor"> | number
  edicao?: Prisma.XOR<Prisma.EdicaoEventoScalarRelationFilter, Prisma.EdicaoEventoWhereInput>
  atribuicoes?: Prisma.AtribuicaoListRelationFilter
}, "id">

export type RevisorOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  nome?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
  _count?: Prisma.RevisorCountOrderByAggregateInput
  _avg?: Prisma.RevisorAvgOrderByAggregateInput
  _max?: Prisma.RevisorMaxOrderByAggregateInput
  _min?: Prisma.RevisorMinOrderByAggregateInput
  _sum?: Prisma.RevisorSumOrderByAggregateInput
}

export type RevisorScalarWhereWithAggregatesInput = {
  AND?: Prisma.RevisorScalarWhereWithAggregatesInput | Prisma.RevisorScalarWhereWithAggregatesInput[]
  OR?: Prisma.RevisorScalarWhereWithAggregatesInput[]
  NOT?: Prisma.RevisorScalarWhereWithAggregatesInput | Prisma.RevisorScalarWhereWithAggregatesInput[]
  id?: Prisma.IntWithAggregatesFilter<"Revisor"> | number
  nome?: Prisma.StringWithAggregatesFilter<"Revisor"> | string
  areas?: Prisma.JsonWithAggregatesFilter<"Revisor">
  edicaoId?: Prisma.IntWithAggregatesFilter<"Revisor"> | number
}

export type RevisorCreateInput = {
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicao: Prisma.EdicaoEventoCreateNestedOneWithoutRevisoresInput
  atribuicoes?: Prisma.AtribuicaoCreateNestedManyWithoutRevisorInput
}

export type RevisorUncheckedCreateInput = {
  id?: number
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId: number
  atribuicoes?: Prisma.AtribuicaoUncheckedCreateNestedManyWithoutRevisorInput
}

export type RevisorUpdateInput = {
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicao?: Prisma.EdicaoEventoUpdateOneRequiredWithoutRevisoresNestedInput
  atribuicoes?: Prisma.AtribuicaoUpdateManyWithoutRevisorNestedInput
}

export type RevisorUncheckedUpdateInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
  atribuicoes?: Prisma.AtribuicaoUncheckedUpdateManyWithoutRevisorNestedInput
}

export type RevisorCreateManyInput = {
  id?: number
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId: number
}

export type RevisorUpdateManyMutationInput = {
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
}

export type RevisorUncheckedUpdateManyInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RevisorListRelationFilter = {
  every?: Prisma.RevisorWhereInput
  some?: Prisma.RevisorWhereInput
  none?: Prisma.RevisorWhereInput
}

export type RevisorOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type RevisorCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  nome?: Prisma.SortOrder
  areas?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type RevisorAvgOrderByAggregateInput = {
  id?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type RevisorMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  nome?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type RevisorMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  nome?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type RevisorSumOrderByAggregateInput = {
  id?: Prisma.SortOrder
  edicaoId?: Prisma.SortOrder
}

export type RevisorScalarRelationFilter = {
  is?: Prisma.RevisorWhereInput
  isNot?: Prisma.RevisorWhereInput
}

export type RevisorCreateNestedManyWithoutEdicaoInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput> | Prisma.RevisorCreateWithoutEdicaoInput[] | Prisma.RevisorUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutEdicaoInput | Prisma.RevisorCreateOrConnectWithoutEdicaoInput[]
  createMany?: Prisma.RevisorCreateManyEdicaoInputEnvelope
  connect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
}

export type RevisorUncheckedCreateNestedManyWithoutEdicaoInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput> | Prisma.RevisorCreateWithoutEdicaoInput[] | Prisma.RevisorUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutEdicaoInput | Prisma.RevisorCreateOrConnectWithoutEdicaoInput[]
  createMany?: Prisma.RevisorCreateManyEdicaoInputEnvelope
  connect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
}

export type RevisorUpdateManyWithoutEdicaoNestedInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput> | Prisma.RevisorCreateWithoutEdicaoInput[] | Prisma.RevisorUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutEdicaoInput | Prisma.RevisorCreateOrConnectWithoutEdicaoInput[]
  upsert?: Prisma.RevisorUpsertWithWhereUniqueWithoutEdicaoInput | Prisma.RevisorUpsertWithWhereUniqueWithoutEdicaoInput[]
  createMany?: Prisma.RevisorCreateManyEdicaoInputEnvelope
  set?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  disconnect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  delete?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  connect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  update?: Prisma.RevisorUpdateWithWhereUniqueWithoutEdicaoInput | Prisma.RevisorUpdateWithWhereUniqueWithoutEdicaoInput[]
  updateMany?: Prisma.RevisorUpdateManyWithWhereWithoutEdicaoInput | Prisma.RevisorUpdateManyWithWhereWithoutEdicaoInput[]
  deleteMany?: Prisma.RevisorScalarWhereInput | Prisma.RevisorScalarWhereInput[]
}

export type RevisorUncheckedUpdateManyWithoutEdicaoNestedInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput> | Prisma.RevisorCreateWithoutEdicaoInput[] | Prisma.RevisorUncheckedCreateWithoutEdicaoInput[]
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutEdicaoInput | Prisma.RevisorCreateOrConnectWithoutEdicaoInput[]
  upsert?: Prisma.RevisorUpsertWithWhereUniqueWithoutEdicaoInput | Prisma.RevisorUpsertWithWhereUniqueWithoutEdicaoInput[]
  createMany?: Prisma.RevisorCreateManyEdicaoInputEnvelope
  set?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  disconnect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  delete?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  connect?: Prisma.RevisorWhereUniqueInput | Prisma.RevisorWhereUniqueInput[]
  update?: Prisma.RevisorUpdateWithWhereUniqueWithoutEdicaoInput | Prisma.RevisorUpdateWithWhereUniqueWithoutEdicaoInput[]
  updateMany?: Prisma.RevisorUpdateManyWithWhereWithoutEdicaoInput | Prisma.RevisorUpdateManyWithWhereWithoutEdicaoInput[]
  deleteMany?: Prisma.RevisorScalarWhereInput | Prisma.RevisorScalarWhereInput[]
}

export type RevisorCreateNestedOneWithoutAtribuicoesInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutAtribuicoesInput, Prisma.RevisorUncheckedCreateWithoutAtribuicoesInput>
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutAtribuicoesInput
  connect?: Prisma.RevisorWhereUniqueInput
}

export type RevisorUpdateOneRequiredWithoutAtribuicoesNestedInput = {
  create?: Prisma.XOR<Prisma.RevisorCreateWithoutAtribuicoesInput, Prisma.RevisorUncheckedCreateWithoutAtribuicoesInput>
  connectOrCreate?: Prisma.RevisorCreateOrConnectWithoutAtribuicoesInput
  upsert?: Prisma.RevisorUpsertWithoutAtribuicoesInput
  connect?: Prisma.RevisorWhereUniqueInput
  update?: Prisma.XOR<Prisma.XOR<Prisma.RevisorUpdateToOneWithWhereWithoutAtribuicoesInput, Prisma.RevisorUpdateWithoutAtribuicoesInput>, Prisma.RevisorUncheckedUpdateWithoutAtribuicoesInput>
}

export type RevisorCreateWithoutEdicaoInput = {
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  atribuicoes?: Prisma.AtribuicaoCreateNestedManyWithoutRevisorInput
}

export type RevisorUncheckedCreateWithoutEdicaoInput = {
  id?: number
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  atribuicoes?: Prisma.AtribuicaoUncheckedCreateNestedManyWithoutRevisorInput
}

export type RevisorCreateOrConnectWithoutEdicaoInput = {
  where: Prisma.RevisorWhereUniqueInput
  create: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput>
}

export type RevisorCreateManyEdicaoInputEnvelope = {
  data: Prisma.RevisorCreateManyEdicaoInput | Prisma.RevisorCreateManyEdicaoInput[]
}

export type RevisorUpsertWithWhereUniqueWithoutEdicaoInput = {
  where: Prisma.RevisorWhereUniqueInput
  update: Prisma.XOR<Prisma.RevisorUpdateWithoutEdicaoInput, Prisma.RevisorUncheckedUpdateWithoutEdicaoInput>
  create: Prisma.XOR<Prisma.RevisorCreateWithoutEdicaoInput, Prisma.RevisorUncheckedCreateWithoutEdicaoInput>
}

export type RevisorUpdateWithWhereUniqueWithoutEdicaoInput = {
  where: Prisma.RevisorWhereUniqueInput
  data: Prisma.XOR<Prisma.RevisorUpdateWithoutEdicaoInput, Prisma.RevisorUncheckedUpdateWithoutEdicaoInput>
}

export type RevisorUpdateManyWithWhereWithoutEdicaoInput = {
  where: Prisma.RevisorScalarWhereInput
  data: Prisma.XOR<Prisma.RevisorUpdateManyMutationInput, Prisma.RevisorUncheckedUpdateManyWithoutEdicaoInput>
}

export type RevisorScalarWhereInput = {
  AND?: Prisma.RevisorScalarWhereInput | Prisma.RevisorScalarWhereInput[]
  OR?: Prisma.RevisorScalarWhereInput[]
  NOT?: Prisma.RevisorScalarWhereInput | Prisma.RevisorScalarWhereInput[]
  id?: Prisma.IntFilter<"Revisor"> | number
  nome?: Prisma.StringFilter<"Revisor"> | string
  areas?: Prisma.JsonFilter<"Revisor">
  edicaoId?: Prisma.IntFilter<"Revisor"> | number
}

export type RevisorCreateWithoutAtribuicoesInput = {
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicao: Prisma.EdicaoEventoCreateNestedOneWithoutRevisoresInput
}

export type RevisorUncheckedCreateWithoutAtribuicoesInput = {
  id?: number
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId: number
}

export type RevisorCreateOrConnectWithoutAtribuicoesInput = {
  where: Prisma.RevisorWhereUniqueInput
  create: Prisma.XOR<Prisma.RevisorCreateWithoutAtribuicoesInput, Prisma.RevisorUncheckedCreateWithoutAtribuicoesInput>
}

export type RevisorUpsertWithoutAtribuicoesInput = {
  update: Prisma.XOR<Prisma.RevisorUpdateWithoutAtribuicoesInput, Prisma.RevisorUncheckedUpdateWithoutAtribuicoesInput>
  create: Prisma.XOR<Prisma.RevisorCreateWithoutAtribuicoesInput, Prisma.RevisorUncheckedCreateWithoutAtribuicoesInput>
  where?: Prisma.RevisorWhereInput
}

export type RevisorUpdateToOneWithWhereWithoutAtribuicoesInput = {
  where?: Prisma.RevisorWhereInput
  data: Prisma.XOR<Prisma.RevisorUpdateWithoutAtribuicoesInput, Prisma.RevisorUncheckedUpdateWithoutAtribuicoesInput>
}

export type RevisorUpdateWithoutAtribuicoesInput = {
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicao?: Prisma.EdicaoEventoUpdateOneRequiredWithoutRevisoresNestedInput
}

export type RevisorUncheckedUpdateWithoutAtribuicoesInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  edicaoId?: Prisma.IntFieldUpdateOperationsInput | number
}

export type RevisorCreateManyEdicaoInput = {
  id?: number
  nome: string
  areas: Prisma.JsonNullValueInput | runtime.InputJsonValue
}

export type RevisorUpdateWithoutEdicaoInput = {
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  atribuicoes?: Prisma.AtribuicaoUpdateManyWithoutRevisorNestedInput
}

export type RevisorUncheckedUpdateWithoutEdicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
  atribuicoes?: Prisma.AtribuicaoUncheckedUpdateManyWithoutRevisorNestedInput
}

export type RevisorUncheckedUpdateManyWithoutEdicaoInput = {
  id?: Prisma.IntFieldUpdateOperationsInput | number
  nome?: Prisma.StringFieldUpdateOperationsInput | string
  areas?: Prisma.JsonNullValueInput | runtime.InputJsonValue
}


/**
 * Count Type RevisorCountOutputType
 */

export type RevisorCountOutputType = {
  atribuicoes: number
}

export type RevisorCountOutputTypeSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  atribuicoes?: boolean | RevisorCountOutputTypeCountAtribuicoesArgs
}

/**
 * RevisorCountOutputType without action
 */
export type RevisorCountOutputTypeDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the RevisorCountOutputType
   */
  select?: Prisma.RevisorCountOutputTypeSelect<ExtArgs> | null
}

/**
 * RevisorCountOutputType without action
 */
export type RevisorCountOutputTypeCountAtribuicoesArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.AtribuicaoWhereInput
}


export type RevisorSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  nome?: boolean
  areas?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
  atribuicoes?: boolean | Prisma.Revisor$atribuicoesArgs<ExtArgs>
  _count?: boolean | Prisma.RevisorCountOutputTypeDefaultArgs<ExtArgs>
}, ExtArgs["result"]["revisor"]>

export type RevisorSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  nome?: boolean
  areas?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["revisor"]>

export type RevisorSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  nome?: boolean
  areas?: boolean
  edicaoId?: boolean
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}, ExtArgs["result"]["revisor"]>

export type RevisorSelectScalar = {
  id?: boolean
  nome?: boolean
  areas?: boolean
  edicaoId?: boolean
}

export type RevisorOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "nome" | "areas" | "edicaoId", ExtArgs["result"]["revisor"]>
export type RevisorInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
  atribuicoes?: boolean | Prisma.Revisor$atribuicoesArgs<ExtArgs>
  _count?: boolean | Prisma.RevisorCountOutputTypeDefaultArgs<ExtArgs>
}
export type RevisorIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}
export type RevisorIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  edicao?: boolean | Prisma.EdicaoEventoDefaultArgs<ExtArgs>
}

export type $RevisorPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Revisor"
  objects: {
    edicao: Prisma.$EdicaoEventoPayload<ExtArgs>
    atribuicoes: Prisma.$AtribuicaoPayload<ExtArgs>[]
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: number
    nome: string
    areas: runtime.JsonValue
    edicaoId: number
  }, ExtArgs["result"]["revisor"]>
  composites: {}
}

export type RevisorGetPayload<S extends boolean | null | undefined | RevisorDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$RevisorPayload, S>

export type RevisorCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<RevisorFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: RevisorCountAggregateInputType | true
  }

export interface RevisorDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Revisor'], meta: { name: 'Revisor' } }
  /**
   * Find zero or one Revisor that matches the filter.
   * @param {RevisorFindUniqueArgs} args - Arguments to find a Revisor
   * @example
   * // Get one Revisor
   * const revisor = await prisma.revisor.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends RevisorFindUniqueArgs>(args: Prisma.SelectSubset<T, RevisorFindUniqueArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Revisor that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {RevisorFindUniqueOrThrowArgs} args - Arguments to find a Revisor
   * @example
   * // Get one Revisor
   * const revisor = await prisma.revisor.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends RevisorFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, RevisorFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Revisor that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorFindFirstArgs} args - Arguments to find a Revisor
   * @example
   * // Get one Revisor
   * const revisor = await prisma.revisor.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends RevisorFindFirstArgs>(args?: Prisma.SelectSubset<T, RevisorFindFirstArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Revisor that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorFindFirstOrThrowArgs} args - Arguments to find a Revisor
   * @example
   * // Get one Revisor
   * const revisor = await prisma.revisor.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends RevisorFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, RevisorFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Revisors that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Revisors
   * const revisors = await prisma.revisor.findMany()
   * 
   * // Get first 10 Revisors
   * const revisors = await prisma.revisor.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const revisorWithIdOnly = await prisma.revisor.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends RevisorFindManyArgs>(args?: Prisma.SelectSubset<T, RevisorFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Revisor.
   * @param {RevisorCreateArgs} args - Arguments to create a Revisor.
   * @example
   * // Create one Revisor
   * const Revisor = await prisma.revisor.create({
   *   data: {
   *     // ... data to create a Revisor
   *   }
   * })
   * 
   */
  create<T extends RevisorCreateArgs>(args: Prisma.SelectSubset<T, RevisorCreateArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Revisors.
   * @param {RevisorCreateManyArgs} args - Arguments to create many Revisors.
   * @example
   * // Create many Revisors
   * const revisor = await prisma.revisor.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends RevisorCreateManyArgs>(args?: Prisma.SelectSubset<T, RevisorCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Revisors and returns the data saved in the database.
   * @param {RevisorCreateManyAndReturnArgs} args - Arguments to create many Revisors.
   * @example
   * // Create many Revisors
   * const revisor = await prisma.revisor.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Revisors and only return the `id`
   * const revisorWithIdOnly = await prisma.revisor.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends RevisorCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, RevisorCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Revisor.
   * @param {RevisorDeleteArgs} args - Arguments to delete one Revisor.
   * @example
   * // Delete one Revisor
   * const Revisor = await prisma.revisor.delete({
   *   where: {
   *     // ... filter to delete one Revisor
   *   }
   * })
   * 
   */
  delete<T extends RevisorDeleteArgs>(args: Prisma.SelectSubset<T, RevisorDeleteArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Revisor.
   * @param {RevisorUpdateArgs} args - Arguments to update one Revisor.
   * @example
   * // Update one Revisor
   * const revisor = await prisma.revisor.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends RevisorUpdateArgs>(args: Prisma.SelectSubset<T, RevisorUpdateArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Revisors.
   * @param {RevisorDeleteManyArgs} args - Arguments to filter Revisors to delete.
   * @example
   * // Delete a few Revisors
   * const { count } = await prisma.revisor.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends RevisorDeleteManyArgs>(args?: Prisma.SelectSubset<T, RevisorDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Revisors.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Revisors
   * const revisor = await prisma.revisor.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends RevisorUpdateManyArgs>(args: Prisma.SelectSubset<T, RevisorUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Revisors and returns the data updated in the database.
   * @param {RevisorUpdateManyAndReturnArgs} args - Arguments to update many Revisors.
   * @example
   * // Update many Revisors
   * const revisor = await prisma.revisor.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Revisors and only return the `id`
   * const revisorWithIdOnly = await prisma.revisor.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends RevisorUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, RevisorUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Revisor.
   * @param {RevisorUpsertArgs} args - Arguments to update or create a Revisor.
   * @example
   * // Update or create a Revisor
   * const revisor = await prisma.revisor.upsert({
   *   create: {
   *     // ... data to create a Revisor
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Revisor we want to update
   *   }
   * })
   */
  upsert<T extends RevisorUpsertArgs>(args: Prisma.SelectSubset<T, RevisorUpsertArgs<ExtArgs>>): Prisma.Prisma__RevisorClient<runtime.Types.Result.GetResult<Prisma.$RevisorPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Revisors.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorCountArgs} args - Arguments to filter Revisors to count.
   * @example
   * // Count the number of Revisors
   * const count = await prisma.revisor.count({
   *   where: {
   *     // ... the filter for the Revisors we want to count
   *   }
   * })
  **/
  count<T extends RevisorCountArgs>(
    args?: Prisma.Subset<T, RevisorCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], RevisorCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Revisor.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends RevisorAggregateArgs>(args: Prisma.Subset<T, RevisorAggregateArgs>): Prisma.PrismaPromise<GetRevisorAggregateType<T>>

  /**
   * Group by Revisor.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {RevisorGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends RevisorGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: RevisorGroupByArgs['orderBy'] }
      : { orderBy?: RevisorGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, RevisorGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetRevisorGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Revisor model
 */
readonly fields: RevisorFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Revisor.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__RevisorClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  edicao<T extends Prisma.EdicaoEventoDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.EdicaoEventoDefaultArgs<ExtArgs>>): Prisma.Prisma__EdicaoEventoClient<runtime.Types.Result.GetResult<Prisma.$EdicaoEventoPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  atribuicoes<T extends Prisma.Revisor$atribuicoesArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.Revisor$atribuicoesArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$AtribuicaoPayload<ExtArgs>, T, "findMany", GlobalOmitOptions> | Null>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Revisor model
 */
export interface RevisorFieldRefs {
  readonly id: Prisma.FieldRef<"Revisor", 'Int'>
  readonly nome: Prisma.FieldRef<"Revisor", 'String'>
  readonly areas: Prisma.FieldRef<"Revisor", 'Json'>
  readonly edicaoId: Prisma.FieldRef<"Revisor", 'Int'>
}
    

// Custom InputTypes
/**
 * Revisor findUnique
 */
export type RevisorFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter, which Revisor to fetch.
   */
  where: Prisma.RevisorWhereUniqueInput
}

/**
 * Revisor findUniqueOrThrow
 */
export type RevisorFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter, which Revisor to fetch.
   */
  where: Prisma.RevisorWhereUniqueInput
}

/**
 * Revisor findFirst
 */
export type RevisorFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter, which Revisor to fetch.
   */
  where?: Prisma.RevisorWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Revisors to fetch.
   */
  orderBy?: Prisma.RevisorOrderByWithRelationInput | Prisma.RevisorOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Revisors.
   */
  cursor?: Prisma.RevisorWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Revisors from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Revisors.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Revisors.
   */
  distinct?: Prisma.RevisorScalarFieldEnum | Prisma.RevisorScalarFieldEnum[]
}

/**
 * Revisor findFirstOrThrow
 */
export type RevisorFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter, which Revisor to fetch.
   */
  where?: Prisma.RevisorWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Revisors to fetch.
   */
  orderBy?: Prisma.RevisorOrderByWithRelationInput | Prisma.RevisorOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Revisors.
   */
  cursor?: Prisma.RevisorWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Revisors from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Revisors.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Revisors.
   */
  distinct?: Prisma.RevisorScalarFieldEnum | Prisma.RevisorScalarFieldEnum[]
}

/**
 * Revisor findMany
 */
export type RevisorFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter, which Revisors to fetch.
   */
  where?: Prisma.RevisorWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Revisors to fetch.
   */
  orderBy?: Prisma.RevisorOrderByWithRelationInput | Prisma.RevisorOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Revisors.
   */
  cursor?: Prisma.RevisorWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Revisors from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Revisors.
   */
  skip?: number
  distinct?: Prisma.RevisorScalarFieldEnum | Prisma.RevisorScalarFieldEnum[]
}

/**
 * Revisor create
 */
export type RevisorCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * The data needed to create a Revisor.
   */
  data: Prisma.XOR<Prisma.RevisorCreateInput, Prisma.RevisorUncheckedCreateInput>
}

/**
 * Revisor createMany
 */
export type RevisorCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Revisors.
   */
  data: Prisma.RevisorCreateManyInput | Prisma.RevisorCreateManyInput[]
}

/**
 * Revisor createManyAndReturn
 */
export type RevisorCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * The data used to create many Revisors.
   */
  data: Prisma.RevisorCreateManyInput | Prisma.RevisorCreateManyInput[]
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Revisor update
 */
export type RevisorUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * The data needed to update a Revisor.
   */
  data: Prisma.XOR<Prisma.RevisorUpdateInput, Prisma.RevisorUncheckedUpdateInput>
  /**
   * Choose, which Revisor to update.
   */
  where: Prisma.RevisorWhereUniqueInput
}

/**
 * Revisor updateMany
 */
export type RevisorUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Revisors.
   */
  data: Prisma.XOR<Prisma.RevisorUpdateManyMutationInput, Prisma.RevisorUncheckedUpdateManyInput>
  /**
   * Filter which Revisors to update
   */
  where?: Prisma.RevisorWhereInput
  /**
   * Limit how many Revisors to update.
   */
  limit?: number
}

/**
 * Revisor updateManyAndReturn
 */
export type RevisorUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * The data used to update Revisors.
   */
  data: Prisma.XOR<Prisma.RevisorUpdateManyMutationInput, Prisma.RevisorUncheckedUpdateManyInput>
  /**
   * Filter which Revisors to update
   */
  where?: Prisma.RevisorWhereInput
  /**
   * Limit how many Revisors to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Revisor upsert
 */
export type RevisorUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * The filter to search for the Revisor to update in case it exists.
   */
  where: Prisma.RevisorWhereUniqueInput
  /**
   * In case the Revisor found by the `where` argument doesn't exist, create a new Revisor with this data.
   */
  create: Prisma.XOR<Prisma.RevisorCreateInput, Prisma.RevisorUncheckedCreateInput>
  /**
   * In case the Revisor was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.RevisorUpdateInput, Prisma.RevisorUncheckedUpdateInput>
}

/**
 * Revisor delete
 */
export type RevisorDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
  /**
   * Filter which Revisor to delete.
   */
  where: Prisma.RevisorWhereUniqueInput
}

/**
 * Revisor deleteMany
 */
export type RevisorDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Revisors to delete
   */
  where?: Prisma.RevisorWhereInput
  /**
   * Limit how many Revisors to delete.
   */
  limit?: number
}

/**
 * Revisor.atribuicoes
 */
export type Revisor$atribuicoesArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Atribuicao
   */
  select?: Prisma.AtribuicaoSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Atribuicao
   */
  omit?: Prisma.AtribuicaoOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.AtribuicaoInclude<ExtArgs> | null
  where?: Prisma.AtribuicaoWhereInput
  orderBy?: Prisma.AtribuicaoOrderByWithRelationInput | Prisma.AtribuicaoOrderByWithRelationInput[]
  cursor?: Prisma.AtribuicaoWhereUniqueInput
  take?: number
  skip?: number
  distinct?: Prisma.AtribuicaoScalarFieldEnum | Prisma.AtribuicaoScalarFieldEnum[]
}

/**
 * Revisor without action
 */
export type RevisorDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Revisor
   */
  select?: Prisma.RevisorSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Revisor
   */
  omit?: Prisma.RevisorOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.RevisorInclude<ExtArgs> | null
}
